{
 "cells": [
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "import plotly.graph_objects as go\n",
    "\n",
    "# Sample data: taxonomic composition of multiple microbiome samples\n",
    "microbiomes['Sample1'] = {\n",
    "    'None;Other': 0.075742067,\n",
    "    'Euryarchaeota': 0.049116324,\n",
    "    'Actinobacteria': 0.004214359,\n",
    "    'Bacteroidetes': 0.084114591,\n",
    "    'Chlamydiae': 5.78339E-05,\n",
    "    'Chloroflexi': 0.007700276,\n",
    "    'Cyanobacteria': 0.005267594,\n",
    "    'Elusimicrobia': 0.000758561,\n",
    "    'Fibrobacteres': 0.010849173,\n",
    "    'Firmicutes': 0.704400609,\n",
    "    'Fusobacteria': 2.9331E-05,\n",
    "    'Lentisphaerae': 0.006111405,\n",
    "    'NA': 0.003288403,\n",
    "    'Nitrospirae': 3.57752E-05,\n",
    "    'Planctomycetes': 0.009601286,\n",
    "    'Proteobacteria': 0.006894289,\n",
    "    'Saccharibacteria': 1.51341E-05,\n",
    "    'Spirochaetae': 0.001372058,\n",
    "    'Synergistetes': 0.019067822,\n",
    "    'Tenericutes': 0.004268445,\n",
    "    'Verrucomicrobia': 0.007094664\n",
    "}\n",
    "\n",
    "# Create a pie chart for each sample\n",
    "for sample in microbiomes:\n",
    "    # Get the taxonomic composition of the sample\n",
    "    composition = microbiomes[sample]\n",
    "\n",
    "    # Create a list of labels and values for the pie chart\n",
    "    labels = []\n",
    "    values = []\n",
    "    for domain in composition:\n",
    "        labels.append('{}:'.format(domain))\n",
    "        values.append(composition[domain])\n",
    "\n",
    "    # Create a pie chart using Plotly\n",
    "    fig = go.Figure(data=[go.Pie(labels=labels, values=values)])\n",
    "    fig.update_layout(title_text='Total Taxonomic Composition', legend=dict(orientation=\"h\", x=-1, y=.8))\n",
    "    fig.show()\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": [
    "import plotly.graph_objects as go\n",
    "\n",
    "# Sample data: taxonomic composition of multiple microbiome samples\n",
    "microbiomes = {\n",
    "    'Sample1': {\n",
    "        'Other': 0.063926449,\n",
    "        'Euryarchaeota': 0.052744073,\n",
    "        'Actinobacteria': 0.004597809,\n",
    "        'Bacteroidetes': 0.069751442,\n",
    "        'Chlamydiae': 4.25673E-05,\n",
    "        'Chloroflexi': 0.007159006,\n",
    "        'Cyanobacteria': 0.005425458,\n",
    "        'Elusimicrobia': 0.000692495,\n",
    "        'Fibrobacteres': 0.009562012,\n",
    "        'Firmicutes': 0.731381689,\n",
    "        'Fusobacteria': 4.7717E-05,\n",
    "        'Lentisphaerae': 0.005097325,\n",
    "        'NA': 0.002225507,\n",
    "        'Nitrospirae': 3.89827E-05,\n",
    "        'Planctomycetes': 0.010283447,\n",
    "        'Proteobacteria': 0.006749815,\n",
    "        'Saccharibacteria': 2.46308E-05,\n",
    "        'Spirochaetae': 0.001025642,\n",
    "        'Synergistetes': 0.021118209,\n",
    "        'Tenericutes': 0.002160078,\n",
    "        'Verrucomicrobia': 0.005945647,\n",
    "    },\n",
    "}\n",
    "plt.figure(dpi=1200)\n",
    "# Create a pie chart for each sample\n",
    "for sample in microbiomes:\n",
    "    # Get the taxonomic composition of the sample\n",
    "    composition = microbiomes[sample]\n",
    "\n",
    "    # Create a list of labels and values for the pie chart\n",
    "    labels = []\n",
    "    values = []\n",
    "    for domain in composition:\n",
    "        percentage = composition[domain] * 100\n",
    "        labels.append('{}: {:.2f}%'.format(domain, percentage))\n",
    "        values.append(composition[domain])\n",
    "\n",
    "    # Create a pie chart using Plotly\n",
    "    fig = go.Figure(data=[go.Pie(labels=labels, values=values)])\n",
    "    fig.update_layout(title_text='Total Taxonomic Composition', legend=dict(orientation=\"h\", x=-1, y=.8))\n",
    "    fig.show()\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": "2 + 2",
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "source": [
    "import numpy as np\n",
    "\n",
    "# Your values\n",
    "values = np.array([0.057767409, 0.047604777, 0.044120724, 0.042986011, 0.069458867, 0.05613369, 0.075359312, 0.058313877, 0.033352441, 0.049190009, 0.01630713, 0.020719511, 0.057951034, 0.073817385, 0.069040461, 0.036791537, 0.07614939, 0.023336887, 0.024526143, 0.039956005, 0.068994538, 0.090589498, 0.099250023, 0.02920478, 0.028913358, 0.034781928])\n",
    "\n",
    "# Standard error\n",
    "standard_error = np.std(values, ddof=1) / np.sqrt(len(values))\n",
    "print(f\"Standard error: {standard_error}\")\n",
    "\n",
    "# Standard deviation\n",
    "standard_deviation = np.std(values, ddof=1)  # ddof=1 calculates sample standard deviation\n",
    "print(f\"Standard deviation: {standard_deviation}\")\n",
    "\n",
    "# Range\n",
    "range_values = np.ptp(values)  # ptp (peak-to-peak) function returns the range of values (maximum - minimum)\n",
    "print(f\"Range: {range_values}\")\n",
    "\n",
    "# Variance\n",
    "variance = np.var(values, ddof=1)  # ddof=1 calculates sample variance\n",
    "print(f\"Variance: {variance}\")\n"
   ],
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "source": [
    "import plotly.graph_objects as go\n",
    "import numpy as np\n",
    "\n",
    "# Creating random data for the plot\n",
    "np.random.seed(0)\n",
    "x, y, z = np.random.multivariate_normal(np.array([0,0,0]), np.eye(3), 200).transpose()\n",
    "\n",
    "# Creating the figure\n",
    "fig = go.Figure(data=[go.Scatter3d(\n",
    "    x=x,\n",
    "    y=y,\n",
    "    z=z,\n",
    "    mode='markers',\n",
    "    marker=dict(\n",
    "        size=8,\n",
    "        color=z,                # set color to an array/list of desired values\n",
    "        colorscale='Viridis',    # choose a colorscale\n",
    "        opacity=0.8\n",
    "    )\n",
    ")])\n",
    "\n",
    "# Adding labels and title\n",
    "fig.update_layout(\n",
    "    scene = dict(\n",
    "        xaxis_title='X',\n",
    "        yaxis_title='Y',\n",
    "        zaxis_title='Z'),\n",
    "    width=700,\n",
    "    margin=dict(r=20, b=10, l=10, t=10))\n",
    "\n",
    "fig.show()\n"
   ],
   "metadata": {
    "collapsed": false,
    "is_executing": true
   },
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "source": [
    "\n",
    "import plotly.graph_objects as go\n",
    "\n",
    "# Sample data: taxonomic composition of multiple microbiome samples\n",
    "microbiomes['Sample1'] = {\n",
    "        'None;Other': 0.075742067,\n",
    "        'Euryarchaeota': 0.049116324,\n",
    "        'Actinobacteria': 0.004214359,\n",
    "        'Bacteroidetes': 0.084114591,\n",
    "        'Chlamydiae': 5.78339E-05,\n",
    "        'Chloroflexi': 0.007700276,\n",
    "        'Cyanobacteria': 0.005267594,\n",
    "        'Elusimicrobia': 0.000758561,\n",
    "        'Fibrobacteres': 0.010849173,\n",
    "        'Firmicutes': 0.704400609,\n",
    "        'Fusobacteria': 2.9331E-05,\n",
    "        'Lentisphaerae': 0.006111405,\n",
    "        'NA': 0.003288403,\n",
    "        'Nitrospirae': 3.57752E-05,\n",
    "        'Planctomycetes': 0.009601286,\n",
    "        'Proteobacteria': 0.006894289,\n",
    "        'Saccharibacteria': 1.51341E-05,\n",
    "        'Spirochaetae': 0.001372058,\n",
    "        'Synergistetes': 0.019067822,\n",
    "        'Tenericutes': 0.004268445,\n",
    "        'Verrucomicrobia': 0.007094664\n",
    "    }\n",
    "\n",
    "plt.figure(dpi=1200)\n",
    "# Create a pie chart for each sample\n",
    "for sample in microbiomes:\n",
    "    # Get the taxonomic composition of the sample\n",
    "    composition = microbiomes[sample]\n",
    "\n",
    "    # Create a list of labels and values for the pie chart\n",
    "    labels = []\n",
    "    values = []\n",
    "    for domain in composition:\n",
    "        percentage = composition[domain] * 100\n",
    "        labels.append('{}: {:.2f}%'.format(domain, percentage))\n",
    "        values.append(composition[domain])\n",
    "\n",
    "    # Create a pie chart using Plotly\n",
    "    fig = go.Figure(data=[go.Pie(labels=labels, values=values)])\n",
    "    fig.update_layout(title_text='Total Taxonomic Composition', legend=dict(orientation=\"h\", x=-1, y=.8))\n",
    "    fig.show()\n"
   ],
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "!conda --version\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "# Load the data\n",
    "otu_table = pd.read_excel(\"/Users/dimitriosgiakoumas/Downloads/Micro/Phylum_Merged.xlsx\")\n",
    "\n",
    "\n",
    "plt.figure(dpi=600)\n",
    "# Transform the data to long format for plotting\n",
    "otu_table_long = otu_table.melt(id_vars=\"OTU\", var_name=\"T\", value_name=\"Relative Abundance\")\n",
    "\n",
    "# Convert the values in Relative Abundance column to percentage\n",
    "otu_table_long['Relative Abundance'] = otu_table_long['Relative Abundance'] * 100\n",
    "\n",
    "# Generate the plot\n",
    "fig, ax = plt.subplots()\n",
    "\n",
    "# Generate a stacked bar plot\n",
    "otu_table_long.pivot_table(index='T', columns='OTU', values='Relative Abundance', fill_value=0).plot(kind='bar', stacked=True, ax=ax, width=.8)\n",
    "\n",
    "# Set y-axis range to 0-100%\n",
    "plt.ylim([0, 100])\n",
    "\n",
    "# Rotate x-axis labels\n",
    "plt.xticks(rotation=90)\n",
    "\n",
    "# Set the x-axis label\n",
    "ax.set_xlabel(\"Sample ID\")\n",
    "\n",
    "# Set the y-axis label\n",
    "ax.set_ylabel(\"Relative Abundance (%)\")\n",
    "\n",
    "# Move the legend to the bottom\n",
    "ax.legend(loc='lower center', bbox_to_anchor=(0.5, -.75), fancybox=True, shadow=True, ncol=5)\n",
    "\n",
    "plt.show()\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "source": [
    "import pandas as pd\n",
    "import plotly.express as px\n",
    "\n",
    "\n",
    "# Example OTU table with 16 samples\n",
    "otu_table = pd.DataFrame({\n",
    "    'OTU': ['Other', 'Euryarchaeota', 'Actinobacteria', 'Bacteroidetes', 'Chlamydiae', 'Chloroflexi', 'Cyanobacteria', 'Elusimicrobia', 'Fibrobacteres', 'Firmicutes', 'Fusobacteria', 'Lentisphaerae', 'NA', 'Nitrospirae', 'Planctomycetes', 'Proteobacteria', 'Saccharibacteria', 'Spirochaetae', 'Synergistetes', 'Tenericutes', 'Verrucomicrobia'],\n",
    "    'T20': [0.0692213560163, 0.0512763463036, 0.00552812411751, 0.0460905208931, 0.0, 0.00294376887991, 0.00377384324431, 0.000573350334169, 0.0061956065961, 0.772371359867, 0.000119804547438, 0.00142053963391, 0.00117237307136, 0.0, 0.00855746767417, 0.00939609950623, 0.0, 0.000650367543237, 0.0173887743139, 0.00139486723089, 0.00192543022669],\n",
    "    'T107': [0.0886399156762, 0.0700123577945, 0.00510667684367, 0.0883764038818, 0.0, 0.00936829862247, 0.00912296005525, 0.0, 0.0122851015883, 0.665294588013, 0.000208992112819, 0.00439792098281, 0.000508850361647, 0.000118125976811, 0.00885944826082, 0.00822338530876, 0.0, 0.000954094428089, 0.0118398575219, 0.00336204703231, 0.0133209755388],\n",
    "    'T203': [0.091516034342, 0.0439206338697, 0.00434792204639, 0.0828312255843, 0.0, 0.0188483524245, 0.00508728950098, 0.00070626144916, 0.0218609989185, 0.655399589486, 0.0, 0.00456862874926, 0.00194221898519, 0.0, 0.0204815820256, 0.00940210554195, 0.0, 0.00150080557947, 0.0326094153479, 0.00264848043435, 0.0023284557152],\n",
    "    'T11': [0.0387414787625, 0.053864708967, 0.00231777661248, 0.0659884635553, 0.0, 0.0028736234924, 0.00484530676455, 0.0, 0.00624016780283, 0.765977975878, 0.0, 0.00684845306765, 0.00293654955427, 0.0, 0.011295228107, 0.011777661248, 0.0, 0.00378605138962, 0.019979024646, 0.000964866282119, 0.00156266386995],\n",
    "    'T212': [0.0662810452229, 0.0916511609093, 0.00777445190519, 0.0822587169323, 0.0, 0.013558773562, 0.00245125798884, 0.00040449801796, 0.00329261386619, 0.667834317612, 0.000226518890057, 0.00792816115201, 0.00574387185503, 0.00020224900898, 0.0129358466144, 0.00594612086401, 0.000275058652213, 0.000493487581911, 0.02409999191, 0.0011892241728, 0.0054526332821],\n",
    "    'T85.1': [0.0510423403203, 0.0536196599079, 0.00868299920113, 0.070557690113, 0.0, 0.000551603454179, 0.00618176284856, 0.000266291322707, 0.00414653631072, 0.739947502568, 0.0, 0.00664777266329, 0.00207326815536, 0.0, 0.0124586297409, 0.00714231369118, 0.0, 0.000884467607563, 0.0186594133983, 0.00222543462548, 0.0149123140716],\n",
    "    'T65': [0.0730575036454, 0.0383787483509, 0.00480106337602, 0.0652210572259, 0.0, 0.00343216514071, 0.0076578944758, 0.00488041979546, 0.0331114660107, 0.728412574025, 0.0, 0.00436460306911, 0.0, 0.0, 0.00758845760879, 0.00395790141949, 0.000119034629158, 0.000247988810745, 0.0178452748212, 0.00344208469314, 0.00348176290286],\n",
    "    'T196': [0.0808211256746, 0.0581534309946, 0.0115940632228, 0.0606399383192, 0.0, 0.0177525057826, 0.00546453353894, 0.00122397841172, 0.00293947571318, 0.692530840401, 0.0, 0.00149383191981, 0.000857748650732, 0.00011565150347, 0.0143022359291, 0.00536815728604, 0.0, 0.000395142636854, 0.0429163454125, 0.00146491904395, 0.00196607555898],\n",
    "    'T99': [0.0524611702876, 0.0459035240017, 0.00296795082352, 0.0678522828586, 0.000129041340153, 0.00292102669983, 0.0018886959786, 0.000164234432922, 0.00140772371076, 0.785157899676, 0.0, 0.00446952278166, 0.00170099948383, 0.000187696494768, 0.00476279855474, 0.00735535638872, 0.0, 0.00082117216461, 0.0105931209235, 0.00208812350429, 0.00716765989395],\n",
    "    'T85.2': [0.0522599376322, 0.0657281622997, 0.00296605613104, 0.0922613713753, 0.00012545252518, 0.00324384386537, 0.0172945266855, 0.0018817878777, 0.00709702856733, 0.685409871322, 0.0, 0.00685508441163, 0.00260762034482, 0.0, 0.0146958672354, 0.00730312914441, 0.0, 0.000143374314492, 0.0252159575612, 0.00278683823793, 0.0121240904692],\n",
    "    'TAK': [0.0483987105493, 0.0419514572786, 0.00371104784482, 0.0433915653975, 0.0, 0.00776550608723, 0.00360027029722, 0.0, 0.00356703703293, 0.806272224745, 0.0, 0.00254788359495, 0.0023374062545, 0.0, 0.00550564411605, 0.00521762249227, 0.0, 0.000919453645135, 0.020504924062, 0.00177244076171, 0.00253680584019],\n",
    "    'T85.3': [0.090085008662, 0.0447403408404, 0.00357895867746, 0.0645421215906, 0.000154439117414, 0.00243073741321, 0.00817184373447, 0.000389455165653, 0.0104078535649, 0.723305534292, 0.000208157071297, 0.0052173562709, 0.000490176329184, 0.0, 0.00637229227939, 0.00932677974296, 0.0, 0.000705048144716, 0.0151014597854, 0.00427057733371, 0.0105018599842],\n",
    "    'T195': [0.0565088200213, 0.0267216305007, 0.00165310008579, 0.0622214317102, 0.0, 0.00525225470297, 0.00124505639373, 0.0, 0.0188223231288, 0.795946765992, 0.0, 0.00277260457427, 0.00187281591998, 0.0, 0.00486513632844, 0.002008830484, 0.0, 0.00269936596288, 0.00947916884639, 0.00177865199104, 0.00615204335726],\n",
    "    'T205': [0.0745835897329, 0.0343140833759, 0.00274304070756, 0.0786094973873, 0.0, 0.0101794970744, 0.000500631003661, 0.0, 0.014841623296, 0.743635206875, 0.0, 0.00377559215261, 0.0032958207741, 0.0, 0.00808310474661, 0.00238842707997, 0.0, 0.00171048926251, 0.0154048331751, 0.00132458619719, 0.00460997715871],\n",
    "    'T159': [0.0588736891759, 0.0825048078668, 0.00325664937044, 0.0711836423673, 0.00027214340143, 0.0111215936718, 0.00707572843717, 0.000589644036431, 0.00576036866359, 0.679269929968, 0.0, 0.0143963859356, 0.00654958452774, 0.0, 0.0098788054719, 0.0094978047099, 0.0, 0.000353786421859, 0.0329021372328, 0.00219529010487, 0.00431800863602],\n",
    "    'TAA': [0.0303314571607, 0.0411641204324, 0.00253506655946, 0.0739971410703, 0.0, 0.00230054498347, 0.0024457250067, 0.0, 0.0010162601626, 0.795340838024, 0.0, 0.00385285446261, 0.00151880639686, 0.0, 0.0138926114536, 0.00368533905119, 0.0, 0.000145180023229, 0.0233516483516, 0.00165281872599, 0.00276958813544]}\n",
    ")\n",
    "otu_table.iloc[:, 1:] *= 100\n",
    "plt.figure(dpi=600)\n",
    "# Calculate average relative abundance for each OTU\n",
    "otu_table['Average'] = otu_table.iloc[:, 1:].mean(axis=1)\n",
    "\n",
    "# Melt DataFrame\n",
    "df_melted = pd.melt(otu_table, id_vars=['OTU'], var_name='T', value_name='Relative Abundance')\n",
    "\n",
    "# Set y-axis range to 0-100%\n",
    "#fig.update_yaxes(range=[0, 100])\n",
    "fig = px.bar(df_melted, x='T', y='Relative Abundance', color='OTU',\n",
    "             title='', barmode='stack',\n",
    "             color_discrete_sequence=px.colors.qualitative.Alphabet)\n",
    "fig.update_layout(\n",
    "    xaxis_title=\"Sample ID\"\n",
    ")\n",
    "# Update layout\n",
    "fig.update_layout(legend=dict(orientation=\"h\", yanchor=\"bottom\", y=1.02, xanchor=\"right\", x=1),\n",
    "                  margin=dict(t=100, b=50, l=50, r=50))\n",
    "fig.update_layout(plot_bgcolor='rgba(255, 255, 255, 1)')\n",
    "# Show plot\n",
    "fig.show()\n"
   ],
   "metadata": {
    "collapsed": false,
    "is_executing": true
   },
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "source": [
    "import pandas as pd\n",
    "import plotly.express as px\n",
    "\n",
    "\n",
    "# Example OTU table with 16 samples\n",
    "otu_table = pd.read_excel(\"/Users/dimitriosgiakoumas/Downloads/Micro/Phylum_Merged.xlsx\")\n",
    "\n",
    "otu_table.iloc[:, 1:] *= 100\n",
    "plt.figure(dpi=1200)\n",
    "# Calculate average relative abundance for each OTU\n",
    "#otu_table['Average'] = otu_table.iloc[:, 1:].mean(axis=1)\n",
    "\n",
    "# Melt DataFrame\n",
    "df_melted = pd.melt(otu_table, id_vars=['OTU'], var_name='T', value_name='Relative Abundance')\n",
    "\n",
    "# Set y-axis range to 0-100%\n",
    "#fig.update_yaxes(range=[0, 100])\n",
    "fig = px.bar(df_melted, x='T', y='Relative Abundance', color='OTU',\n",
    "             title='', barmode='stack',\n",
    "             color_discrete_sequence=px.colors.qualitative.Alphabet)\n",
    "fig.update_layout(\n",
    "    xaxis_title=\"Sample ID\"\n",
    ")\n",
    "# Update layout\n",
    "fig.update_layout(legend=dict(orientation=\"h\", yanchor=\"bottom\", y=1.02, xanchor=\"right\", x=1),\n",
    "                  margin=dict(t=100, b=50, l=50, r=50))\n",
    "fig.update_layout(plot_bgcolor='rgba(255, 255, 255, 1)')\n",
    "# Show plot\n",
    "fig.show()\n"
   ],
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "source": [
    "import pandas as pd\n",
    "import plotly.express as px\n",
    "\n",
    "# Load the data\n",
    "otu_table = pd.read_excel(\"/Users/dimitriosgiakoumas/Downloads/Micro/Phylum_Merged.xlsx\")\n",
    "\n",
    "# Calculate average relative abundance for each OTU\n",
    "#otu_table['Average'] = otu_table.iloc[:, 1:].mean(axis=1)\n",
    "\n",
    "# Transform the data to long format for plotting\n",
    "otu_table_long = otu_table.melt(id_vars=\"OTU\", var_name=\"T\", value_name=\"Relative Abundance\")\n",
    "\n",
    "# Convert the values in Relative Abundance column to percentage\n",
    "otu_table_long['Relative Abundance'] = otu_table_long['Relative Abundance'] * 100\n",
    "\n",
    "# Pivot the table for plotly\n",
    "pivot_table = otu_table_long.pivot_table(index='T', columns='OTU', values='Relative Abundance', fill_value=0)\n",
    "\n",
    "# Generate the plot with plotly\n",
    "fig = px.bar(pivot_table, labels={\"value\": \"Relative Abundance (%)\", \"variable\": \"OTU\", \"index\": \"Sample ID\"},\n",
    "             #title=\"Sample ID vs Relative Abundance\",\n",
    "             orientation='v', barmode='stack')\n",
    "\n",
    "# Update layout\n",
    "fig.update_layout(\n",
    "    xaxis=dict(tickangle=-45),\n",
    "    yaxis=dict(range=[0,100]),\n",
    "    legend=dict(orientation=\"h\", yanchor=\"bottom\", y=1.02, xanchor=\"right\", x=1)\n",
    ")\n",
    "\n",
    "fig.show()\n"
   ],
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "`import plotly.graph_objects as go\n",
    "\n",
    "# Sample data: taxonomic composition of multiple microbiome samples\n",
    "microbiomes = {\n",
    "    'Sample1': {\n",
    "        'Other': 0.063926449,\n",
    "        'Euryarchaeota': 0.052744073,\n",
    "        'Actinobacteria': 0.004597809,\n",
    "        'Bacteroidetes': 0.069751442,\n",
    "        'Chlamydiae': 4.25673E-05,\n",
    "        'Chloroflexi': 0.007159006,\n",
    "        'Cyanobacteria': 0.005425458,\n",
    "        'Elusimicrobia': 0.000692495,\n",
    "        'Fibrobacteres': 0.009562012,\n",
    "        'Firmicutes': 0.731381689,\n",
    "        'Fusobacteria': 4.7717E-05,\n",
    "        'Lentisphaerae': 0.005097325,\n",
    "        'NA': 0.002225507,\n",
    "        'Nitrospirae': 3.89827E-05,\n",
    "        'Planctomycetes': 0.010283447,\n",
    "        'Proteobacteria': 0.006749815,\n",
    "        'Saccharibacteria': 2.46308E-05,\n",
    "        'Spirochaetae': 0.001025642,\n",
    "        'Synergistetes': 0.021118209,\n",
    "        'Tenericutes': 0.002160078,\n",
    "        'Verrucomicrobia': 0.005945647,\n",
    "    },\n",
    "}\n",
    "plt.figure(dpi=1200)\n",
    "# Create a pie chart for each sample\n",
    "for sample in microbiomes:\n",
    "    # Get the taxonomic composition of the sample\n",
    "    composition = microbiomes[sample]\n",
    "\n",
    "    # Create a list of labels and values for the pie chart\n",
    "    labels = []\n",
    "    values = []\n",
    "    for domain in composition:\n",
    "        percentage = composition[domain] * 100\n",
    "        labels.append('{}: {:.2f}%'.format(domain, percentage))\n",
    "        values.append(composition[domain])\n",
    "\n",
    "    # Create a pie chart using Plotly\n",
    "    fig = go.Figure(data=[go.Pie(labels=labels, values=values)])\n",
    "    fig.update_layout(title_text='Total Taxonomic Composition', legend=dict(orientation=\"h\", x=-1, y=.8))\n",
    "    fig.show()\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "import plotly.subplots as sp\n",
    "import plotly.graph_objects as go\n",
    "\n",
    "# Sample data: taxonomic composition of multiple microbiome samples\n",
    "microbiomes = {\n",
    "    'Sample1': {\n",
    "        'Bacteria': {\n",
    "            'Firmicutes': 0.35,\n",
    "            'Bacteroidetes': 0.25,\n",
    "            'Proteobacteria': 0.20,\n",
    "            'Actinobacteria': 0.10,\n",
    "            'Fusobacteria': 0.05,\n",
    "            'Other': 0.05,\n",
    "        },\n",
    "        'Archaea': {\n",
    "            'Euryarchaeota': 0.60,\n",
    "            'Crenarchaeota': 0.20,\n",
    "            'Other': 0.20,\n",
    "        },\n",
    "    },\n",
    "    'Sample2': {\n",
    "        'Bacteria': {\n",
    "            'Firmicutes': 0.40,\n",
    "            'Bacteroidetes': 0.20,\n",
    "            'Proteobacteria': 0.15,\n",
    "            'Actinobacteria': 0.10,\n",
    "            'Fusobacteria': 0.10,\n",
    "            'Other': 0.05,\n",
    "        },\n",
    "        'Archaea': {\n",
    "            'Euryarchaeota': 0.70,\n",
    "            'Crenarchaeota': 0.10,\n",
    "            'Other': 0.20,\n",
    "        },\n",
    "    },\n",
    "    'Sample3': {\n",
    "        'Bacteria': {\n",
    "            'Firmicutes': 0.30,\n",
    "            'Bacteroidetes': 0.30,\n",
    "            'Proteobacteria': 0.10,\n",
    "            'Actinobacteria': 0.10,\n",
    "            'Fusobacteria': 0.10,\n",
    "            'Other': 0.10,\n",
    "        },\n",
    "        'Archaea': {\n",
    "            'Euryarchaeota': 0.80,\n",
    "            'Other': 0.20,\n",
    "        },\n",
    "    },\n",
    "}\n",
    "\n",
    "# Create a subplot of pie charts for each sample\n",
    "fig = sp.make_subplots(rows=len(microbiomes), cols=1, specs=[[{'type':'pie'}] for i in range(len(microbiomes))], subplot_titles=list(microbiomes.keys()))\n",
    "\n",
    "for i, sample in enumerate(microbiomes):\n",
    "    # Get the taxonomic composition of the sample\n",
    "    composition = microbiomes[sample]\n",
    "\n",
    "    # Create a list of labels and values for the pie chart\n",
    "    labels = []\n",
    "    values = []\n",
    "    for domain in composition:\n",
    "        for phylum in composition[domain]:\n",
    "            labels.append('{}: {}'.format(domain, phylum))\n",
    "            values.append(composition[domain][phylum])\n",
    "\n",
    "    # Add a pie chart to the subplot\n",
    "    fig.add_trace(go.Pie(labels=labels, values=values), row=i+1, col=1)\n",
    "    fig.update_layout(title='Taxonomic Composition of Multiple Microbiome Samples', height=300, width=300)\n",
    "\n",
    "fig.show()\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "import matplotlib.pyplot as plt\n",
    "\n",
    "# Sample data: taxonomic composition of multiple microbiome samples\n",
    "microbiomes = {\n",
    "    'Sample1': {\n",
    "        'Bacteria': {\n",
    "            'Firmicutes': 0.35,\n",
    "            'Bacteroidetes': 0.25,\n",
    "            'Proteobacteria': 0.20,\n",
    "            'Actinobacteria': 0.10,\n",
    "            'Fusobacteria': 0.05,\n",
    "            'Other': 0.05,\n",
    "        },\n",
    "        'Archaea': {\n",
    "            'Euryarchaeota': 0.60,\n",
    "            'Crenarchaeota': 0.20,\n",
    "            'Other': 0.20,\n",
    "        },\n",
    "    },\n",
    "    'Sample2': {\n",
    "        'Bacteria': {\n",
    "            'Firmicutes': 0.40,\n",
    "            'Bacteroidetes': 0.20,\n",
    "            'Proteobacteria': 0.15,\n",
    "            'Actinobacteria': 0.10,\n",
    "            'Fusobacteria': 0.10,\n",
    "            'Other': 0.05,\n",
    "        },\n",
    "        'Archaea': {\n",
    "            'Euryarchaeota': 0.70,\n",
    "            'Crenarchaeota': 0.10,\n",
    "            'Other': 0.20,\n",
    "        },\n",
    "    },\n",
    "    'Sample3': {\n",
    "        'Bacteria': {\n",
    "            'Firmicutes': 0.30,\n",
    "            'Bacteroidetes': 0.30,\n",
    "            'Proteobacteria': 0.10,\n",
    "            'Actinobacteria': 0.10,\n",
    "            'Fusobacteria': 0.10,\n",
    "            'Other': 0.10,\n",
    "        },\n",
    "        'Archaea': {\n",
    "            'Euryarchaeota': 0.80,\n",
    "            'Other': 0.20,\n",
    "        },\n",
    "    },\n",
    "}\n",
    "\n",
    "# Create a subplot of pie charts for each sample\n",
    "fig, axs = plt.subplots(len(microbiomes), 1, figsize=(8, 8))\n",
    "fig.suptitle('Taxonomic Composition of Multiple Microbiome Samples')\n",
    "\n",
    "for i, sample in enumerate(microbiomes):\n",
    "    # Get the taxonomic composition of the sample\n",
    "    composition = microbiomes[sample]\n",
    "\n",
    "    # Create a list of labels and values for the pie chart\n",
    "    labels = []\n",
    "    values = []\n",
    "    for domain in composition:\n",
    "        for phylum in composition[domain]:\n",
    "            labels.append('{}: {}'.format(domain, phylum))\n",
    "            values.append(composition[domain][phylum])\n",
    "\n",
    "    # Add a pie chart to the subplot\n",
    "    axs[i].pie(values, labels=labels, autopct='%1.1f%%', startangle=90)\n",
    "    axs[i].axis('equal')\n",
    "    axs[i].set_title(sample)\n",
    "\n",
    "plt.show()\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": "",
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "from math import pi\n",
    "\n",
    "import pandas as pd\n",
    "\n",
    "from bokeh.models import BasicTicker, PrintfTickFormatter\n",
    "from bokeh.plotting import figure, show\n",
    "from bokeh.sampledata.unemployment1948 import data\n",
    "from bokeh.transform import linear_cmap\n",
    "\n",
    "data['Year'] = data['Year'].astype(str)\n",
    "data = data.set_index('Year')\n",
    "data.drop('Annual', axis=1, inplace=True)\n",
    "data.columns.name = 'Month'\n",
    "\n",
    "years = list(data.index)\n",
    "months = list(reversed(data.columns))\n",
    "\n",
    "# reshape to 1D array or rates with a month and year for each row.\n",
    "df = pd.DataFrame(data.stack(), columns=['rate']).reset_index()\n",
    "\n",
    "# this is the colormap from the original NYTimes plot\n",
    "colors = [\"#75968f\", \"#a5bab7\", \"#c9d9d3\", \"#e2e2e2\", \"#dfccce\", \"#ddb7b1\", \"#cc7878\", \"#933b41\", \"#550b1d\"]\n",
    "\n",
    "TOOLS = \"hover,save,pan,box_zoom,reset,wheel_zoom\"\n",
    "\n",
    "p = figure(title=f\"US Unemployment ({years[0]} - {years[-1]})\",\n",
    "           x_range=years, y_range=months,\n",
    "           x_axis_location=\"above\", width=900, height=400,\n",
    "           tools=TOOLS, toolbar_location='below',\n",
    "           tooltips=[('date', '@Month @Year'), ('rate', '@rate%')])\n",
    "\n",
    "p.grid.grid_line_color = None\n",
    "p.axis.axis_line_color = None\n",
    "p.axis.major_tick_line_color = None\n",
    "p.axis.major_label_text_font_size = \"7px\"\n",
    "p.axis.major_label_standoff = 0\n",
    "p.xaxis.major_label_orientation = pi / 3\n",
    "\n",
    "r = p.rect(x=\"Year\", y=\"Month\", width=1, height=1, source=df,\n",
    "           fill_color=linear_cmap(\"rate\", colors, low=df.rate.min(), high=df.rate.max()),\n",
    "           line_color=None)\n",
    "\n",
    "p.add_layout(r.construct_color_bar(\n",
    "    major_label_text_font_size=\"7px\",\n",
    "    ticker=BasicTicker(desired_num_ticks=len(colors)),\n",
    "    formatter=PrintfTickFormatter(format=\"%d%%\"),\n",
    "    label_standoff=6,\n",
    "    border_line_color=None,\n",
    "    padding=5\n",
    "), 'right')\n",
    "\n",
    "show(p)"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "from bokeh.models import ColumnDataSource\n",
    "from bokeh.palettes import GnBu3, OrRd3\n",
    "from bokeh.plotting import figure, show\n",
    "\n",
    "fruits = ['Apples', 'Pears', 'Nectarines', 'Plums', 'Grapes', 'Strawberries']\n",
    "years = [\"2015\", \"2016\", \"2017\"]\n",
    "\n",
    "Bacteria = {'fruits' : fruits,\n",
    "           '2015'   : [2, 1, 4, 3, 2, 4],\n",
    "           '2016'   : [5, 3, 4, 2, 4, 6],\n",
    "           '2017'   : [3, 2, 4, 4, 5, 3]}\n",
    "Archea = {'fruits' : fruits,\n",
    "           '2015'   : [-1, 0, -1, -3, -2, -1],\n",
    "           '2016'   : [-2, -1, -3, -1, -2, -2],\n",
    "           '2017'   : [-1, -2, -1, 0, -2, -2]}\n",
    "\n",
    "# Calculate the total value for each fruit across all years\n",
    "total_values = {}\n",
    "for fruit in fruits:\n",
    "    total_value = sum([exports[year][i] - Archea[year][i] for i, f in enumerate(fruits) if f == fruit for year in years])\n",
    "    total_values[fruit] = total_value\n",
    "\n",
    "# Normalize the values to percentages\n",
    "for year in years:\n",
    "    for i, fruit in enumerate(fruits):\n",
    "        exports[year][i] = exports[year][i] / total_values[fruit] * 10\n",
    "        imports[year][i] = imports[year][i] / total_values[fruit] * 10\n",
    "\n",
    "# Flip the orientation of the plot\n",
    "p = figure(x_range=fruits, width=700, y_range=(0, 10), title=\"Fruit import/export, by year\",\n",
    "           toolbar_location=None, tools='')\n",
    "\n",
    "p.vbar_stack(years, x='fruits', width=0.9, color=GnBu3, source=ColumnDataSource(exports),\n",
    "             legend_label=[f\"{year} exports\" for year in years])\n",
    "\n",
    "p.vbar_stack(years, x='fruits', width=0.9, color=OrRd3, source=ColumnDataSource(imports),\n",
    "             legend_label=[f\"{year} imports\" for year in years])\n",
    "\n",
    "p.x_range.range_padding = 0.1\n",
    "p.xgrid.grid_line_color = None\n",
    "p.legend.location = \"top_left\"\n",
    "p.axis.minor_tick_line_color = None\n",
    "p.outline_line_color = None\n",
    "\n",
    "# Set the y-axis formatter to display percentages\n",
    "#from bokeh.models import NumeralTickFormatter\n",
    "#p.yaxis.formatter = NumeralTickFormatter(format=\"0 %\")\n",
    "\n",
    "show(p)\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "source": [
    "import os\n",
    "print(\"PYTHONPATH:\", os.environ.get('PYTHONPATH'))\n",
    "print(\"PATH:\", os.environ.get('PATH'))"
   ],
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "\n",
    "plt.style.use('_mpl-gallery')\n",
    "\n",
    "# make data\n",
    "x = np.arange(0, 10, 2)\n",
    "ay = [1, 1.25, 2, 2.75, 3]\n",
    "by = [1, 1, 1, 1, 1]\n",
    "cy = [2, 1, 2, 1, 2]\n",
    "y = np.vstack([ay, by, cy])\n",
    "\n",
    "# plot\n",
    "fig, ax = plt.subplots()\n",
    "\n",
    "ax.stackplot(x, y)\n",
    "\n",
    "ax.set(xlim=(0, 8), xticks=np.arange(1, 8),\n",
    "       ylim=(0, 8), yticks=np.arange(1, 8))\n",
    "\n",
    "plt.show()"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "# Load required libraries\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import biom\n",
    "from phyloseq import *\n",
    "\n",
    "# Example biom table and sample metadata\n",
    "biom_table = biom.Table(np.array([[20, 10, 0], [5, 15, 2], [1, 2, 3]]),\n",
    "                        observation_ids=['OTU1', 'OTU2', 'OTU3'],\n",
    "                        sample_ids=['sample1', 'sample2', 'sample3'])\n",
    "sample_metadata = pd.DataFrame({'Sample': ['sample1', 'sample2', 'sample3'],\n",
    "                                'Condition': ['A', 'A', 'B']},\n",
    "                               index=['sample1', 'sample2', 'sample3'])\n",
    "\n",
    "# Example taxonomy table\n",
    "tax_table = pd.DataFrame({'Kingdom': ['Bacteria', 'Bacteria', 'Bacteria'],\n",
    "                          'Phylum': ['Firmicutes', 'Actinobacteria', 'Proteobacteria'],\n",
    "                          'Class': ['Bacilli', 'Actinobacteria', 'Gammaproteobacteria']},\n",
    "                         index=['OTU1', 'OTU2', 'OTU3'])\n",
    "\n",
    "# Convert biom table to pandas dataframe\n",
    "otu_table = biom_table.to_dataframe()\n",
    "\n",
    "# Create phyloseq object\n",
    "physeq = phyloseq(otu_table, tax_table, sample_metadata)\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "\n",
    "category_names = ['Strongly disagree', 'Disagree',\n",
    "                  'Neither agree nor disagree', 'Agree', 'Strongly agree']\n",
    "results = {\n",
    "    'Question 1': [10, 15, 17, 32, 26],\n",
    "    'Question 2': [26, 22, 29, 10, 13],\n",
    "    'Question 3': [35, 37, 7, 2, 19],\n",
    "    'Question 4': [32, 11, 9, 15, 33],\n",
    "    'Question 5': [21, 29, 5, 5, 40],\n",
    "    'Question 6': [8, 19, 5, 30, 38]\n",
    "}\n",
    "\n",
    "\n",
    "def survey(results, category_names):\n",
    "    \"\"\"\n",
    "    Parameters\n",
    "    ----------\n",
    "    results : dict\n",
    "        A mapping from question labels to a list of answers per category.\n",
    "        It is assumed all lists contain the same number of entries and that\n",
    "        it matches the length of *category_names*.\n",
    "    category_names : list of str\n",
    "        The category labels.\n",
    "    \"\"\"\n",
    "    labels = list(results.keys())\n",
    "    data = np.array(list(results.values()))\n",
    "    data_cum = data.cumsum(axis=1)\n",
    "    category_colors = plt.colormaps['RdYlGn'](\n",
    "        np.linspace(0.15, 0.85, data.shape[1]))\n",
    "\n",
    "    fig, ax = plt.subplots(figsize=(9.2, 5))\n",
    "    ax.invert_yaxis()\n",
    "    ax.xaxis.set_visible(False)\n",
    "    ax.set_xlim(0, np.sum(data, axis=1).max())\n",
    "\n",
    "    for i, (colname, color) in enumerate(zip(category_names, category_colors)):\n",
    "        widths = data[:, i]\n",
    "        starts = data_cum[:, i] - widths\n",
    "        rects = ax.barh(labels, widths, left=starts, height=0.5,\n",
    "                        label=colname, color=color)\n",
    "\n",
    "        r, g, b, _ = color\n",
    "        text_color = 'white' if r * g * b < 0.5 else 'darkgrey'\n",
    "        ax.bar_label(rects, label_type='center', color=text_color)\n",
    "    ax.legend(ncols=len(category_names), bbox_to_anchor=(0, 1),\n",
    "              loc='lower left', fontsize='small')\n",
    "\n",
    "    return fig, ax\n",
    "\n",
    "\n",
    "survey(results, category_names)\n",
    "plt.show()"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "fig, ax = plt.subplots()\n",
    "\n",
    "size = 0.3\n",
    "vals = np.array([[60., 32.], [37., 40.], [29., 10.]])\n",
    "\n",
    "cmap = plt.colormaps[\"tab20c\"]\n",
    "outer_colors = cmap(np.arange(3)*4)\n",
    "inner_colors = cmap([1, 2, 5, 6, 9, 10])\n",
    "\n",
    "ax.pie(vals.sum(axis=1), radius=1, colors=outer_colors,\n",
    "       wedgeprops=dict(width=size, edgecolor='w'))\n",
    "\n",
    "ax.pie(vals.flatten(), radius=1-size, colors=inner_colors,\n",
    "       wedgeprops=dict(width=size, edgecolor='w'))\n",
    "\n",
    "ax.set(aspect=\"equal\", title='Pie plot with `ax.pie`')\n",
    "plt.show()"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "import plotly.express as px\n",
    "\n",
    "wide_df = px.data.medals_wide()\n",
    "\n",
    "fig = px.bar(wide_df, x=\"nation\", y=[\"gold\", \"silver\", \"bronze\"], title=\"Wide-Form Input\")\n",
    "fig.show()"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "import matplotlib.pyplot as plt\n",
    "\n",
    "# Sample data\n",
    "x = [1, 2, 3, 4, 5]\n",
    "y = [2, 4, 6, 8, 10]\n",
    "\n",
    "# Create a line graph\n",
    "plt.plot(x, y)\n",
    "\n",
    "# Add labels and a title\n",
    "plt.xlabel('X-axis label')\n",
    "plt.ylabel('Y-axis label')\n",
    "plt.title('Line Graph Example')\n",
    "\n",
    "# Display the graph\n",
    "plt.show()\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "import plotly.express as px\n",
    "\n",
    "long_df = px.data.medals_long()\n",
    "\n",
    "fig = px.bar(long_df, x=\"nation\", y=\"count\", color=\"medal\", title=\"Long-Form Input\")\n",
    "fig.show()\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "import plotly.express as px\n",
    "\n",
    "long_df = px.data.medals_long()\n",
    "\n",
    "fig = px.bar(long_df, x=\"nation\", y=\"count\", color=\"medal\", title=\"Long-Form Input\")\n",
    "fig.update_yaxes(range=[0, 100])\n",
    "fig.show()\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "import pandas as pd\n",
    "import plotly.express as px\n",
    "\n",
    "\n",
    "# Example OTU table with 16 samples\n",
    "otu_table = pd.DataFrame({\n",
    "    'OTU': ['Other', 'Euryarchaeota', 'Actinobacteria', 'Bacteroidetes', 'Chlamydiae', 'Chloroflexi', 'Cyanobacteria', 'Elusimicrobia', 'Fibrobacteres', 'Firmicutes', 'Fusobacteria', 'Lentisphaerae', 'NA', 'Nitrospirae', 'Planctomycetes', 'Proteobacteria', 'Saccharibacteria', 'Spirochaetae', 'Synergistetes', 'Tenericutes', 'Verrucomicrobia'],\n",
    "    'T20': [0.0692213560163, 0.0512763463036, 0.00552812411751, 0.0460905208931, 0.0, 0.00294376887991, 0.00377384324431, 0.000573350334169, 0.0061956065961, 0.772371359867, 0.000119804547438, 0.00142053963391, 0.00117237307136, 0.0, 0.00855746767417, 0.00939609950623, 0.0, 0.000650367543237, 0.0173887743139, 0.00139486723089, 0.00192543022669],\n",
    "    'T107': [0.0886399156762, 0.0700123577945, 0.00510667684367, 0.0883764038818, 0.0, 0.00936829862247, 0.00912296005525, 0.0, 0.0122851015883, 0.665294588013, 0.000208992112819, 0.00439792098281, 0.000508850361647, 0.000118125976811, 0.00885944826082, 0.00822338530876, 0.0, 0.000954094428089, 0.0118398575219, 0.00336204703231, 0.0133209755388],\n",
    "    'T203': [0.091516034342, 0.0439206338697, 0.00434792204639, 0.0828312255843, 0.0, 0.0188483524245, 0.00508728950098, 0.00070626144916, 0.0218609989185, 0.655399589486, 0.0, 0.00456862874926, 0.00194221898519, 0.0, 0.0204815820256, 0.00940210554195, 0.0, 0.00150080557947, 0.0326094153479, 0.00264848043435, 0.0023284557152],\n",
    "    'T11': [0.0387414787625, 0.053864708967, 0.00231777661248, 0.0659884635553, 0.0, 0.0028736234924, 0.00484530676455, 0.0, 0.00624016780283, 0.765977975878, 0.0, 0.00684845306765, 0.00293654955427, 0.0, 0.011295228107, 0.011777661248, 0.0, 0.00378605138962, 0.019979024646, 0.000964866282119, 0.00156266386995],\n",
    "    'T212': [0.0662810452229, 0.0916511609093, 0.00777445190519, 0.0822587169323, 0.0, 0.013558773562, 0.00245125798884, 0.00040449801796, 0.00329261386619, 0.667834317612, 0.000226518890057, 0.00792816115201, 0.00574387185503, 0.00020224900898, 0.0129358466144, 0.00594612086401, 0.000275058652213, 0.000493487581911, 0.02409999191, 0.0011892241728, 0.0054526332821],\n",
    "    'T85.1': [0.0510423403203, 0.0536196599079, 0.00868299920113, 0.070557690113, 0.0, 0.000551603454179, 0.00618176284856, 0.000266291322707, 0.00414653631072, 0.739947502568, 0.0, 0.00664777266329, 0.00207326815536, 0.0, 0.0124586297409, 0.00714231369118, 0.0, 0.000884467607563, 0.0186594133983, 0.00222543462548, 0.0149123140716],\n",
    "    'T65': [0.0730575036454, 0.0383787483509, 0.00480106337602, 0.0652210572259, 0.0, 0.00343216514071, 0.0076578944758, 0.00488041979546, 0.0331114660107, 0.728412574025, 0.0, 0.00436460306911, 0.0, 0.0, 0.00758845760879, 0.00395790141949, 0.000119034629158, 0.000247988810745, 0.0178452748212, 0.00344208469314, 0.00348176290286],\n",
    "    'T196': [0.0808211256746, 0.0581534309946, 0.0115940632228, 0.0606399383192, 0.0, 0.0177525057826, 0.00546453353894, 0.00122397841172, 0.00293947571318, 0.692530840401, 0.0, 0.00149383191981, 0.000857748650732, 0.00011565150347, 0.0143022359291, 0.00536815728604, 0.0, 0.000395142636854, 0.0429163454125, 0.00146491904395, 0.00196607555898],\n",
    "    'T99': [0.0524611702876, 0.0459035240017, 0.00296795082352, 0.0678522828586, 0.000129041340153, 0.00292102669983, 0.0018886959786, 0.000164234432922, 0.00140772371076, 0.785157899676, 0.0, 0.00446952278166, 0.00170099948383, 0.000187696494768, 0.00476279855474, 0.00735535638872, 0.0, 0.00082117216461, 0.0105931209235, 0.00208812350429, 0.00716765989395],\n",
    "    'T85.2': [0.0522599376322, 0.0657281622997, 0.00296605613104, 0.0922613713753, 0.00012545252518, 0.00324384386537, 0.0172945266855, 0.0018817878777, 0.00709702856733, 0.685409871322, 0.0, 0.00685508441163, 0.00260762034482, 0.0, 0.0146958672354, 0.00730312914441, 0.0, 0.000143374314492, 0.0252159575612, 0.00278683823793, 0.0121240904692],\n",
    "    'TAK': [0.0483987105493, 0.0419514572786, 0.00371104784482, 0.0433915653975, 0.0, 0.00776550608723, 0.00360027029722, 0.0, 0.00356703703293, 0.806272224745, 0.0, 0.00254788359495, 0.0023374062545, 0.0, 0.00550564411605, 0.00521762249227, 0.0, 0.000919453645135, 0.020504924062, 0.00177244076171, 0.00253680584019],\n",
    "    'T85.3': [0.090085008662, 0.0447403408404, 0.00357895867746, 0.0645421215906, 0.000154439117414, 0.00243073741321, 0.00817184373447, 0.000389455165653, 0.0104078535649, 0.723305534292, 0.000208157071297, 0.0052173562709, 0.000490176329184, 0.0, 0.00637229227939, 0.00932677974296, 0.0, 0.000705048144716, 0.0151014597854, 0.00427057733371, 0.0105018599842],\n",
    "    'T195': [0.0565088200213, 0.0267216305007, 0.00165310008579, 0.0622214317102, 0.0, 0.00525225470297, 0.00124505639373, 0.0, 0.0188223231288, 0.795946765992, 0.0, 0.00277260457427, 0.00187281591998, 0.0, 0.00486513632844, 0.002008830484, 0.0, 0.00269936596288, 0.00947916884639, 0.00177865199104, 0.00615204335726],\n",
    "    'T205': [0.0745835897329, 0.0343140833759, 0.00274304070756, 0.0786094973873, 0.0, 0.0101794970744, 0.000500631003661, 0.0, 0.014841623296, 0.743635206875, 0.0, 0.00377559215261, 0.0032958207741, 0.0, 0.00808310474661, 0.00238842707997, 0.0, 0.00171048926251, 0.0154048331751, 0.00132458619719, 0.00460997715871],\n",
    "    'T159': [0.0588736891759, 0.0825048078668, 0.00325664937044, 0.0711836423673, 0.00027214340143, 0.0111215936718, 0.00707572843717, 0.000589644036431, 0.00576036866359, 0.679269929968, 0.0, 0.0143963859356, 0.00654958452774, 0.0, 0.0098788054719, 0.0094978047099, 0.0, 0.000353786421859, 0.0329021372328, 0.00219529010487, 0.00431800863602],\n",
    "    'TAA': [0.0303314571607, 0.0411641204324, 0.00253506655946, 0.0739971410703, 0.0, 0.00230054498347, 0.0024457250067, 0.0, 0.0010162601626, 0.795340838024, 0.0, 0.00385285446261, 0.00151880639686, 0.0, 0.0138926114536, 0.00368533905119, 0.0, 0.000145180023229, 0.0233516483516, 0.00165281872599, 0.00276958813544]}\n",
    ")\n",
    "otu_table.iloc[:, 1:] *= 100\n",
    "plt.figure(dpi=600)\n",
    "# Calculate average relative abundance for each OTU\n",
    "otu_table['Average'] = otu_table.iloc[:, 1:].mean(axis=1)\n",
    "\n",
    "# Melt DataFrame\n",
    "df_melted = pd.melt(otu_table, id_vars=['OTU'], var_name='T', value_name='Relative Abundance')\n",
    "\n",
    "# Set y-axis range to 0-100%\n",
    "fig.update_yaxes(range=[0, 100])\n",
    "fig = px.bar(df_melted, x='T', y='Relative Abundance', color='OTU',\n",
    "             title='', barmode='stack',\n",
    "             color_discrete_sequence=px.colors.qualitative.Alphabet)\n",
    "fig.update_layout(\n",
    "    xaxis_title=\"Sample ID\"\n",
    ")\n",
    "# Update layout\n",
    "fig.update_layout(legend=dict(orientation=\"h\", yanchor=\"bottom\", y=1.02, xanchor=\"right\", x=1),\n",
    "                  margin=dict(t=100, b=50, l=50, r=50))\n",
    "fig.update_layout(plot_bgcolor='rgba(255, 255, 255, 1)')\n",
    "# Show plot\n",
    "fig.show()\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "import pandas as pd\n",
    "import plotly.express as px\n",
    "\n",
    "\n",
    "# Example OTU table with 21 OTUs and 2 samples\n",
    "otu_table = pd.DataFrame({\n",
    "    'OTU': ['Archea', 'OTU2', 'OTU3', 'OTU4', 'OTU5', 'OTU6', 'OTU7', 'OTU8', 'OTU9', 'OTU10', 'OTU11', 'OTU12', 'OTU13', 'OTU14', 'OTU15', 'OTU16', 'OTU17', 'OTU18', 'OTU19', 'OTU20', 'OTU21'],\n",
    "    'Sample1': [0.0692213560163, 0.0512763463036, 0.00552812411751, 0.0460905208931, 0.0, 0.00294376887991, 0.00377384324431, 0.000573350334169, 0.0061956065961, 0.772371359867, 0.000119804547438, 0.00142053963391, 0.00117237307136, 0.0, 0.00855746767417, 0.00939609950623, 0.0, 0.000650367543237, 0.0173887743139, 0.00139486723089, 0.00192543022669],\n",
    "    'Sample2': [0.0886399156762, 0.0700123577945, 0.00510667684367, 0.0883764038818, 0.0, 0.00936829862247, 0.00912296005525, 0.0, 0.0122851015883, 0.665294588013, 0.000208992112819, 0.00439792098281, 0.000508850361647, 0.000118125976811, 0.00885944826082, 0.00822338530876, 0.0, 0.000954094428089, 0.0118398575219, 0.00336204703231, 0.0133209755388]\n",
    "})\n",
    "# Multiply values by 100\n",
    "otu_table.iloc[:, 1:] *= 100\n",
    "# Melt the OTU table\n",
    "melted_otu_table = pd.melt(otu_table, id_vars='OTU', var_name='Sample', value_name='Relative Abundance')\n",
    "\n",
    "# Create plot\n",
    "fig = px.bar(melted_otu_table, x='Sample', y='Relative Abundance', color='OTU',\n",
    "             title='Sample Relative Abundance', barmode='stack')\n",
    "# Set y-axis range to 0-100%\n",
    "fig.update_yaxes(range=[0, 100])\n",
    "\n",
    "# Update layout\n",
    "fig.update_layout(legend=dict(orientation=\"h\", yanchor=\"bottom\", y=1.005, xanchor=\"right\", x=1),\n",
    "                  margin=dict(t=100, b=50, l=50, r=50))\n",
    "\n",
    "\n",
    "\n",
    "# Show plot\n",
    "fig.show()\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "# Create green Bars\n",
    "plt.bar(r, greenBars, color='#b5ffb9', edgecolor='white', width=barWidth, label=\"group A\")\n",
    "# Create orange Bars\n",
    "plt.bar(r, orangeBars, bottom=greenBars, color='#f9bc86', edgecolor='white', width=barWidth, label=\"group B\")\n",
    "# Create blue Bars\n",
    "plt.bar(r, blueBars, bottom=[i+j for i,j in zip(greenBars, orangeBars)], color='#a3acff', edgecolor='white', width=barWidth, label=\"group C\")\n",
    "\n",
    "# Custom x axis\n",
    "plt.xticks(r, names)\n",
    "plt.xlabel(\"group\")\n",
    "\n",
    "# Add a legend\n",
    "plt.legend(loc='upper left', bbox_to_anchor=(1,1), ncol=1)\n",
    "\n",
    "# Show graphic\n",
    "plt.show()"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "\n",
    "# Generate some sample data\n",
    "x = np.linspace(0, 10, 100)\n",
    "y1 = np.sin(x)\n",
    "y2 = np.cos(x)\n",
    "\n",
    "# Create the first subplot\n",
    "plt.subplot(2, 1, 1)  # 2 rows, 1 column, subplot number 1\n",
    "plt.plot(x, y1, color='red')\n",
    "plt.title('Subplot 1')\n",
    "plt.xlabel('X-axis label')\n",
    "plt.ylabel('Y-axis label')\n",
    "plt.text(-0.125, 1.25, 'A', transform=plt.gca().transAxes,\n",
    "fontsize=16, fontweight='bold', va='top')\n",
    "\n",
    "# Create the second subplot\n",
    "plt.subplot(2, 1, 2)  # 2 rows, 1 column, subplot number 2\n",
    "plt.plot(x, y2, color='blue')\n",
    "plt.title('Subplot 2')\n",
    "plt.xlabel('X-axis label')\n",
    "plt.ylabel('Y-axis label')\n",
    "plt.text(-0.125, 1.25, 'B', transform=plt.gca().transAxes,\n",
    "fontsize=16, fontweight='bold', va='top')\n",
    "\n",
    "# Add some padding between the subplots\n",
    "plt.subplots_adjust(hspace=1)\n",
    "\n",
    "# Show the plot\n",
    "plt.show()\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "import requests\n",
    "import re\n",
    "import PyPDF2\n",
    "\n",
    "# Download the PDF from the URL\n",
    "url = 'https://www.cell.com/molecular-therapy-family/molecular-therapy/pdf/S1525-0016(22)00099-5.pdf'\n",
    "response = requests.get(url)\n",
    "with open('document.pdf', 'wb') as f:\n",
    "    f.write(response.content)\n",
    "\n",
    "# Extract text from the PDF\n",
    "pdf_file = open('document.pdf', 'rb')\n",
    "pdf_reader = PyPDF2.PdfFileReader(pdf_file)\n",
    "text = ''\n",
    "for i in range(pdf_reader.numPages):\n",
    "    text += pdf_reader.getPage(i).extractText()\n",
    "\n",
    "# Count the number of times \"and\" appears at the beginning of a sentence\n",
    "count = len(re.findall(r'(?:^|\\.\\s+)(and)', text, re.IGNORECASE))\n",
    "\n",
    "print(f'The word \"and\" appears at the beginning of a sentence {count} times.')\n",
    "\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "import requests\n",
    "import PyPDF2\n",
    "\n",
    "url = \"https://opencommons.uconn.edu/cgi/viewcontent.cgi?article=1510&context=law_review\"\n",
    "\n",
    "# Download the PDF from the URL\n",
    "response = requests.get(url)\n",
    "with open('document.pdf', 'wb') as f:\n",
    "    f.write(response.content)\n",
    "\n",
    "# Read the PDF and extract text\n",
    "with open('document.pdf', 'rb') as f:\n",
    "    pdfReader = PyPDF2.PdfFileReader(f)\n",
    "    numPages = pdfReader.numPages\n",
    "    text = \"\"\n",
    "    for page in range(numPages):\n",
    "        pageObj = pdfReader.getPage(page)\n",
    "        text += pageObj.extractText()\n",
    "\n",
    "# Split the text into sentences and print each sentence that starts with \"and\"\n",
    "sentences = text.split('.')\n",
    "for sentence in sentences:\n",
    "    if sentence.startswith('And'):\n",
    "        print(sentence.strip() + '.')\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "import requests\n",
    "import io\n",
    "import PyPDF2\n",
    "\n",
    "# Fetch PDF file from URL\n",
    "url = 'https://www.cell.com/molecular-therapy-family/molecular-therapy/pdf/S1525-0016(22)00099-5.pdf'\n",
    "response = requests.get(url)\n",
    "pdf_file = io.BytesIO(response.content)\n",
    "\n",
    "# Read PDF content\n",
    "pdf_reader = PyPDF2.PdfFileReader(pdf_file)\n",
    "num_pages = pdf_reader.numPages\n",
    "text = ''\n",
    "for i in range(num_pages):\n",
    "    page = pdf_reader.getPage(i)\n",
    "    text += page.extractText()\n",
    "\n",
    "# Split text into sentences\n",
    "sentences = text.split('. ')\n",
    "for sentence in sentences:\n",
    "    if sentence.startswith('And'):\n",
    "        print(sentence.strip() + '.')\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "import requests\n",
    "from bs4 import BeautifulSoup\n",
    "\n",
    "# Send a request to the website\n",
    "url = 'https://opencommons.uconn.edu/cgi/viewcontent.cgi?article=1510&context=law_review'\n",
    "response = requests.get(url)\n",
    "\n",
    "# Parse the HTML content with BeautifulSoup\n",
    "soup = BeautifulSoup(response.content, 'html.parser')\n",
    "\n",
    "# Find all the paragraphs in the document\n",
    "paragraphs = soup.find_all('p')\n",
    "\n",
    "# Initialize a counter for the number of sentences starting with \"and\"\n",
    "and_count = 0\n",
    "\n",
    "# Loop through each paragraph and count the number of sentences starting with \"and\"\n",
    "for p in paragraphs:\n",
    "    sentences = p.text.split('.')\n",
    "    for s in sentences:\n",
    "        if s.strip().startswith('The'):\n",
    "            and_count += 1\n",
    "\n",
    "# Print the result\n",
    "print(f'The word \"The\" is used to begin a sentence {and_count} times in the document.')\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "# Install the library if you haven't already\n",
    "# !pip install ete3\n",
    "\n",
    "from ete3 import Tree, TreeStyle, NodeStyle\n",
    "\n",
    "# Read the Newick-formatted string\n",
    "newick_string = \"(( (Bacteroides fragilis NCTC 9343_Phylum_Bacteriodetes:0.14775, Bifidobacterium longum subsp. longum JCM 1217_Phylum_Actinobacteria:0.13892) :0.01857, Helicobacter pylori strain ATCC 43504_Phylum_Proteobacteria:0.11945) :0.00449, Ruminococcus bromii strain ATCC 27255_Phylum_Firmicutes:0.11881, (Clostridium butyricum strain VPI3266_Phylum_Firmicutes:0.03359,Clostridium perfringens ATCC 13124_Phylum_Firmicutes:0.04132) :0.05316);\"\n",
    "t = Tree(newick_string)\n",
    "\n",
    "# Configure the tree style\n",
    "ts = TreeStyle()\n",
    "ts.show_leaf_name = True\n",
    "ts.show_branch_length = True\n",
    "ts.show_branch_support = True\n",
    "\n",
    "# Plot the tree\n",
    "t.render(\"%%inline\", w=800, units=\"px\", tree_style=ts)\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "import re\n",
    "from ete3 import Tree, TreeStyle, NodeStyle\n",
    "\n",
    "# Read the Newick-formatted string\n",
    "newick_string = \"(( (Bacteroides fragilis NCTC 9343_Phylum_Bacteriodetes:0.14775, Bifidobacterium longum subsp. longum JCM 1217_Phylum_Actinobacteria:0.13892) :0.01857, Helicobacter pylori strain ATCC 43504_Phylum_Proteobacteria:0.11945) :0.00449, Ruminococcus bromii strain ATCC 27255_Phylum_Firmicutes:0.11881, (Clostridium butyricum strain VPI3266_Phylum_Firmicutes:0.03359,Clostridium perfringens ATCC 13124_Phylum_Firmicutes:0.04132) :0.05316);\"\n",
    "\n",
    "# Find all decimal numbers in the string\n",
    "decimals = re.findall(r\"\\d+\\.\\d+\", newick_string)\n",
    "\n",
    "# Replace each decimal number with a rounded version\n",
    "for decimal in decimals:\n",
    "    rounded_decimal = \"{:.4f}\".format(float(decimal))\n",
    "    newick_string = newick_string.replace(decimal, rounded_decimal)\n",
    "\n",
    "t = Tree(newick_string)\n",
    "\n",
    "# Configure the tree style\n",
    "ts = TreeStyle()\n",
    "ts.show_leaf_name = True\n",
    "ts.show_branch_length = True\n",
    "ts.show_branch_support = True\n",
    "\n",
    "# Plot the tree\n",
    "t.render(\"%%inline\", w=800, units=\"px\", tree_style=ts)\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": "!/usr/local/bin/python3.9 -m pip install ete3",
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "from ete3 import Tree, TreeStyle, NodeStyle, TextFace, add_face_to_node\n",
    "\n",
    "# Define a layout function to customize node and branch appearance\n",
    "def layout(node):\n",
    "    if node.is_leaf():\n",
    "        species_name = TextFace(node.name, fsize=12)\n",
    "        add_face_to_node(species_name, node, column=0, position=\"branch-right\")\n",
    "\n",
    "    if not node.is_root():\n",
    "        branch_length = TextFace(\"{:.4f}\".format(node.dist), fsize=10)\n",
    "        add_face_to_node(branch_length, node, column=0, position=\"branch-top\")\n",
    "\n",
    "# Read the Newick-formatted string\n",
    "newick_string = \"(( (Bacteroides fragilis NCTC 9343_Phylum_Bacteriodetes:0.14775, Bifidobacterium longum subsp. longum JCM 1217_Phylum_Actinobacteria:0.13892) :0.01857, Helicobacter pylori strain ATCC 43504_Phylum_Proteobacteria:0.11945) :0.00449, Ruminococcus bromii strain ATCC 27255_Phylum_Firmicutes:0.11881, (Clostridium butyricum strain VPI3266_Phylum_Firmicutes:0.03359,Clostridium perfringens ATCC 13124_Phylum_Firmicutes:0.04132) :0.05316);\"\n",
    "t = Tree(newick_string)\n",
    "\n",
    "# Configure the tree style\n",
    "ts = TreeStyle()\n",
    "ts.show_leaf_name = False\n",
    "ts.show_branch_length = False\n",
    "ts.show_branch_support = True\n",
    "ts.layout_fn = layout\n",
    "\n",
    "# Plot the tree\n",
    "t.render(\"%%inline\", w=800, units=\"px\", tree_style=ts)\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "from ete3 import Tree, TreeStyle, NodeStyle\n",
    "\n",
    "# Read the Newick-formatted string\n",
    "newick_string = \"(( (Bacteroides fragilis NCTC 9343_Phylum_Bacteriodetes:0.14775, Bifidobacterium longum subsp. longum JCM 1217_Phylum_Actinobacteria:0.13892) :0.01857, Helicobacter pylori strain ATCC 43504_Phylum_Proteobacteria:0.11945) :0.00449, Ruminococcus bromii strain ATCC 27255_Phylum_Firmicutes:0.11881, (Clostridium butyricum strain VPI3266_Phylum_Firmicutes:0.03359,Clostridium perfringens ATCC 13124_Phylum_Firmicutes:0.04132) :0.05316);\"\n",
    "t = Tree(newick_string)\n",
    "\n",
    "# Configure the tree style\n",
    "ts = TreeStyle()\n",
    "ts.show_leaf_name = True\n",
    "ts.show_branch_length = True\n",
    "ts.show_branch_support = True\n",
    "ts.show_scale = True\n",
    "\n",
    "# Set scale length and label with 4 decimal places\n",
    "scale_length = 0.05\n",
    "ts.scale = scale_length\n",
    "ts.scale_formatter = lambda value: \"{:.4f}\".format(value)\n",
    "\n",
    "# Plot the tree\n",
    "t.render(\"%%inline\", w=800, units=\"px\", tree_style=ts)\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "from ete3 import Tree, TreeStyle, NodeStyle, TextFace\n",
    "\n",
    "# Read the Newick-formatted string\n",
    "newick_string = \"(( (Bacteroides fragilis NCTC 9343_Phylum_Bacteriodetes:0.14775, Bifidobacterium longum subsp. longum JCM 1217_Phylum_Actinobacteria:0.13892) :0.01857, Helicobacter pylori strain ATCC 43504_Phylum_Proteobacteria:0.11945) :0.00449, Ruminococcus bromii strain ATCC 27255_Phylum_Firmicutes:0.11881, (Clostridium butyricum strain VPI3266_Phylum_Firmicutes:0.03359,Clostridium perfringens ATCC 13124_Phylum_Firmicutes:0.04132) :0.05316);\"\n",
    "t = Tree(newick_string)\n",
    "\n",
    "# Configure the tree style\n",
    "ts = TreeStyle()\n",
    "ts.show_leaf_name = True\n",
    "ts.show_branch_length = True\n",
    "ts.show_branch_support = True\n",
    "ts.show_scale = False  # Disable the default scale bar\n",
    "\n",
    "# Set scale length and create a custom scale bar label with 4 decimal places\n",
    "#scale_length = 1\n",
    "#scale_label = TextFace(\"{:.4f}\".format(scale_length), fsize=10)\n",
    "\n",
    "# Add custom scale bar label to the tree style\n",
    "#ts.legend.add_face(scale_label, column=0)\n",
    "#ts.legend_position = 3  # Position the legend at the bottom of the tree\n",
    "\n",
    "# Plot the tree\n",
    "t.render(\"%%inline\", w=800, units=\"px\", tree_style=ts)\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "from ete3 import Tree, TreeStyle, NodeStyle, faces, AttrFace\n",
    "\n",
    "# Read the Newick-formatted string\n",
    "newick_string = \"(( (Bacteroides fragilis NCTC 9343_Phylum_Bacteriodetes:0.14775, Bifidobacterium longum subsp. longum JCM 1217_Phylum_Actinobacteria:0.13892) :0.01857, Helicobacter pylori strain ATCC 43504_Phylum_Proteobacteria:0.11945) :0.00449, Ruminococcus bromii strain ATCC 27255_Phylum_Firmicutes:0.11881, (Clostridium butyricum strain VPI3266_Phylum_Firmicutes:0.03359,Clostridium perfringens ATCC 13124_Phylum_Firmicutes:0.04132) :0.05316);\"\n",
    "t = Tree(newick_string)\n",
    "\n",
    "# Function to customize leaf nodes\n",
    "def custom_leaf_node(name):\n",
    "    # Create an italic face for the name\n",
    "    name_face = AttrFace(\"name\", fsize=14, fgcolor=\"black\", ftype=\"Arial Italic\")\n",
    "\n",
    "    # Add the face to the leaf node\n",
    "    faces.add_face_to_node(name_face, name, column=0, position=\"branch-right\")\n",
    "\n",
    "# Apply custom_leaf_node function to all leaves\n",
    "for leaf in t.iter_leaves():\n",
    "    custom_leaf_node(leaf)\n",
    "\n",
    "# Configure the tree style\n",
    "ts = TreeStyle()\n",
    "ts.show_leaf_name = False  # Set to False since we are using custom_leaf_node function\n",
    "ts.show_branch_length = True\n",
    "ts.show_branch_support = True\n",
    "\n",
    "# Plot the tree\n",
    "t.render(\"%%inline\", w=800, units=\"px\", tree_style=ts)\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "from ete3 import Tree, TreeStyle, NodeStyle, faces, AttrFace\n",
    "\n",
    "# Read the Newick-formatted string\n",
    "newick_string = \"(( (Bacteroides fragilis NCTC 9343_Phylum_Bacteriodetes:0.14775, Bifidobacterium longum subsp. longum JCM 1217_Phylum_Actinobacteria:0.13892) :0.01857, Helicobacter pylori strain ATCC 43504_Phylum_Proteobacteria:0.11945) :0.00449, Ruminococcus bromii strain ATCC 27255_Phylum_Firmicutes:0.11881, (Clostridium butyricum strain VPI3266_Phylum_Firmicutes:0.03359,Clostridium perfringens ATCC 13124_Phylum_Firmicutes:0.04132) :0.05316);\"\n",
    "t = Tree(newick_string)\n",
    "\n",
    "# Define the layout function\n",
    "def custom_layout(node):\n",
    "    if node.is_leaf():\n",
    "        # Create an italic face for the name\n",
    "        name_face = AttrFace(\"name\", fsize=14, fgcolor=\"black\", ftype=\"Arial Italic\")\n",
    "\n",
    "        # Add the face to the leaf node\n",
    "        faces.add_face_to_node(name_face, node, column=0, position=\"branch-right\")\n",
    "\n",
    "# Configure the tree style\n",
    "ts = TreeStyle()\n",
    "ts.show_leaf_name = False  # Set to False since we are using custom_layout function\n",
    "ts.show_branch_length = True\n",
    "ts.show_branch_support = True\n",
    "ts.layout_fn = custom_layout  # Set the custom layout function\n",
    "\n",
    "# Plot the tree\n",
    "t.render(\"%%inline\", w=800, units=\"px\", tree_style=ts)\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "from ete3 import Tree, TreeStyle, NodeStyle, faces, TextFace\n",
    "\n",
    "# Read the Newick-formatted string\n",
    "newick_string = \"(( (Bacteroides fragilis NCTC 9343_Phylum_Bacteriodetes:0.14775, Bifidobacterium longum subsp. longum JCM 1217_Phylum_Actinobacteria:0.13892) :0.01857, Helicobacter pylori strain ATCC 43504_Phylum_Proteobacteria:0.11945) :0.00449, Ruminococcus bromii strain ATCC 27255_Phylum_Firmicutes:0.11881, (Clostridium butyricum strain VPI3266_Phylum_Firmicutes:0.03359,Clostridium perfringens ATCC 13124_Phylum_Firmicutes:0.04132) :0.05316);\"\n",
    "t = Tree(newick_string)\n",
    "\n",
    "# Define the layout function\n",
    "def custom_layout(node):\n",
    "    if node.is_leaf():\n",
    "        # Create an italic face for the name\n",
    "        name_face = TextFace(node.name, fsize=14, fgcolor=\"black\", ftype=\"Arial\", fstyle=\"italic\")\n",
    "\n",
    "        # Add the face to the leaf node\n",
    "        faces.add_face_to_node(name_face, node, column=0, position=\"branch-right\")\n",
    "\n",
    "# Configure the tree style\n",
    "ts = TreeStyle()\n",
    "ts.show_leaf_name = False  # Set to False since we are using custom_layout function\n",
    "ts.show_branch_length = True\n",
    "ts.show_branch_support = True\n",
    "ts.layout_fn = custom_layout  # Set the custom layout function\n",
    "\n",
    "# Plot the tree\n",
    "t.render(\"%%inline\", w=800, units=\"px\", tree_style=ts)\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "Pipeline modified from Qiime2 Moving Pictures Tutorial\n",
    "Alyssa Demko\n",
    "\n",
    "Project: Belize 2015 sediment culture-independent vs dependent comparison\n",
    "06.13.2020\n",
    "\n",
    "Only looking at 2020 controls with Vibrio spike in\n",
    "\n",
    "\n",
    "#Activate Qiime2\n",
    "###to activate: source activate qiime2-2020.2\n",
    "###to exit environment: conda deactivate\n",
    "\n",
    "#import files, note need absolute file path included in manifest doc\n",
    "##files need to be unzipped\n",
    "\n",
    "#Make sure you are in the proper directory (folder) and your files are there as well\n",
    "pwd: /Users/alyssademko/Desktop/BZ15_2019/2020_ReAnalysis\n",
    "#files:\n",
    "pe-33-manifest.csv #file with path to raw reads for each sample\n",
    "BZ15_metadata.tsv # include all metadata here for easier visualization/stats analysis later on\n",
    "raw_reads/ #make sure they are unzipped!\n",
    "\n",
    "#Import files\n",
    "qiime tools import \\\n",
    "    --type 'SampleData[PairedEndSequencesWithQuality]' \\\n",
    "           --input-path pe-33-manifest-bzcontrols2.csv \\\n",
    "                        --output-path demux.qza \\\n",
    "                                      --input-format PairedEndFastqManifestPhred33\n",
    "\n",
    "qiime demux summarize \\\n",
    "            --i-data demux.qza \\\n",
    "                     --o-visualization DEMUX_SEQ_SUMMARY_VIZ.qzv\n",
    "\n",
    "\n",
    "#DADA2 denoising (takes a few hours- run overnight)\n",
    "#this trims off the primers and trims the sequences (Reverse were more crappy which is why they were trimmed shorter), also removes chimeras (default is consensus method)\n",
    "#decide metrics after looking at demux.qzv above\n",
    "\n",
    "qiime dada2 denoise-paired \\\n",
    "            --i-demultiplexed-seqs demux.qza \\\n",
    "                                   --p-trunc-len-f 250 \\\n",
    "                                                   --p-trunc-len-r 155 \\\n",
    "                                                                   --p-trim-left-f 19 \\\n",
    "                                                                                   --p-trim-left-r 20 \\\n",
    "                                                                                                   --o-table table.qza \\\n",
    "                                                                                                             --o-representative-sequences rep-seqs.qza \\\n",
    "                                                                                                                                          --o-denoising-stats denoising-stats.qza\n",
    "\n",
    "\n",
    "\n",
    "#visualize results from feature table\n",
    "qiime feature-table summarize \\\n",
    "                    --i-table table.qza \\\n",
    "                              --o-visualization table.qzv \\\n",
    "                                                --m-sample-metadata-file BZ15_metadata.tsv\n",
    "\n",
    "#visualize results summaries\n",
    "qiime feature-table tabulate-seqs \\\n",
    "                    --i-data rep-seqs.qza \\\n",
    "                             --o-visualization rep-seqs.qzv\n",
    "\n",
    "#visualize denoising stats\n",
    "qiime metadata tabulate \\\n",
    "               --m-input-file denoising-stats.qza \\\n",
    "                              --o-visualization denoising-stats.qzv\n",
    "\n",
    "#everything looks good from files above\n",
    "#lowest depth after above step is AD069 @ 68,943 followed by AD072 @ 87,757\n",
    "#will subsequently still need to remove chloroplast, singletons, etc\n",
    "\n",
    "\n",
    "#Generate a tree for phylogenetic diversity analyses, start @ 1pm, finished 1:56pm\n",
    "qiime phylogeny align-to-tree-mafft-fasttree \\\n",
    "                --i-sequences rep-seqs.qza \\\n",
    "                              --o-alignment aligned-rep-seqs.qza \\\n",
    "                                            --o-masked-alignment masked-aligned-rep-seqs.qza \\\n",
    "                                                                 --o-tree unrooted-tree.qza \\\n",
    "                                                                          --o-rooted-tree rooted-tree.qza\n",
    "\n",
    "\n",
    "#Taxonomy\n",
    "#make sure to use the classifier trained on the right scikit for this to work\n",
    "#can take a few hours\n",
    "qiime feature-classifier classify-sklearn \\\n",
    "                         --i-classifier silva-132-99-515-806-nb-classifier.qza \\\n",
    "                                        --i-reads rep-seqs.qza \\\n",
    "                                                  --o-classification silva-taxonomy.qza\n",
    "\n",
    "\n",
    "qiime metadata tabulate \\\n",
    "               --m-input-file silva-taxonomy.qza \\\n",
    "                              --o-visualization silva-taxonomy.qzv\n",
    "\n",
    "\n",
    "# need to get rid of chloroplasts! might as well do mitochondria too although not usually an issue\n",
    "qiime taxa filter-table \\\n",
    "           --i-table table.qza \\\n",
    "                     --i-taxonomy silva-taxonomy.qza \\\n",
    "                                  --p-exclude mitochondria,chloroplast \\\n",
    "                                                           --o-filtered-table silva-filtered-table.qza\n",
    "\n",
    "#visualize results from feature table\n",
    "qiime feature-table summarize \\\n",
    "                    --i-table silva-filtered-table.qza \\\n",
    "                              --o-visualization silva-filtered-table.qzv \\\n",
    "                                                --m-sample-metadata-file BZ15_metadata.tsv\n",
    "\n",
    "\n",
    "\n",
    "#Taxonomy Bar plots\n",
    "#visualize filtered taxa bar plots\n",
    "qiime taxa barplot \\\n",
    "           --i-table silva-filtered-table.qza \\\n",
    "                     --i-taxonomy silva-taxonomy.qza \\\n",
    "                                  --m-metadata-file BZ15_metadata.tsv \\\n",
    "                                                    --o-visualization silva-filtered-taxa-bar-plots.qzv\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "#####Filtering step specific to my experiment (threshold set at 70 reads)\n",
    "\n",
    "qiime feature-table filter-features \\\n",
    "                    --i-table silva-filtered-table.qza \\\n",
    "                              --p-min-frequency 70 \\\n",
    "                                                --o-filtered-table silva-70-filtered-table.qza\n",
    "\n",
    "#visualize\n",
    "qiime feature-table summarize \\\n",
    "                    --i-table silva-70-filtered-table.qza \\\n",
    "                              --o-visualization silva-70-filtered-table.qzv \\\n",
    "                                                --m-sample-metadata-file BZ15_metadata.tsv\n",
    "\n",
    "#Now lowest depth is AD069 @ 56,983 followed by AD067 @ 66,676\n",
    "\n",
    "Taxonomy Bar plots\n",
    "#visualize filtered taxa bar plots\n",
    "qiime taxa barplot \\\n",
    "           --i-table silva-70-filtered-table.qza \\\n",
    "                     --i-taxonomy silva-taxonomy.qza \\\n",
    "                                  --m-metadata-file BZ15_metadata.tsv \\\n",
    "                                                    --o-visualization silva-70-filtered-taxa-bar-plots.qzv\n",
    "\n",
    "\n",
    "#Results indicate that there are still ~50 genera that we only see in CD sampling! this includes streptomyces\n",
    "#continue with analysis, first rarefaction\n",
    "\n",
    "\n",
    "ALPHA RAREFACTION\n",
    "#Alpha rarefaction makes table of diversity at different depths for your curve\n",
    "#for first one, lets look at the unlimited diversity by doing max depth of 266,648\n",
    "qiime diversity alpha-rarefaction \\\n",
    "                --i-table silva-70-filtered-table.qza \\\n",
    "                          --i-phylogeny rooted-tree.qza \\\n",
    "                                        --p-max-depth 266648 \\\n",
    "                                                      --m-metadata-file BZ15_metadata.tsv \\\n",
    "                                                                        --o-visualization alpha-rarefaction.qzv\n",
    "\n",
    "#not enough steps, redo with >10 (default)\n",
    "qiime diversity alpha-rarefaction \\\n",
    "                --i-table silva-70-filtered-table.qza \\\n",
    "                          --i-phylogeny rooted-tree.qza \\\n",
    "                                        --p-steps 20 \\\n",
    "                                                  --p-max-depth 266648 \\\n",
    "                                                                --m-metadata-file BZ15_metadata.tsv \\\n",
    "                                                                                  --o-visualization alpha-rarefaction-20steps.qzv\n",
    "\n",
    "\n",
    "\n",
    "#### Lets Rarefy to 56,983 to include all samples and go from there...\n",
    "\n",
    "qiime diversity alpha-rarefaction \\\n",
    "                --i-table silva-70-filtered-table.qza \\\n",
    "                          --i-phylogeny rooted-tree.qza \\\n",
    "                                        --p-max-depth 56983 \\\n",
    "                                                      --m-metadata-file BZ15_metadata.tsv \\\n",
    "                                                                        --o-visualization alpha-rarefaction-56k.qzv\n",
    "\n",
    "#I think it looks okay in terms of being saturated\n",
    "\n",
    "\n",
    "#now do diversity core metrices\n",
    "qiime diversity core-metrics-phylogenetic \\\n",
    "                --i-phylogeny rooted-tree.qza \\\n",
    "                              --i-table silva-70-filtered-table.qza \\\n",
    "                                        --p-sampling-depth 56983 \\\n",
    "                                                           --m-metadata-file BZ15_metadata.tsv \\\n",
    "                                                                             --output-dir 56k/core-metrics-results-56k#%%\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "source": [
    "import plotly.graph_objects as go\n",
    "\n",
    "# Sample data: taxonomic composition of multiple microbiome samples\n",
    "microbiomes = {\n",
    "    'Sample1': {\n",
    "        'Other': 0.063926449,\n",
    "        'Euryarchaeota': 0.052744073,\n",
    "        'Actinobacteria': 0.004597809,\n",
    "        'Bacteroidetes': 0.069751442,\n",
    "        'Chlamydiae': 4.25673E-05,\n",
    "        'Chloroflexi': 0.007159006,\n",
    "        'Cyanobacteria': 0.005425458,\n",
    "        'Elusimicrobia': 0.000692495,\n",
    "        'Fibrobacteres': 0.009562012,\n",
    "        'Firmicutes': 0.731381689,\n",
    "        'Fusobacteria': 4.7717E-05,\n",
    "        'Lentisphaerae': 0.005097325,\n",
    "        'NA': 0.002225507,\n",
    "        'Nitrospirae': 3.89827E-05,\n",
    "        'Planctomycetes': 0.010283447,\n",
    "        'Proteobacteria': 0.006749815,\n",
    "        'Saccharibacteria': 2.46308E-05,\n",
    "        'Spirochaetae': 0.001025642,\n",
    "        'Synergistetes': 0.021118209,\n",
    "        'Tenericutes': 0.002160078,\n",
    "        'Verrucomicrobia': 0.005945647,\n",
    "    },\n",
    "}\n",
    "\n",
    "# Create a pie chart for each sample\n",
    "for sample in microbiomes:\n",
    "    # Get the taxonomic composition of the sample\n",
    "    composition = microbiomes[sample]\n",
    "\n",
    "    # Create a list of labels and values for the pie chart\n",
    "    labels = []\n",
    "    values = []\n",
    "    for domain in composition:\n",
    "        labels.append('{}:'.format(domain))\n",
    "        values.append(composition[domain])\n",
    "\n",
    "    # Create a pie chart using Plotly\n",
    "    fig = go.Figure(data=[go.Pie(labels=labels, values=values)])\n",
    "    fig.update_layout(title_text='Total Taxonomic Composition', legend=dict(orientation=\"h\", x=-1, y=.8))\n",
    "    fig.show()\n"
   ],
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "import plotly.graph_objects as go\n",
    "\n",
    "# Sample data: taxonomic composition of multiple microbiome samples\n",
    "microbiomes = {\n",
    "    'Sample1': {\n",
    "        'Other': 0.063926449,\n",
    "        'Euryarchaeota': 0.052744073,\n",
    "        'Actinobacteria': 0.004597809,\n",
    "        'Bacteroidetes': 0.069751442,\n",
    "        'Chlamydiae': 4.25673E-05,\n",
    "        'Chloroflexi': 0.007159006,\n",
    "        'Cyanobacteria': 0.005425458,\n",
    "        'Elusimicrobia': 0.000692495,\n",
    "        'Fibrobacteres': 0.009562012,\n",
    "        'Firmicutes': 0.731381689,\n",
    "        'Fusobacteria': 4.7717E-05,\n",
    "        'Lentisphaerae': 0.005097325,\n",
    "        'NA': 0.002225507,\n",
    "        'Nitrospirae': 3.89827E-05,\n",
    "        'Planctomycetes': 0.010283447,\n",
    "        'Proteobacteria': 0.006749815,\n",
    "        'Saccharibacteria': 2.46308E-05,\n",
    "        'Spirochaetae': 0.001025642,\n",
    "        'Synergistetes': 0.021118209,\n",
    "        'Tenericutes': 0.002160078,\n",
    "        'Verrucomicrobia': 0.005945647,\n",
    "    },\n",
    "}\n",
    "plt.figure(dpi=1200)\n",
    "# Create a pie chart for each sample\n",
    "for sample in microbiomes:\n",
    "    # Get the taxonomic composition of the sample\n",
    "    composition = microbiomes[sample]\n",
    "\n",
    "    # Create a list of labels and values for the pie chart\n",
    "    labels = []\n",
    "    values = []\n",
    "    for domain in composition:\n",
    "        percentage = composition[domain] * 100\n",
    "        labels.append('{}: {:.2f}%'.format(domain, percentage))\n",
    "        values.append(composition[domain])\n",
    "\n",
    "    # Create a pie chart using Plotly\n",
    "    fig = go.Figure(data=[go.Pie(labels=labels, values=values)])\n",
    "    fig.update_layout(title_text='Total Taxonomic Composition', legend=dict(orientation=\"h\", x=-1, y=.8))\n",
    "    fig.show()\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": [
    "import plotly.graph_objects as go\n",
    "\n",
    "# Sample data: taxonomic composition of multiple microbiome samples\n",
    "microbiomes = {\n",
    "    'Sample1': {\n",
    "        'Other': 0.063926449,\n",
    "        'Euryarchaeota': 0.052744073,\n",
    "        'Actinobacteria': 0.004597809,\n",
    "        'Bacteroidetes': 0.069751442,\n",
    "        'Chlamydiae': 4.25673E-05,\n",
    "        'Chloroflexi': 0.007159006,\n",
    "        'Cyanobacteria': 0.005425458,\n",
    "        'Elusimicrobia': 0.000692495,\n",
    "        'Fibrobacteres': 0.009562012,\n",
    "        'Firmicutes': 0.731381689,\n",
    "        'Fusobacteria': 4.7717E-05,\n",
    "        'Lentisphaerae': 0.005097325,\n",
    "        'NA': 0.002225507,\n",
    "        'Nitrospirae': 3.89827E-05,\n",
    "        'Planctomycetes': 0.010283447,\n",
    "        'Proteobacteria': 0.006749815,\n",
    "        'Saccharibacteria': 2.46308E-05,\n",
    "        'Spirochaetae': 0.001025642,\n",
    "        'Synergistetes': 0.021118209,\n",
    "        'Tenericutes': 0.002160078,\n",
    "        'Verrucomicrobia': 0.005945647,\n",
    "    },\n",
    "}\n",
    "plt.figure(dpi=1200)\n",
    "# Create a pie chart for each sample\n",
    "for sample in microbiomes:\n",
    "    # Get the taxonomic composition of the sample\n",
    "    composition = microbiomes[sample]\n",
    "\n",
    "    # Create a list of labels and values for the pie chart\n",
    "    labels = []\n",
    "    values = []\n",
    "    for domain in composition:\n",
    "        percentage = composition[domain] * 100\n",
    "        labels.append('{}: {:.2f}%'.format(domain, percentage))\n",
    "        values.append(composition[domain])\n",
    "\n",
    "    # Create a pie chart using Plotly\n",
    "    fig = go.Figure(data=[go.Pie(labels=labels, values=values)])\n",
    "    fig.update_layout(title_text='Total Taxonomic Composition', legend=dict(orientation=\"h\", x=-1, y=.8))\n",
    "    fig.show()\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "import plotly.subplots as sp\n",
    "import plotly.graph_objects as go\n",
    "\n",
    "# Sample data: taxonomic composition of multiple microbiome samples\n",
    "microbiomes = {\n",
    "    'Sample1': {\n",
    "        'Bacteria': {\n",
    "            'Firmicutes': 0.35,\n",
    "            'Bacteroidetes': 0.25,\n",
    "            'Proteobacteria': 0.20,\n",
    "            'Actinobacteria': 0.10,\n",
    "            'Fusobacteria': 0.05,\n",
    "            'Other': 0.05,\n",
    "        },\n",
    "        'Archaea': {\n",
    "            'Euryarchaeota': 0.60,\n",
    "            'Crenarchaeota': 0.20,\n",
    "            'Other': 0.20,\n",
    "        },\n",
    "    },\n",
    "    'Sample2': {\n",
    "        'Bacteria': {\n",
    "            'Firmicutes': 0.40,\n",
    "            'Bacteroidetes': 0.20,\n",
    "            'Proteobacteria': 0.15,\n",
    "            'Actinobacteria': 0.10,\n",
    "            'Fusobacteria': 0.10,\n",
    "            'Other': 0.05,\n",
    "        },\n",
    "        'Archaea': {\n",
    "            'Euryarchaeota': 0.70,\n",
    "            'Crenarchaeota': 0.10,\n",
    "            'Other': 0.20,\n",
    "        },\n",
    "    },\n",
    "    'Sample3': {\n",
    "        'Bacteria': {\n",
    "            'Firmicutes': 0.30,\n",
    "            'Bacteroidetes': 0.30,\n",
    "            'Proteobacteria': 0.10,\n",
    "            'Actinobacteria': 0.10,\n",
    "            'Fusobacteria': 0.10,\n",
    "            'Other': 0.10,\n",
    "        },\n",
    "        'Archaea': {\n",
    "            'Euryarchaeota': 0.80,\n",
    "            'Other': 0.20,\n",
    "        },\n",
    "    },\n",
    "}\n",
    "\n",
    "# Create a subplot of pie charts for each sample\n",
    "fig = sp.make_subplots(rows=len(microbiomes), cols=1, specs=[[{'type':'pie'}] for i in range(len(microbiomes))], subplot_titles=list(microbiomes.keys()))\n",
    "\n",
    "for i, sample in enumerate(microbiomes):\n",
    "    # Get the taxonomic composition of the sample\n",
    "    composition = microbiomes[sample]\n",
    "\n",
    "    # Create a list of labels and values for the pie chart\n",
    "    labels = []\n",
    "    values = []\n",
    "    for domain in composition:\n",
    "        for phylum in composition[domain]:\n",
    "            labels.append('{}: {}'.format(domain, phylum))\n",
    "            values.append(composition[domain][phylum])\n",
    "\n",
    "    # Add a pie chart to the subplot\n",
    "    fig.add_trace(go.Pie(labels=labels, values=values), row=i+1, col=1)\n",
    "    fig.update_layout(title='Taxonomic Composition of Multiple Microbiome Samples', height=300, width=300)\n",
    "\n",
    "fig.show()\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": [
    "import plotly.graph_objects as go\n",
    "\n",
    "# New data: taxonomic composition of multiple microbiome samples\n",
    "microbiomes = {\n",
    "    'Sample1': {\n",
    "        'Euryarchaeota': 0.072714802,\n",
    "        'Bacteroidetes': 0.07093182,\n",
    "        'Chloroflexi': 0.005602122,\n",
    "        'Fibrobacteres': 0.012134267,\n",
    "        'Firmicutes': 0.736295756,\n",
    "        'Proteobacteria': 0.003416498,\n",
    "        'Synergistetes': 0.004515492,\n",
    "        'Verrucomicrobia': 0.014220982,\n",
    "        'seq130': 0.003823125,\n",
    "        'seq171': 0.003160981,\n",
    "        'seq4': 0.048508221,\n",
    "    },\n",
    "}\n",
    "\n",
    "plt.figure(dpi=1200)\n",
    "# Create a pie chart for each sample\n",
    "for sample in microbiomes:\n",
    "    # Get the taxonomic composition of the sample\n",
    "    composition = microbiomes[sample]\n",
    "\n",
    "    # Create a list of labels and values for the pie chart\n",
    "    labels = []\n",
    "    values = []\n",
    "    for domain in composition:\n",
    "        percentage = composition[domain] * 100\n",
    "        labels.append('{}: {:.2f}%'.format(domain, percentage))\n",
    "        values.append(composition[domain])\n",
    "\n",
    "    # Create a pie chart using Plotly\n",
    "    fig = go.Figure(data=[go.Pie(labels=labels, values=values)])\n",
    "    fig.update_layout(title_text='Core Microbiome', legend=dict(orientation=\"h\", x=-1, y=.8))\n",
    "    fig.show()\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "import matplotlib.pyplot as plt\n",
    "\n",
    "# Sample data: taxonomic composition of multiple microbiome samples\n",
    "microbiomes = {\n",
    "    'Sample1': {\n",
    "        'Bacteria': {\n",
    "            'Firmicutes': 0.35,\n",
    "            'Bacteroidetes': 0.25,\n",
    "            'Proteobacteria': 0.20,\n",
    "            'Actinobacteria': 0.10,\n",
    "            'Fusobacteria': 0.05,\n",
    "            'Other': 0.05,\n",
    "        },\n",
    "        'Archaea': {\n",
    "            'Euryarchaeota': 0.60,\n",
    "            'Crenarchaeota': 0.20,\n",
    "            'Other': 0.20,\n",
    "        },\n",
    "    },\n",
    "    'Sample2': {\n",
    "        'Bacteria': {\n",
    "            'Firmicutes': 0.40,\n",
    "            'Bacteroidetes': 0.20,\n",
    "            'Proteobacteria': 0.15,\n",
    "            'Actinobacteria': 0.10,\n",
    "            'Fusobacteria': 0.10,\n",
    "            'Other': 0.05,\n",
    "        },\n",
    "        'Archaea': {\n",
    "            'Euryarchaeota': 0.70,\n",
    "            'Crenarchaeota': 0.10,\n",
    "            'Other': 0.20,\n",
    "        },\n",
    "    },\n",
    "    'Sample3': {\n",
    "        'Bacteria': {\n",
    "            'Firmicutes': 0.30,\n",
    "            'Bacteroidetes': 0.30,\n",
    "            'Proteobacteria': 0.10,\n",
    "            'Actinobacteria': 0.10,\n",
    "            'Fusobacteria': 0.10,\n",
    "            'Other': 0.10,\n",
    "        },\n",
    "        'Archaea': {\n",
    "            'Euryarchaeota': 0.80,\n",
    "            'Other': 0.20,\n",
    "        },\n",
    "    },\n",
    "}\n",
    "\n",
    "# Create a subplot of pie charts for each sample\n",
    "fig, axs = plt.subplots(len(microbiomes), 1, figsize=(8, 8))\n",
    "fig.suptitle('Taxonomic Composition of Multiple Microbiome Samples')\n",
    "\n",
    "for i, sample in enumerate(microbiomes):\n",
    "    # Get the taxonomic composition of the sample\n",
    "    composition = microbiomes[sample]\n",
    "\n",
    "    # Create a list of labels and values for the pie chart\n",
    "    labels = []\n",
    "    values = []\n",
    "    for domain in composition:\n",
    "        for phylum in composition[domain]:\n",
    "            labels.append('{}: {}'.format(domain, phylum))\n",
    "            values.append(composition[domain][phylum])\n",
    "\n",
    "    # Add a pie chart to the subplot\n",
    "    axs[i].pie(values, labels=labels, autopct='%1.1f%%', startangle=90)\n",
    "    axs[i].axis('equal')\n",
    "    axs[i].set_title(sample)\n",
    "\n",
    "plt.show()\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "fig, ax = plt.subplots()\n",
    "\n",
    "size = 0.3\n",
    "vals = np.array([[60., 32.], [37., 40.], [29., 10.]])\n",
    "\n",
    "cmap = plt.colormaps[\"tab20c\"]\n",
    "outer_colors = cmap(np.arange(3)*4)\n",
    "inner_colors = cmap([1, 2, 5, 6, 9, 10])\n",
    "\n",
    "ax.pie(vals.sum(axis=1), radius=1, colors=outer_colors,\n",
    "       wedgeprops=dict(width=size, edgecolor='w'))\n",
    "\n",
    "ax.pie(vals.flatten(), radius=1-size, colors=inner_colors,\n",
    "       wedgeprops=dict(width=size, edgecolor='w'))\n",
    "\n",
    "ax.set(aspect=\"equal\", title='Pie plot with `ax.pie`')\n",
    "plt.show()"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "import plotly.express as px\n",
    "\n",
    "wide_df = px.data.medals_wide()\n",
    "\n",
    "fig = px.bar(wide_df, x=\"nation\", y=[\"gold\", \"silver\", \"bronze\"], title=\"Wide-Form Input\")\n",
    "fig.show()"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "import matplotlib.pyplot as plt\n",
    "\n",
    "# Sample data\n",
    "x = [1, 2, 3, 4, 5]\n",
    "y = [2, 4, 6, 8, 10]\n",
    "\n",
    "# Create a line graph\n",
    "plt.plot(x, y)\n",
    "\n",
    "# Add labels and a title\n",
    "plt.xlabel('X-axis label')\n",
    "plt.ylabel('Y-axis label')\n",
    "plt.title('Line Graph Example')\n",
    "\n",
    "# Display the graph\n",
    "plt.show()\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "import pandas as pd\n",
    "import plotly.express as px\n",
    "\n",
    "\n",
    "# Example OTU table with 16 samples\n",
    "otu_table = pd.DataFrame({\n",
    "    'OTU': ['Other', 'Euryarchaeota', 'Actinobacteria', 'Bacteroidetes', 'Chlamydiae', 'Chloroflexi', 'Cyanobacteria', 'Elusimicrobia', 'Fibrobacteres', 'Firmicutes', 'Fusobacteria', 'Lentisphaerae', 'NA', 'Nitrospirae', 'Planctomycetes', 'Proteobacteria', 'Saccharibacteria', 'Spirochaetae', 'Synergistetes', 'Tenericutes', 'Verrucomicrobia'],\n",
    "    'T20': [0.0692213560163, 0.0512763463036, 0.00552812411751, 0.0460905208931, 0.0, 0.00294376887991, 0.00377384324431, 0.000573350334169, 0.0061956065961, 0.772371359867, 0.000119804547438, 0.00142053963391, 0.00117237307136, 0.0, 0.00855746767417, 0.00939609950623, 0.0, 0.000650367543237, 0.0173887743139, 0.00139486723089, 0.00192543022669],\n",
    "    'T107': [0.0886399156762, 0.0700123577945, 0.00510667684367, 0.0883764038818, 0.0, 0.00936829862247, 0.00912296005525, 0.0, 0.0122851015883, 0.665294588013, 0.000208992112819, 0.00439792098281, 0.000508850361647, 0.000118125976811, 0.00885944826082, 0.00822338530876, 0.0, 0.000954094428089, 0.0118398575219, 0.00336204703231, 0.0133209755388],\n",
    "    'T203': [0.091516034342, 0.0439206338697, 0.00434792204639, 0.0828312255843, 0.0, 0.0188483524245, 0.00508728950098, 0.00070626144916, 0.0218609989185, 0.655399589486, 0.0, 0.00456862874926, 0.00194221898519, 0.0, 0.0204815820256, 0.00940210554195, 0.0, 0.00150080557947, 0.0326094153479, 0.00264848043435, 0.0023284557152],\n",
    "    'T11': [0.0387414787625, 0.053864708967, 0.00231777661248, 0.0659884635553, 0.0, 0.0028736234924, 0.00484530676455, 0.0, 0.00624016780283, 0.765977975878, 0.0, 0.00684845306765, 0.00293654955427, 0.0, 0.011295228107, 0.011777661248, 0.0, 0.00378605138962, 0.019979024646, 0.000964866282119, 0.00156266386995],\n",
    "    'T212': [0.0662810452229, 0.0916511609093, 0.00777445190519, 0.0822587169323, 0.0, 0.013558773562, 0.00245125798884, 0.00040449801796, 0.00329261386619, 0.667834317612, 0.000226518890057, 0.00792816115201, 0.00574387185503, 0.00020224900898, 0.0129358466144, 0.00594612086401, 0.000275058652213, 0.000493487581911, 0.02409999191, 0.0011892241728, 0.0054526332821],\n",
    "    'T85.1': [0.0510423403203, 0.0536196599079, 0.00868299920113, 0.070557690113, 0.0, 0.000551603454179, 0.00618176284856, 0.000266291322707, 0.00414653631072, 0.739947502568, 0.0, 0.00664777266329, 0.00207326815536, 0.0, 0.0124586297409, 0.00714231369118, 0.0, 0.000884467607563, 0.0186594133983, 0.00222543462548, 0.0149123140716],\n",
    "    'T65': [0.0730575036454, 0.0383787483509, 0.00480106337602, 0.0652210572259, 0.0, 0.00343216514071, 0.0076578944758, 0.00488041979546, 0.0331114660107, 0.728412574025, 0.0, 0.00436460306911, 0.0, 0.0, 0.00758845760879, 0.00395790141949, 0.000119034629158, 0.000247988810745, 0.0178452748212, 0.00344208469314, 0.00348176290286],\n",
    "    'T196': [0.0808211256746, 0.0581534309946, 0.0115940632228, 0.0606399383192, 0.0, 0.0177525057826, 0.00546453353894, 0.00122397841172, 0.00293947571318, 0.692530840401, 0.0, 0.00149383191981, 0.000857748650732, 0.00011565150347, 0.0143022359291, 0.00536815728604, 0.0, 0.000395142636854, 0.0429163454125, 0.00146491904395, 0.00196607555898],\n",
    "    'T99': [0.0524611702876, 0.0459035240017, 0.00296795082352, 0.0678522828586, 0.000129041340153, 0.00292102669983, 0.0018886959786, 0.000164234432922, 0.00140772371076, 0.785157899676, 0.0, 0.00446952278166, 0.00170099948383, 0.000187696494768, 0.00476279855474, 0.00735535638872, 0.0, 0.00082117216461, 0.0105931209235, 0.00208812350429, 0.00716765989395],\n",
    "    'T85.2': [0.0522599376322, 0.0657281622997, 0.00296605613104, 0.0922613713753, 0.00012545252518, 0.00324384386537, 0.0172945266855, 0.0018817878777, 0.00709702856733, 0.685409871322, 0.0, 0.00685508441163, 0.00260762034482, 0.0, 0.0146958672354, 0.00730312914441, 0.0, 0.000143374314492, 0.0252159575612, 0.00278683823793, 0.0121240904692],\n",
    "    'TAK': [0.0483987105493, 0.0419514572786, 0.00371104784482, 0.0433915653975, 0.0, 0.00776550608723, 0.00360027029722, 0.0, 0.00356703703293, 0.806272224745, 0.0, 0.00254788359495, 0.0023374062545, 0.0, 0.00550564411605, 0.00521762249227, 0.0, 0.000919453645135, 0.020504924062, 0.00177244076171, 0.00253680584019],\n",
    "    'T85.3': [0.090085008662, 0.0447403408404, 0.00357895867746, 0.0645421215906, 0.000154439117414, 0.00243073741321, 0.00817184373447, 0.000389455165653, 0.0104078535649, 0.723305534292, 0.000208157071297, 0.0052173562709, 0.000490176329184, 0.0, 0.00637229227939, 0.00932677974296, 0.0, 0.000705048144716, 0.0151014597854, 0.00427057733371, 0.0105018599842],\n",
    "    'T195': [0.0565088200213, 0.0267216305007, 0.00165310008579, 0.0622214317102, 0.0, 0.00525225470297, 0.00124505639373, 0.0, 0.0188223231288, 0.795946765992, 0.0, 0.00277260457427, 0.00187281591998, 0.0, 0.00486513632844, 0.002008830484, 0.0, 0.00269936596288, 0.00947916884639, 0.00177865199104, 0.00615204335726],\n",
    "    'T205': [0.0745835897329, 0.0343140833759, 0.00274304070756, 0.0786094973873, 0.0, 0.0101794970744, 0.000500631003661, 0.0, 0.014841623296, 0.743635206875, 0.0, 0.00377559215261, 0.0032958207741, 0.0, 0.00808310474661, 0.00238842707997, 0.0, 0.00171048926251, 0.0154048331751, 0.00132458619719, 0.00460997715871],\n",
    "    'T159': [0.0588736891759, 0.0825048078668, 0.00325664937044, 0.0711836423673, 0.00027214340143, 0.0111215936718, 0.00707572843717, 0.000589644036431, 0.00576036866359, 0.679269929968, 0.0, 0.0143963859356, 0.00654958452774, 0.0, 0.0098788054719, 0.0094978047099, 0.0, 0.000353786421859, 0.0329021372328, 0.00219529010487, 0.00431800863602],\n",
    "    'TAA': [0.0303314571607, 0.0411641204324, 0.00253506655946, 0.0739971410703, 0.0, 0.00230054498347, 0.0024457250067, 0.0, 0.0010162601626, 0.795340838024, 0.0, 0.00385285446261, 0.00151880639686, 0.0, 0.0138926114536, 0.00368533905119, 0.0, 0.000145180023229, 0.0233516483516, 0.00165281872599, 0.00276958813544]}\n",
    ")\n",
    "otu_table.iloc[:, 1:] *= 100\n",
    "plt.figure(dpi=600)\n",
    "# Calculate average relative abundance for each OTU\n",
    "otu_table['Average'] = otu_table.iloc[:, 1:].mean(axis=1)\n",
    "\n",
    "# Melt DataFrame\n",
    "df_melted = pd.melt(otu_table, id_vars=['OTU'], var_name='T', value_name='Relative Abundance')\n",
    "\n",
    "# Set y-axis range to 0-100%\n",
    "fig.update_yaxes(range=[0, 100])\n",
    "fig = px.bar(df_melted, x='T', y='Relative Abundance', color='OTU',\n",
    "             title='', barmode='stack',\n",
    "             color_discrete_sequence=px.colors.qualitative.Alphabet)\n",
    "fig.update_layout(\n",
    "    xaxis_title=\"Sample ID\"\n",
    ")\n",
    "# Update layout\n",
    "fig.update_layout(legend=dict(orientation=\"h\", yanchor=\"bottom\", y=1.02, xanchor=\"right\", x=1),\n",
    "                  margin=dict(t=100, b=50, l=50, r=50))\n",
    "fig.update_layout(plot_bgcolor='rgba(255, 255, 255, 1)')\n",
    "# Show plot\n",
    "fig.show()\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "import numpy as np\n",
    "from scipy.linalg import solve\n",
    "from sympy import symbols, Eq, integrate, sin, pi\n",
    "\n",
    "# Solving a system of linear equations using SciPy\n",
    "# 3x + 2y = 2\n",
    "# x - 4y = 11\n",
    "\n",
    "A = np.array([[3, 2], [1, -4]])\n",
    "B = np.array([2, 11])\n",
    "x, y = solve(A, B)\n",
    "print(\"Solving linear equations with SciPy:\")\n",
    "print(f\"x = {x:.2f}, y = {y:.2f}\\n\")\n",
    "\n",
    "# Performing symbolic integration using SymPy\n",
    "x = symbols('x')\n",
    "expr = sin(2 * pi * x)\n",
    "integral = integrate(expr, (x, 0, 1))\n",
    "print(\"Symbolic integration with SymPy:\")\n",
    "print(f\"Integral of sin(2*pi*x) from 0 to 1: {integral}\")\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "from sympy import symbols, Eq, solve, simplify_logic\n",
    "\n",
    "# Define the symbols used in the logic equation\n",
    "A, B, C = symbols(\"A B C\")\n",
    "\n",
    "# Define the logic equation\n",
    "equation = Eq((A & ~B) | (B & C), True)\n",
    "\n",
    "# Solve the equation for all combinations of A, B, and C\n",
    "solutions = []\n",
    "\n",
    "for a_val in [True, False]:\n",
    "    for b_val in [True, False]:\n",
    "        for c_val in [True, False]:\n",
    "            if equation.subs({A: a_val, B: b_val, C: c_val}):\n",
    "                solutions.append({A: a_val, B: b_val, C: c_val})\n",
    "\n",
    "print(\"Solutions:\")\n",
    "for solution in solutions:\n",
    "    print(solution)\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": "/usr/local/bin/python3.9 -m pip install sympy",
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": "!/usr/local/bin/python3.9 -m pip install PdfReader\n",
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "from ete3 import Tree, TreeStyle, NodeStyle, faces, TextFace\n",
    "\n",
    "# Read the Newick-formatted string\n",
    "newick_string = \"(( (Bacteroides fragilis NCTC 9343_Phylum_Bacteriodetes:0.14775, Bifidobacterium longum subsp. longum JCM 1217_Phylum_Actinobacteria:0.13892) :0.01857, Helicobacter pylori strain ATCC 43504_Phylum_Proteobacteria:0.11945) :0.00449, Ruminococcus bromii strain ATCC 27255_Phylum_Firmicutes:0.11881, (Clostridium butyricum strain VPI3266_Phylum_Firmicutes:0.03359,Clostridium perfringens ATCC 13124_Phylum_Firmicutes:0.04132) :0.05316);\"\n",
    "t = Tree(newick_string)\n",
    "\n",
    "# Define the layout function\n",
    "def custom_layout(node):\n",
    "    if node.is_leaf():\n",
    "        # Create an italic face for the name\n",
    "        name_face = TextFace(node.name, fsize=14, fgcolor=\"black\", ftype=\"Arial\", fstyle=\"italic\")\n",
    "\n",
    "        # Add the face to the leaf node\n",
    "        faces.add_face_to_node(name_face, node, column=0, position=\"branch-right\")\n",
    "\n",
    "# Configure the tree style\n",
    "ts = TreeStyle()\n",
    "ts.show_leaf_name = False  # Set to False since we are using custom_layout function\n",
    "ts.show_branch_length = True\n",
    "ts.show_branch_support = True\n",
    "ts.layout_fn = custom_layout  # Set the custom layout function\n",
    "\n",
    "# Plot the tree\n",
    "t.render(\"%%inline\", w=800, units=\"px\", tree_style=ts)\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "Pipeline modified from Qiime2 Moving Pictures Tutorial\n",
    "Alyssa Demko\n",
    "\n",
    "Project: Belize 2015 sediment culture-independent vs dependent comparison\n",
    "06.13.2020\n",
    "\n",
    "Only looking at 2020 controls with Vibrio spike in\n",
    "\n",
    "\n",
    "#Activate Qiime2\n",
    "###to activate: source activate qiime2-2020.2\n",
    "###to exit environment: conda deactivate\n",
    "\n",
    "#import files, note need absolute file path included in manifest doc\n",
    "##files need to be unzipped\n",
    "\n",
    "#Make sure you are in the proper directory (folder) and your files are there as well\n",
    "pwd: /Users/alyssademko/Desktop/BZ15_2019/2020_ReAnalysis\n",
    "#files:\n",
    "pe-33-manifest.csv #file with path to raw reads for each sample\n",
    "BZ15_metadata.tsv # include all metadata here for easier visualization/stats analysis later on\n",
    "raw_reads/ #make sure they are unzipped!\n",
    "\n",
    "#Import files\n",
    "qiime tools import \\\n",
    "    --type 'SampleData[PairedEndSequencesWithQuality]' \\\n",
    "           --input-path pe-33-manifest-bzcontrols2.csv \\\n",
    "                        --output-path demux.qza \\\n",
    "                                      --input-format PairedEndFastqManifestPhred33\n",
    "\n",
    "qiime demux summarize \\\n",
    "            --i-data demux.qza \\\n",
    "                     --o-visualization DEMUX_SEQ_SUMMARY_VIZ.qzv\n",
    "\n",
    "\n",
    "#DADA2 denoising (takes a few hours- run overnight)\n",
    "#this trims off the primers and trims the sequences (Reverse were more crappy which is why they were trimmed shorter), also removes chimeras (default is consensus method)\n",
    "#decide metrics after looking at demux.qzv above\n",
    "\n",
    "qiime dada2 denoise-paired \\\n",
    "            --i-demultiplexed-seqs demux.qza \\\n",
    "                                   --p-trunc-len-f 250 \\\n",
    "                                                   --p-trunc-len-r 155 \\\n",
    "                                                                   --p-trim-left-f 19 \\\n",
    "                                                                                   --p-trim-left-r 20 \\\n",
    "                                                                                                   --o-table table.qza \\\n",
    "                                                                                                             --o-representative-sequences rep-seqs.qza \\\n",
    "                                                                                                                                          --o-denoising-stats denoising-stats.qza\n",
    "\n",
    "\n",
    "\n",
    "#visualize results from feature table\n",
    "qiime feature-table summarize \\\n",
    "                    --i-table table.qza \\\n",
    "                              --o-visualization table.qzv \\\n",
    "                                                --m-sample-metadata-file BZ15_metadata.tsv\n",
    "\n",
    "#visualize results summaries\n",
    "qiime feature-table tabulate-seqs \\\n",
    "                    --i-data rep-seqs.qza \\\n",
    "                             --o-visualization rep-seqs.qzv\n",
    "\n",
    "#visualize denoising stats\n",
    "qiime metadata tabulate \\\n",
    "               --m-input-file denoising-stats.qza \\\n",
    "                              --o-visualization denoising-stats.qzv\n",
    "\n",
    "#everything looks good from files above\n",
    "#lowest depth after above step is AD069 @ 68,943 followed by AD072 @ 87,757\n",
    "#will subsequently still need to remove chloroplast, singletons, etc\n",
    "\n",
    "\n",
    "#Generate a tree for phylogenetic diversity analyses, start @ 1pm, finished 1:56pm\n",
    "qiime phylogeny align-to-tree-mafft-fasttree \\\n",
    "                --i-sequences rep-seqs.qza \\\n",
    "                              --o-alignment aligned-rep-seqs.qza \\\n",
    "                                            --o-masked-alignment masked-aligned-rep-seqs.qza \\\n",
    "                                                                 --o-tree unrooted-tree.qza \\\n",
    "                                                                          --o-rooted-tree rooted-tree.qza\n",
    "\n",
    "\n",
    "#Taxonomy\n",
    "#make sure to use the classifier trained on the right scikit for this to work\n",
    "#can take a few hours\n",
    "qiime feature-classifier classify-sklearn \\\n",
    "                         --i-classifier silva-132-99-515-806-nb-classifier.qza \\\n",
    "                                        --i-reads rep-seqs.qza \\\n",
    "                                                  --o-classification silva-taxonomy.qza\n",
    "\n",
    "\n",
    "qiime metadata tabulate \\\n",
    "               --m-input-file silva-taxonomy.qza \\\n",
    "                              --o-visualization silva-taxonomy.qzv\n",
    "\n",
    "\n",
    "# need to get rid of chloroplasts! might as well do mitochondria too although not usually an issue\n",
    "qiime taxa filter-table \\\n",
    "           --i-table table.qza \\\n",
    "                     --i-taxonomy silva-taxonomy.qza \\\n",
    "                                  --p-exclude mitochondria,chloroplast \\\n",
    "                                                           --o-filtered-table silva-filtered-table.qza\n",
    "\n",
    "#visualize results from feature table\n",
    "qiime feature-table summarize \\\n",
    "                    --i-table silva-filtered-table.qza \\\n",
    "                              --o-visualization silva-filtered-table.qzv \\\n",
    "                                                --m-sample-metadata-file BZ15_metadata.tsv\n",
    "\n",
    "\n",
    "\n",
    "#Taxonomy Bar plots\n",
    "#visualize filtered taxa bar plots\n",
    "qiime taxa barplot \\\n",
    "           --i-table silva-filtered-table.qza \\\n",
    "                     --i-taxonomy silva-taxonomy.qza \\\n",
    "                                  --m-metadata-file BZ15_metadata.tsv \\\n",
    "                                                    --o-visualization silva-filtered-taxa-bar-plots.qzv\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "#####Filtering step specific to my experiment (threshold set at 70 reads)\n",
    "\n",
    "qiime feature-table filter-features \\\n",
    "                    --i-table silva-filtered-table.qza \\\n",
    "                              --p-min-frequency 70 \\\n",
    "                                                --o-filtered-table silva-70-filtered-table.qza\n",
    "\n",
    "#visualize\n",
    "qiime feature-table summarize \\\n",
    "                    --i-table silva-70-filtered-table.qza \\\n",
    "                              --o-visualization silva-70-filtered-table.qzv \\\n",
    "                                                --m-sample-metadata-file BZ15_metadata.tsv\n",
    "\n",
    "#Now lowest depth is AD069 @ 56,983 followed by AD067 @ 66,676\n",
    "\n",
    "Taxonomy Bar plots\n",
    "#visualize filtered taxa bar plots\n",
    "qiime taxa barplot \\\n",
    "           --i-table silva-70-filtered-table.qza \\\n",
    "                     --i-taxonomy silva-taxonomy.qza \\\n",
    "                                  --m-metadata-file BZ15_metadata.tsv \\\n",
    "                                                    --o-visualization silva-70-filtered-taxa-bar-plots.qzv\n",
    "\n",
    "\n",
    "#Results indicate that there are still ~50 genera that we only see in CD sampling! this includes streptomyces\n",
    "#continue with analysis, first rarefaction\n",
    "\n",
    "\n",
    "ALPHA RAREFACTION\n",
    "#Alpha rarefaction makes table of diversity at different depths for your curve\n",
    "#for first one, lets look at the unlimited diversity by doing max depth of 266,648\n",
    "qiime diversity alpha-rarefaction \\\n",
    "                --i-table silva-70-filtered-table.qza \\\n",
    "                          --i-phylogeny rooted-tree.qza \\\n",
    "                                        --p-max-depth 266648 \\\n",
    "                                                      --m-metadata-file BZ15_metadata.tsv \\\n",
    "                                                                        --o-visualization alpha-rarefaction.qzv\n",
    "\n",
    "#not enough steps, redo with >10 (default)\n",
    "qiime diversity alpha-rarefaction \\\n",
    "                --i-table silva-70-filtered-table.qza \\\n",
    "                          --i-phylogeny rooted-tree.qza \\\n",
    "                                        --p-steps 20 \\\n",
    "                                                  --p-max-depth 266648 \\\n",
    "                                                                --m-metadata-file BZ15_metadata.tsv \\\n",
    "                                                                                  --o-visualization alpha-rarefaction-20steps.qzv\n",
    "\n",
    "\n",
    "\n",
    "#### Lets Rarefy to 56,983 to include all samples and go from there...\n",
    "\n",
    "qiime diversity alpha-rarefaction \\\n",
    "                --i-table silva-70-filtered-table.qza \\\n",
    "                          --i-phylogeny rooted-tree.qza \\\n",
    "                                        --p-max-depth 56983 \\\n",
    "                                                      --m-metadata-file BZ15_metadata.tsv \\\n",
    "                                                                        --o-visualization alpha-rarefaction-56k.qzv\n",
    "\n",
    "#I think it looks okay in terms of being saturated\n",
    "\n",
    "\n",
    "#now do diversity core metrices\n",
    "qiime diversity core-metrics-phylogenetic \\\n",
    "                --i-phylogeny rooted-tree.qza \\\n",
    "                              --i-table silva-70-filtered-table.qza \\\n",
    "                                        --p-sampling-depth 56983 \\\n",
    "                                                           --m-metadata-file BZ15_metadata.tsv \\\n",
    "                                                                             --output-dir 56k/core-metrics-results-56k\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "qiime tools import \\\n",
    "    --type 'SampleData[PairedEndSequencesWithQuality]' \\\n",
    "           --input-path /Users/dimitriosgiakoumas/Downloads/Zymo_rawdata_qiime2_Analysis_DG/sample-metadata.tsv \\\n",
    "                        --input-format CasavaOneEightSingleLanePerSampleDirFmt \\\n",
    "                                       --output-path demux-paired-end.qza"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "Pipeline modified from Qiime2 Moving Pictures Tutorial\n",
    "Alyssa Demko\n",
    "\n",
    "Project: Belize 2015 sediment culture-independent vs dependent comparison\n",
    "06.13.2020\n",
    "\n",
    "Only looking at 2020 controls with Vibrio spike in\n",
    "\n",
    "\n",
    "#Activate Qiime2\n",
    "###to activate: source activate qiime2-2020.2\n",
    "###to exit environment: conda deactivate\n",
    "\n",
    "#import files, note need absolute file path included in manifest doc\n",
    "##files need to be unzipped\n",
    "\n",
    "#Make sure you are in the proper directory (folder) and your files are there as well\n",
    "pwd: /Users/alyssademko/Desktop/BZ15_2019/2020_ReAnalysis\n",
    "#files:\n",
    "pe-33-manifest.csv #file with path to raw reads for each sample\n",
    "BZ15_metadata.tsv # include all metadata here for easier visualization/stats analysis later on\n",
    "raw_reads/ #make sure they are unzipped!\n",
    "\n",
    "#Import files\n",
    "qiime tools import \\\n",
    "    --type 'SampleData[PairedEndSequencesWithQuality]' \\\n",
    "           --input-path pe-33-manifest-bzcontrols2.csv \\\n",
    "                        --output-path demux.qza \\\n",
    "                                      --input-format PairedEndFastqManifestPhred33\n",
    "\n",
    "qiime demux summarize \\\n",
    "            --i-data demux.qza \\\n",
    "                     --o-visualization DEMUX_SEQ_SUMMARY_VIZ.qzv\n",
    "\n",
    "\n",
    "#DADA2 denoising (takes a few hours- run overnight)\n",
    "#this trims off the primers and trims the sequences (Reverse were more crappy which is why they were trimmed shorter), also removes chimeras (default is consensus method)\n",
    "#decide metrics after looking at demux.qzv above\n",
    "\n",
    "qiime dada2 denoise-paired \\\n",
    "            --i-demultiplexed-seqs demux.qza \\\n",
    "                                   --p-trunc-len-f 250 \\\n",
    "                                                   --p-trunc-len-r 155 \\\n",
    "                                                                   --p-trim-left-f 19 \\\n",
    "                                                                                   --p-trim-left-r 20 \\\n",
    "                                                                                                   --o-table table.qza \\\n",
    "                                                                                                             --o-representative-sequences rep-seqs.qza \\\n",
    "                                                                                                                                          --o-denoising-stats denoising-stats.qza\n",
    "\n",
    "\n",
    "\n",
    "#visualize results from feature table\n",
    "qiime feature-table summarize \\\n",
    "                    --i-table table.qza \\\n",
    "                              --o-visualization table.qzv \\\n",
    "                                                --m-sample-metadata-file BZ15_metadata.tsv\n",
    "\n",
    "#visualize results summaries\n",
    "qiime feature-table tabulate-seqs \\\n",
    "                    --i-data rep-seqs.qza \\\n",
    "                             --o-visualization rep-seqs.qzv\n",
    "\n",
    "#visualize denoising stats\n",
    "qiime metadata tabulate \\\n",
    "               --m-input-file denoising-stats.qza \\\n",
    "                              --o-visualization denoising-stats.qzv\n",
    "\n",
    "#everything looks good from files above\n",
    "#lowest depth after above step is AD069 @ 68,943 followed by AD072 @ 87,757\n",
    "#will subsequently still need to remove chloroplast, singletons, etc\n",
    "\n",
    "\n",
    "#Generate a tree for phylogenetic diversity analyses, start @ 1pm, finished 1:56pm\n",
    "qiime phylogeny align-to-tree-mafft-fasttree \\\n",
    "                --i-sequences rep-seqs.qza \\\n",
    "                              --o-alignment aligned-rep-seqs.qza \\\n",
    "                                            --o-masked-alignment masked-aligned-rep-seqs.qza \\\n",
    "                                                                 --o-tree unrooted-tree.qza \\\n",
    "                                                                          --o-rooted-tree rooted-tree.qza\n",
    "\n",
    "\n",
    "#Taxonomy\n",
    "#make sure to use the classifier trained on the right scikit for this to work\n",
    "#can take a few hours\n",
    "qiime feature-classifier classify-sklearn \\\n",
    "                         --i-classifier silva-132-99-515-806-nb-classifier.qza \\\n",
    "                                        --i-reads rep-seqs.qza \\\n",
    "                                                  --o-classification silva-taxonomy.qza\n",
    "\n",
    "\n",
    "qiime metadata tabulate \\\n",
    "               --m-input-file silva-taxonomy.qza \\\n",
    "                              --o-visualization silva-taxonomy.qzv\n",
    "\n",
    "\n",
    "# need to get rid of chloroplasts! might as well do mitochondria too although not usually an issue\n",
    "qiime taxa filter-table \\\n",
    "           --i-table table.qza \\\n",
    "                     --i-taxonomy silva-taxonomy.qza \\\n",
    "                                  --p-exclude mitochondria,chloroplast \\\n",
    "                                                           --o-filtered-table silva-filtered-table.qza\n",
    "\n",
    "#visualize results from feature table\n",
    "qiime feature-table summarize \\\n",
    "                    --i-table silva-filtered-table.qza \\\n",
    "                              --o-visualization silva-filtered-table.qzv \\\n",
    "                                                --m-sample-metadata-file BZ15_metadata.tsv\n",
    "\n",
    "\n",
    "\n",
    "#Taxonomy Bar plots\n",
    "#visualize filtered taxa bar plots\n",
    "qiime taxa barplot \\\n",
    "           --i-table silva-filtered-table.qza \\\n",
    "                     --i-taxonomy silva-taxonomy.qza \\\n",
    "                                  --m-metadata-file BZ15_metadata.tsv \\\n",
    "                                                    --o-visualization silva-filtered-taxa-bar-plots.qzv\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "#####Filtering step specific to my experiment (threshold set at 70 reads)\n",
    "\n",
    "qiime feature-table filter-features \\\n",
    "                    --i-table silva-filtered-table.qza \\\n",
    "                              --p-min-frequency 70 \\\n",
    "                                                --o-filtered-table silva-70-filtered-table.qza\n",
    "\n",
    "#visualize\n",
    "qiime feature-table summarize \\\n",
    "                    --i-table silva-70-filtered-table.qza \\\n",
    "                              --o-visualization silva-70-filtered-table.qzv \\\n",
    "                                                --m-sample-metadata-file BZ15_metadata.tsv\n",
    "\n",
    "#Now lowest depth is AD069 @ 56,983 followed by AD067 @ 66,676\n",
    "\n",
    "Taxonomy Bar plots\n",
    "#visualize filtered taxa bar plots\n",
    "qiime taxa barplot \\\n",
    "           --i-table silva-70-filtered-table.qza \\\n",
    "                     --i-taxonomy silva-taxonomy.qza \\\n",
    "                                  --m-metadata-file BZ15_metadata.tsv \\\n",
    "                                                    --o-visualization silva-70-filtered-taxa-bar-plots.qzv\n",
    "\n",
    "\n",
    "#Results indicate that there are still ~50 genera that we only see in CD sampling! this includes streptomyces\n",
    "#continue with analysis, first rarefaction\n",
    "\n",
    "\n",
    "ALPHA RAREFACTION\n",
    "#Alpha rarefaction makes table of diversity at different depths for your curve\n",
    "#for first one, lets look at the unlimited diversity by doing max depth of 266,648\n",
    "qiime diversity alpha-rarefaction \\\n",
    "                --i-table silva-70-filtered-table.qza \\\n",
    "                          --i-phylogeny rooted-tree.qza \\\n",
    "                                        --p-max-depth 266648 \\\n",
    "                                                      --m-metadata-file BZ15_metadata.tsv \\\n",
    "                                                                        --o-visualization alpha-rarefaction.qzv\n",
    "\n",
    "#not enough steps, redo with >10 (default)\n",
    "qiime diversity alpha-rarefaction \\\n",
    "                --i-table silva-70-filtered-table.qza \\\n",
    "                          --i-phylogeny rooted-tree.qza \\\n",
    "                                        --p-steps 20 \\\n",
    "                                                  --p-max-depth 266648 \\\n",
    "                                                                --m-metadata-file BZ15_metadata.tsv \\\n",
    "                                                                                  --o-visualization alpha-rarefaction-20steps.qzv\n",
    "\n",
    "\n",
    "\n",
    "#### Lets Rarefy to 56,983 to include all samples and go from there...\n",
    "\n",
    "qiime diversity alpha-rarefaction \\\n",
    "                --i-table silva-70-filtered-table.qza \\\n",
    "                          --i-phylogeny rooted-tree.qza \\\n",
    "                                        --p-max-depth 56983 \\\n",
    "                                                      --m-metadata-file BZ15_metadata.tsv \\\n",
    "                                                                        --o-visualization alpha-rarefaction-56k.qzv\n",
    "\n",
    "#I think it looks okay in terms of being saturated\n",
    "\n",
    "\n",
    "#now do diversity core metrices\n",
    "qiime diversity core-metrics-phylogenetic \\\n",
    "                --i-phylogeny rooted-tree.qza \\\n",
    "                              --i-table silva-70-filtered-table.qza \\\n",
    "                                        --p-sampling-depth 56983 \\\n",
    "                                                           --m-metadata-file BZ15_metadata.tsv \\\n",
    "                                                                             --output-dir 56k/core-metrics-results-56k\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "/Users/dimitriosgiakoumas/Downloads/QIIME2_Workshop_2022.2/Section_1_QIIME2/manifest.tsv\n",
    "qiime tools import --type 'SampleData[PairedEndSequencesWithQuality]' --input-path /Users/dimitriosgiakoumas/Downloads/QIIME2_Workshop_2022.2/Section_1_QIIME2/manifest.csv --output-path output.seqs --input-format 'PairedEndFastqManifestPhred33V2'\n",
    "sample-id forward-absolute-filepath reverse-absolute-filepath\n",
    "\n",
    "qiime tools import --type 'SampleData[PairedEndSequencesWithQuality]' --input-path /Users/dimitriosgiakoumas/Downloads/QIIME2_Workshop_2022.2/Section_1_QIIME2/manifest.csv --output-path raw_sequences.qza --input-format 'PairedEndFastqManifestPhred33V2'\n",
    "qiime tools import \\\n",
    "    --type MultiplexedPairedEndBarcodeInSequence \\\n",
    "           --input-path /Users/dimitriosgiakoumas/Downloads/QIIME2_Workshop_2022.2/Section_1_QIIME2/manifest.csv \\\n",
    "                        --output-path multiplexed-seqs.qza\n",
    "\n",
    "qiime tools import \\\n",
    "    --type MultiplexedPairedEndBarcodeInSequence \\\n",
    "           --input-path /Users/dimitriosgiakoumas/Downloads/QIIME2_Workshop_2022.2/Section_1_QIIME2 \\\n",
    "                                                                                --output-path multiplexed-seqs.qza\n",
    "\n",
    "qiime tools import \\\n",
    "    --type MultiplexedPairedEndBarcodeInSequence \\\n",
    "           --input-path /Users/dimitriosgiakoumas/Downloads/muxed-pe-barcode-in-seq \\\n",
    "                        --output-path multiplexed-seqs.qza\n",
    "\n",
    "qiime tools import \\\n",
    "    --type EMPPairedEndSequences \\\n",
    "           --input-path /Users/dimitriosgiakoumas/Downloads/muxed-pe-barcode-in-seq \\\n",
    "                        --output-path emp-paired-end-sequences.qza\n",
    "\n",
    "\n",
    "qiime demux summarize --i-data /Users/dimitriosgiakoumas/Downloads/muxed-pe-barcode-in-seq/emp-paired-end-sequences.qza --o-visualization output.file\n",
    "\n",
    "qiime demux emp-paired \\\n",
    "            --m-barcodes-file sample-metadata.tsv \\\n",
    "                              --m-barcodes-column barcode-sequence \\\n",
    "                                                  --p-rev-comp-mapping-barcodes \\\n",
    "                                                  --i-seqs /Users/dimitriosgiakoumas/Downloads/muxed-pe-barcode-in-seq/emp-paired-end-sequences.qza \\\n",
    "                                                           --o-per-sample-sequences demux-full.qza \\\n",
    "                                                                                    --o-error-correction-details demux-details.qza\n",
    "\n",
    "\n",
    "biom convert -i /Users/dimitriosgiakoumas/Downloads/Otu_tables 3/otu_table_filtered_L6.biom -o output.tsv --to-tsv\n",
    "\n",
    "\n",
    "python -m venv biom-env\n",
    "\n",
    "\n",
    "source biom-env/bin/activate\n",
    "\n",
    "\n",
    "pip install biom-format\n",
    "pip install \"numpy>=1.16.5,<1.23.0\"\n",
    "pip install scipy\n",
    "\n",
    "\n",
    "biom convert -i \"/Users/dimitriosgiakoumas/Downloads/Otu_tables 3/otu_table_filtered_L6.biom\" -o output.tsv --to-tsv\n",
    "\n",
    "python -m venv biom-env\n",
    "\n",
    "\n",
    "source biom-env/bin/activate\n",
    "biom convert -i \"/Users/dimitriosgiakoumas/Downloads/Otu_tables 3/otu_table_filtered_L5.biom\" -o outputl5.tsv --to-tsv\n",
    "/Users/dimitriosgiakoumas/Downloads/Otu_tables 3/otu_table_filtered_L5.biom\n",
    "\n",
    "python -m venv biom-env\n",
    "\n",
    "\n",
    "source biom-env/bin/activate\n",
    "biom convert -i \"/Users/dimitriosgiakoumas/Downloads/Otu_tables 3/otu_table_filtered_L4.biom\" -o outputl4.tsv --to-tsv\n",
    "\n",
    "\n",
    "python -m venv biom-env\n",
    "\n",
    "\n",
    "source biom-env/bin/activate\n",
    "biom convert -i \"/Users/dimitriosgiakoumas/Downloads/Otu_tables 3/otu_table_filtered_L3.biom\" -o outputl3.tsv --to-tsv\n",
    "\n",
    "python -m venv biom-env\n",
    "\n",
    "\n",
    "source biom-env/bin/activate\n",
    "biom convert -i \"/Users/dimitriosgiakoumas/Downloads/Otu_tables 3/otu_table_filtered_L2.biom\" -o outputl2.tsv --to-tsv\n",
    "python -m venv biom-env\n",
    "\n",
    "\n",
    "source biom-env/bin/activate\n",
    "biom convert -i \"/Users/dimitriosgiakoumas/Desktop/GutMicrobiome_16SV4/otu_table_filtered_even112500.biom\" -o outputeven112500.tsv --to-tsv\n",
    "python -m venv biom-env\n",
    "\n",
    "\n",
    "source biom-env/bin/activate\n",
    "biom convert -i \"/Users/dimitriosgiakoumas/Downloads/Otu_tables 3/otu_table_filtered.biom\" -o outputfiltered.tsv --to-tsv\n",
    "\n",
    "/Users/dimitriosgiakoumas/Downloads/otu_bacteriaonly.biom\n",
    "\n",
    "source biom-env/bin/activate\n",
    "biom convert -i \"/Users/dimitriosgiakoumas/Downloads/otu_bacteriaonly.biom\" -o outputfiltered.tsv --to-tsv\n",
    "\n",
    "source biom-env/bin/activate\n",
    "biom convert -i \"/Users/dimitriosgiakoumas/Desktop/zr9410.16S_230202.zymo/00...AllSamples.Bac16Sv34/DADA2_ASV_Distribution/ASV_Table.biom\" -o outputeven112500.tsv --to-tsv\n",
    "python -m venv biom-env\n",
    "/Users/dimitriosgiakoumas/Desktop/zr9410.16S_230202.zymo/00...AllSamples.Bac16Sv34/DADA2_ASV_Distribution/ASV_Table.biom"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "qiime tools import \\\n",
    "    --type EMPPairedEndSequences \\\n",
    "           --input-path /Users/dimitriosgiakoumas/Downloads/New_Folder_With_Items \\\n",
    "                        --output-path emp-paired-end-sequences.qza\n",
    "\n",
    "/Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/zr9410_1V3V4_R1.fastq\n",
    "/Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/zr9410_1V3V4_R2.fastq\n",
    "/Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/zr9410_2V3V4_R1.fastq\n",
    "/Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/zr9410_2V3V4_R2.fastq\n",
    "\n",
    "qiime tools import \\\n",
    "--type 'SampleData[PairedEndSequencesWithQuality]' \\\n",
    "--input-path /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/Manafest.csv/sample-metadata.tsv \\\n",
    "--output-path paired-end-demux.qza \\\n",
    "\n",
    "--input-format PairedEndFastqManifestPhred33V2\n",
    "qiime demux summarize \\\n",
    "--i-data /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/paired-end-demux.qza \\\n",
    "--o-visualization DEMUX_SEQ_SUMMARY_VIZ.qzv\n",
    "\n",
    "\n",
    "qiime dada2 denoise-paired \\\n",
    "--i-demultiplexed-seqs /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/paired-end-demux.qza \\\n",
    "--p-trunc-len-f 321 \\\n",
    "--p-trunc-len-r 281 \\\n",
    "--p-trim-left-f 0 \\\n",
    "--p-trim-left-r 1 \\\n",
    "--o-table table.qza \\\n",
    "--o-representative-sequences rep-seqs.qza \\\n",
    "--o-denoising-stats denoising-stats.qza\n",
    "\n",
    "\n",
    "#visualize results from feature table\n",
    "qiime feature-table summarize \\\n",
    "--i-table /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/table.qza \\\n",
    "--o-visualization table.qzv \\\n",
    "--m-sample-metadata-file BZ15_metadata.tsv\n",
    "\n",
    "#visualize results summaries\n",
    "qiime feature-table tabulate-seqs \\\n",
    "--i-data /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/rep-seqs.qza \\\n",
    "--o-visualization rep-seqs.qzv\n",
    "\n",
    "#visualize denoising stats\n",
    "qiime metadata tabulate \\\n",
    "--m-input-file /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/denoising-stats.qza \\\n",
    "--o-visualization denoising-stats.qzv\n",
    "\n",
    "#Generate a tree for phylogenetic diversity analyses, start @ 1pm, finished 1:56pm\n",
    "qiime phylogeny align-to-tree-mafft-fasttree \\\n",
    "--i-sequences /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/rep-seqs.qza \\\n",
    "--o-alignment aligned-rep-seqs.qza \\\n",
    "--o-masked-alignment masked-aligned-rep-seqs.qza \\\n",
    "--o-tree unrooted-tree.qza \\\n",
    "--o-rooted-tree rooted-tree.qza\n",
    "\n",
    "#Taxonomy\n",
    "#make sure to use the classifier trained on the right scikit for this to work\n",
    "#can take a few hours\n",
    "qiime feature-classifier classify-sklearn \\\n",
    "--i-classifier /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/silva-138-99-515-806-nb-classifier.qza \\\n",
    "--i-reads /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/rep-seqs.qza \\\n",
    "--o-classification silva-taxonomy.qza\n",
    "\n",
    "\n",
    "qiime metadata tabulate \\\n",
    "--m-input-file /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/silva-taxonomy.qza \\\n",
    "--o-visualization silva-taxonomy.qzv\n",
    "\n",
    "\n",
    "# need to get rid of chloroplasts! might as well do mitochondria too although not usually an issue\n",
    "qiime taxa filter-table \\\n",
    "--i-table /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/table.qza \\\n",
    "--i-taxonomy /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/silva-taxonomy.qza \\\n",
    "--p-exclude mitochondria,chloroplast \\\n",
    "--o-filtered-table silva-filtered-table.qza\n",
    "\n",
    "#visualize results from feature table\n",
    "qiime feature-table summarize \\\n",
    "--i-table /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/silva-filtered-table.qza \\\n",
    "--o-visualization silva-filtered-table.qzv \\\n",
    "--m-sample-metadata-file /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/Manafest.csv/sample-metadata_zr9410_1V3V4.tsv\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "#Taxonomy Bar plots\n",
    "#visualize filtered taxa bar plots\n",
    "qiime taxa barplot \\\n",
    "--i-table /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/silva-filtered-table.qza \\\n",
    "--i-taxonomy /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/silva-taxonomy.qza \\\n",
    "--m-metadata-file /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/Manafest.csv/sample-metadata_zr9410_1V3V4.tsv \\\n",
    "--o-visualization silva-filtered-taxa-bar-plots.qzv\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "#####Filtering step specific to my experiment (threshold set at 70 reads)\n",
    "\n",
    "qiime feature-table filter-features \\\n",
    "--i-table silva-filtered-table.qza \\\n",
    "--p-min-frequency 70 \\\n",
    "--o-filtered-table silva-70-filtered-table.qza\n",
    "\n",
    "#visualize\n",
    "qiime feature-table summarize \\\n",
    "--i-table silva-70-filtered-table.qza \\\n",
    "--o-visualization silva-70-filtered-table.qzv \\\n",
    "--m-sample-metadata-file BZ15_metadata.tsv\n",
    "\n",
    "#Now lowest depth is AD069 @ 56,983 followed by AD067 @ 66,676\n",
    "\n",
    "Taxonomy Bar plots\n",
    "#visualize filtered taxa bar plots\n",
    "qiime taxa barplot \\\n",
    "--i-table silva-70-filtered-table.qza \\\n",
    "--i-taxonomy silva-taxonomy.qza \\\n",
    "--m-metadata-file BZ15_metadata.tsv \\\n",
    "--o-visualization silva-70-filtered-taxa-bar-plots.qzv\n",
    "\n",
    "\n",
    "#Results indicate that there are still ~50 genera that we only see in CD sampling! this includes streptomyces\n",
    "#continue with analysis, first rarefaction\n",
    "\n",
    "\n",
    "ALPHA RAREFACTION\n",
    "#Alpha rarefaction makes table of diversity at different depths for your curve\n",
    "#for first one, lets look at the unlimited diversity by doing max depth of 266,648\n",
    "qiime diversity alpha-rarefaction \\\n",
    "--i-table /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/silva-filtered-table.qza \\\n",
    "--i-phylogeny /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/rooted-tree.qza \\\n",
    "--p-max-depth 20000 \\\n",
    "--m-metadata-file /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/Manafest.csv/sample-metadata_zr9410_1V3V4.tsv \\\n",
    "--o-visualization alpha-rarefaction.qzv\n",
    "\n",
    "#not enough steps, redo with >10 (default)\n",
    "qiime diversity alpha-rarefaction \\\n",
    "--i-table silva-70-filtered-table.qza \\\n",
    "--i-phylogeny rooted-tree.qza \\\n",
    "--p-steps 20 \\\n",
    "--p-max-depth 266648 \\\n",
    "--m-metadata-file BZ15_metadata.tsv \\\n",
    "--o-visualization alpha-rarefaction-20steps.qzv\n",
    "\n",
    "\n",
    "\n",
    "#### Lets Rarefy to 56,983 to include all samples and go from there...\n",
    "\n",
    "qiime diversity alpha-rarefaction \\\n",
    "--i-table silva-70-filtered-table.qza \\\n",
    "--i-phylogeny rooted-tree.qza \\\n",
    "--p-max-depth 56983 \\\n",
    "--m-metadata-file BZ15_metadata.tsv \\\n",
    "--o-visualization alpha-rarefaction-56k.qzv\n",
    "\n",
    "#I think it looks okay in terms of being saturated\n",
    "\n",
    "\n",
    "#now do diversity core metrices\n",
    "qiime diversity core-metrics-phylogenetic \\\n",
    "--i-phylogeny /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/rooted-tree.qza \\\n",
    "--i-table /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/silva-filtered-table.qza \\\n",
    "--p-sampling-depth 20000 \\\n",
    "--m-metadata-file /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/Manafest.csv/sample-metadata_zr9410_1V3V4.tsv \\\n",
    "--output-dir 56k/core-metrics-results-56k\n",
    "\n",
    "\n",
    "\n",
    "qiime phylogeny fasttree --i-alignment {input.filtered_aligned_seqs} --o-tree {output.tree} --verbose --p-n-threads {threads}\n",
    "\n",
    "qiime phylogeny midpoint-root --i-tree {input.tree} --o-rooted-tree {output.tree}"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "qiime tools import \\\n",
    "    --type 'SampleData[PairedEndSequencesWithQuality]' \\\n",
    "           --input-path /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/muxed-pe-barcode-in-seq/Undetermined_S0_L001_I1_001.fastq.gz \\\n",
    "           --output-path demux-paired-end.qza \\\n",
    "                         --input-format PairedEndFastqManifestPhred33\n",
    "\n",
    "qiime tools import \\\n",
    "    --type 'SampleData[PairedEndSequencesWithQuality]' \\\n",
    "           --input-path /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/muxed-pe-barcode-in-seq/Undetermined_S0_L001_R2_001.fastq.gz \\\n",
    "           --output-path demux-paired-end.qza \\\n",
    "                         --input-format PairedEndFastqManifestPhred33\n",
    "\n",
    "qiime tools import \\\n",
    "--type 'SampleData[PairedEndSequencesWithQuality]' \\\n",
    "--input-path /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/muxed-pe-barcode-in-seq/Undetermined_S0_L001_R1_001.fastq.gz \\\n",
    "--output-path demux-paired-end.qza \\\n",
    "--input-format PairedEndFastqManifestPhred33\n",
    "\n",
    "/Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/muxed-pe-barcode-in-seq/Undetermined_S0_L001_I1_001.fastq.gz\n",
    "/Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/muxed-pe-barcode-in-seq/Undetermined_S0_L001_R1_001.fastq.gz\n",
    "/Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/muxed-pe-barcode-in-seq/Undetermined_S0_L001_R2_001.fastq.gz\n",
    "qiime tools import \\\n",
    "    --type EMPPairedEndSequences \\\n",
    "           --input-path /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/muxed-pe-barcode-in-seq \\\n",
    "                        --output-path emp-paired-end-sequences.qza\n",
    "\n",
    "qiime tools import \\\n",
    "    --type EMPPairedEndSequences \\\n",
    "           --input-path /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/emp-paired-end-sequences \\\n",
    "                        --output-path emp-paired-end-sequences.qza\n",
    "/Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/muxed-pe-barcode-in-seq\n",
    "\n",
    "-rw-r--r-- 1 username groupname 12345 Jan  1 00:00 /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/emp-paired-end-sequences \\\n",
    "\n",
    "\n",
    "qiime demux emp-paired \\\n",
    "--m-barcodes-file /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/emp-paired-end-sequences/Archbold_Barcode_Sample_Metadata.tsv \\\n",
    "--m-barcodes-column barcode-sequence \\\n",
    "--p-rev-comp-mapping-barcodes \\\n",
    "--i-seqs /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/emp-paired-end-sequences/emp-paired-end-sequences.qza \\\n",
    "--o-per-sample-sequences demux-full.qza \\\n",
    "--o-error-correction-details demux-details.qza\n",
    "\n",
    "qiime demux summarize \\\n",
    "--i-data /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/demux-details.qza \\\n",
    "--o-visualization demux-details.qzv\n",
    "\n",
    "denoise step archbold when 25th percetile gets below 30 i will truncate it A common approach is to truncate the sequences at a point where the quality scores start to fall below a certain threshold, such as 30, which corresponds to a 1 in 1000 chance of an error.\n",
    "0 forwawrd left\n",
    "251 forward right\n",
    "\n",
    "reverse reads were pretty bad so i am leaving them out and only denoising the forward reads\n",
    "GPT4 gernerated code:\n",
    "qiime dada2 denoise-single \\\n",
    "--i-demultiplexed-seqs /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/demux-full.qza \\\n",
    "--p-trim-left 0 \\\n",
    "--p-trunc-len 251 \\\n",
    "--o-representative-sequences rep-seqs-dada2.qza \\\n",
    "--o-table table-dada2.qza \\\n",
    "--o-denoising-stats stats-dada2.qza\n",
    "\n",
    "qiime dada2 denoise-paired \\\n",
    "--i-demultiplexed-seqs /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/demux-full.qza \\\n",
    "--p-trim-left-f 0 \\\n",
    "--p-trim-left-r 0 \\\n",
    "--p-trunc-len-f 251 \\\n",
    "--p-trunc-len-r 222 \\\n",
    "--o-representative-sequences rep-seqs-dada2.2.qza \\\n",
    "--o-table table-dada2.2.qza \\\n",
    "--o-denoising-stats stats-dada2.2.qza\n",
    "\n",
    "i took pretrained nieve bayes classifiers from qiime2 website\n",
    "Michael S Robeson II, Devon R O’Rourke, Benjamin D Kaehler, Michal Ziemski, Matthew R Dillon, Jeffrey T Foster, Nicholas A Bokulich. RESCRIPt: Reproducible sequence taxonomy reference database management for the masses. bioRxiv 2020.10.05.326504; doi: https://doi.org/10.1101/2020.10.05.326504\n",
    "\n",
    "Bokulich, N.A., Kaehler, B.D., Rideout, J.R. et al. Optimizing taxonomic classification of marker-gene amplicon sequences with QIIME 2’s q2-feature-classifier plugin. Microbiome 6, 90 (2018). https://doi.org/10.1186/s40168-018-0470-z\n",
    "\n",
    "qiime feature-classifier classify-sklearn \\\n",
    "--i-classifier /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/gg-13-8-99-515-806-nb-classifier.qza \\\n",
    "--i-reads /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/rep-seqs-dada2.qza \\\n",
    "--o-classification taxonomy.qza\n",
    "\n",
    "qiime feature-classifier classify-sklearn \\\n",
    "--i-classifier /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/silva-138-99-515-806-nb-classifier.qza \\\n",
    "--i-reads /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/rep-seqs-dada2.qza \\\n",
    "--o-classification taxonomy.qza\n",
    "\n",
    "\n",
    "qiime metadata tabulate \\\n",
    "--m-input-file /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/taxonomy.qza \\\n",
    "--o-visualization taxonomygg.qzv\n",
    "\n",
    "qiime metadata tabulate \\\n",
    "               --m-input-file /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/taxonomysilva.qza \\\n",
    "               --o-visualization taxonomysilva.qzv\n",
    "\n",
    "qiime taxa barplot \\\n",
    "--i-table /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/table-dada2.qza \\\n",
    "--i-taxonomy /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/taxonomysilva.qza \\\n",
    "--m-metadata-file /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/emp-paired-end-sequences/Archbold_Barcode_Sample_Metadata.tsv \\\n",
    "--o-visualization taxa-bar-plotssilva.qzv\n",
    "\n",
    "\n",
    "\n",
    "qiime vsearch dereplicate-sequences \\\n",
    "--i-sequences /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/rep-seqs-dada2.qza \\\n",
    "--o-clustering clustered_sequences.qza \\\n",
    "--p-perc-identity 0.97 \\\n",
    "--p-threads 4\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "source": [
    "qiime tools import \\\n",
    "    --type EMPPairedEndSequences \\\n",
    "           --input-path /Users/dimitriosgiakoumas/Downloads/New_Folder_With_Items \\\n",
    "           --output-path emp-paired-end-sequences.qza\n",
    "\n",
    "\n",
    "/Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/zr9410_1V3V4_R1.fastq\n",
    "/Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/zr9410_1V3V4_R2.fastq\n",
    "/Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/zr9410_2V3V4_R1.fastq\n",
    "/Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/zr9410_2V3V4_R2.fastq\n",
    "\n",
    "qiime tools import \\\n",
    "--type 'SampleData[PairedEndSequencesWithQuality]' \\\n",
    "--input-path /Users/dimitriosgiakoumas/Downloads/Zymo_rawdata_qiime2_Analysis_DG/sample-metadata.tsv \\\n",
    "--output-path paired-end-demux.qza \\\n",
    "\n",
    "qiime tools import \\\n",
    "    --type 'SampleData[PairedEndSequencesWithQuality]' \\\n",
    "           --input-path /path/to/your/reads \\\n",
    "           --output-path demux.qza \\\n",
    "                         --input-format CasavaOneEightSingleLanePerSampleDirFmt\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "(qiime2-2023.2) dimitriosgiakoumas@Dimitrioss-MacBook-Air Zymo_rawdata_qiime2_Analysis_DG % qiime tools import \\\n",
    "    --type 'SampleData[PairedEndSequencesWithQuality]' \\\n",
    "           --input-path /Users/dimitriosgiakoumas/Downloads/Zymo_rawdata_qiime2_Analysis_DG/sample-metadata.tsv \\\n",
    "           --output-path paired-end-demux.qza \\\n",
    "                         > import.log\n",
    "\n",
    "\n",
    "--input-format PairedEndFastqManifestPhred33V2\n",
    "qiime demux summarize \\\n",
    "            --i-data /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/paired-end-demux.qza \\\n",
    "                                                                       --o-visualization DEMUX_SEQ_SUMMARY_VIZ.qzv\n",
    "\n",
    "\n",
    "qiime dada2 denoise-paired \\\n",
    "            --i-demultiplexed-seqs /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/paired-end-demux.qza \\\n",
    "                                                                                     --p-trunc-len-f 321 \\\n",
    "                                                                                                     --p-trunc-len-r 281 \\\n",
    "                                                                                                                     --p-trim-left-f 0 \\\n",
    "                                                                                                                                     --p-trim-left-r 1 \\\n",
    "                                                                                                                                                     --o-table table.qza \\\n",
    "                                                                                                                                                               --o-representative-sequences rep-seqs.qza \\\n",
    "                                                                                                                                                                                            --o-denoising-stats denoising-stats.qza\n",
    "\n",
    "\n",
    "#visualize results from feature table\n",
    "qiime feature-table summarize \\\n",
    "                    --i-table /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/table.qza \\\n",
    "                                                                                --o-visualization table.qzv \\\n",
    "                                                                                                  --m-sample-metadata-file BZ15_metadata.tsv\n",
    "\n",
    "#visualize results summaries\n",
    "qiime feature-table tabulate-seqs \\\n",
    "                    --i-data /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/rep-seqs.qza \\\n",
    "                                                                               --o-visualization rep-seqs.qzv\n",
    "\n",
    "#visualize denoising stats\n",
    "qiime metadata tabulate \\\n",
    "               --m-input-file /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/denoising-stats.qza \\\n",
    "                                                                                --o-visualization denoising-stats.qzv\n",
    "\n",
    "#Generate a tree for phylogenetic diversity analyses, start @ 1pm, finished 1:56pm\n",
    "qiime phylogeny align-to-tree-mafft-fasttree \\\n",
    "                --i-sequences /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/rep-seqs.qza \\\n",
    "                                                                                --o-alignment aligned-rep-seqs.qza \\\n",
    "                                                                                              --o-masked-alignment masked-aligned-rep-seqs.qza \\\n",
    "                                                                                                                   --o-tree unrooted-tree.qza \\\n",
    "                                                                                                                            --o-rooted-tree rooted-tree.qza\n",
    "\n",
    "#Taxonomy\n",
    "#make sure to use the classifier trained on the right scikit for this to work\n",
    "#can take a few hours\n",
    "qiime feature-classifier classify-sklearn \\\n",
    "                         --i-classifier /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/silva-138-99-515-806-nb-classifier.qza \\\n",
    "                                                                                          --i-reads /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/rep-seqs.qza \\\n",
    "                                                                                                                                                      --o-classification silva-taxonomy.qza\n",
    "\n",
    "\n",
    "qiime metadata tabulate \\\n",
    "               --m-input-file /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/silva-taxonomy.qza \\\n",
    "                                                                                --o-visualization silva-taxonomy.qzv\n",
    "\n",
    "\n",
    "# need to get rid of chloroplasts! might as well do mitochondria too although not usually an issue\n",
    "qiime taxa filter-table \\\n",
    "           --i-table /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/table.qza \\\n",
    "                                                                       --i-taxonomy /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/silva-taxonomy.qza \\\n",
    "                                                                                                                                      --p-exclude mitochondria,chloroplast \\\n",
    "                                                                                                                                                               --o-filtered-table silva-filtered-table.qza\n",
    "\n",
    "#visualize results from feature table\n",
    "qiime feature-table summarize \\\n",
    "                    --i-table /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/silva-filtered-table.qza \\\n",
    "                                                                                --o-visualization silva-filtered-table.qzv \\\n",
    "                                                                                                  --m-sample-metadata-file /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/Manafest.csv/sample-metadata_zr9410_1V3V4.tsv\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "#Taxonomy Bar plots\n",
    "#visualize filtered taxa bar plots\n",
    "qiime taxa barplot \\\n",
    "           --i-table /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/silva-filtered-table.qza \\\n",
    "                                                                       --i-taxonomy /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/silva-taxonomy.qza \\\n",
    "                                                                                                                                      --m-metadata-file /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/Manafest.csv/sample-metadata_zr9410_1V3V4.tsv \\\n",
    "                                                                                                                                                                                                          --o-visualization silva-filtered-taxa-bar-plots.qzv\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "#####Filtering step specific to my experiment (threshold set at 70 reads)\n",
    "\n",
    "qiime feature-table filter-features \\\n",
    "                    --i-table silva-filtered-table.qza \\\n",
    "                              --p-min-frequency 70 \\\n",
    "                                                --o-filtered-table silva-70-filtered-table.qza\n",
    "\n",
    "#visualize\n",
    "qiime feature-table summarize \\\n",
    "                    --i-table silva-70-filtered-table.qza \\\n",
    "                              --o-visualization silva-70-filtered-table.qzv \\\n",
    "                                                --m-sample-metadata-file BZ15_metadata.tsv\n",
    "\n",
    "#Now lowest depth is AD069 @ 56,983 followed by AD067 @ 66,676\n",
    "\n",
    "Taxonomy Bar plots\n",
    "#visualize filtered taxa bar plots\n",
    "qiime taxa barplot \\\n",
    "           --i-table silva-70-filtered-table.qza \\\n",
    "                     --i-taxonomy silva-taxonomy.qza \\\n",
    "                                  --m-metadata-file BZ15_metadata.tsv \\\n",
    "                                                    --o-visualization silva-70-filtered-taxa-bar-plots.qzv\n",
    "\n",
    "\n",
    "#Results indicate that there are still ~50 genera that we only see in CD sampling! this includes streptomyces\n",
    "#continue with analysis, first rarefaction\n",
    "\n",
    "\n",
    "ALPHA RAREFACTION\n",
    "#Alpha rarefaction makes table of diversity at different depths for your curve\n",
    "#for first one, lets look at the unlimited diversity by doing max depth of 266,648\n",
    "qiime diversity alpha-rarefaction \\\n",
    "                --i-table /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/silva-filtered-table.qza \\\n",
    "                                                                            --i-phylogeny /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/rooted-tree.qza \\\n",
    "                                                                                                                                            --p-max-depth 20000 \\\n",
    "                                                                                                                                                          --m-metadata-file /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/Manafest.csv/sample-metadata_zr9410_1V3V4.tsv \\\n",
    "                                                                                                                                                                                                                              --o-visualization alpha-rarefaction.qzv\n",
    "\n",
    "#not enough steps, redo with >10 (default)\n",
    "qiime diversity alpha-rarefaction \\\n",
    "                --i-table silva-70-filtered-table.qza \\\n",
    "                          --i-phylogeny rooted-tree.qza \\\n",
    "                                        --p-steps 20 \\\n",
    "                                                  --p-max-depth 266648 \\\n",
    "                                                                --m-metadata-file BZ15_metadata.tsv \\\n",
    "                                                                                  --o-visualization alpha-rarefaction-20steps.qzv\n",
    "\n",
    "\n",
    "\n",
    "#### Lets Rarefy to 56,983 to include all samples and go from there...\n",
    "\n",
    "qiime diversity alpha-rarefaction \\\n",
    "                --i-table silva-70-filtered-table.qza \\\n",
    "                          --i-phylogeny rooted-tree.qza \\\n",
    "                                        --p-max-depth 56983 \\\n",
    "                                                      --m-metadata-file BZ15_metadata.tsv \\\n",
    "                                                                        --o-visualization alpha-rarefaction-56k.qzv\n",
    "\n",
    "#I think it looks okay in terms of being saturated\n",
    "\n",
    "\n",
    "#now do diversity core metrices\n",
    "qiime diversity core-metrics-phylogenetic \\\n",
    "--i-phylogeny /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/rooted-tree.qza \\               --i-table /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/silva-filtered-table.qza \\\n",
    "--p-sampling-depth 20000 \\\n",
    "--m-metadata-file /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/Manafest.csv/sample-metadata_zr9410_1V3V4.tsv \\\n",
    "--output-dir 56k/core-metrics-results-56k\n",
    "\n",
    "qiime phylogeny fasttree --i-alignment {input.filtered_aligned_seqs} --o-tree {output.tree} --verbose --p-n-threads {threads}\n",
    "\n",
    "qiime phylogeny midpoint-root --i-tree {input.tree} --o-rooted-tree {output.tree}"
   ],
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "source": [
    "qiime taxa filter-table \\\n",
    "           --i-table /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/table-dada2.qza \\\n",
    "                                                                       --i-taxonomy /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/taxonomy.2.qza \\\n",
    "                                                                                                                                      --p-exclude mitochondria,chloroplast \\\n",
    "                                                                                                                                                               --o-filtered-table silva-filtered-table.qza\n",
    "\n",
    "qiime feature-table filter-features \\\n",
    "--i-table /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/table-dada2.qza\n",
    "--o-filtered-table filtered-table.qza\n",
    "\n",
    "qiime taxa filter-table \\\n",
    "           --i-table table.qza \\\n",
    "                     --i-taxonomy taxonomy.qza \\\n",
    "                                  --p-exclude mitochondria,chloroplast \\\n",
    "                                                           --o-filtered-table table-no-mitochondria-no-chloroplast.qza\n",
    "\n",
    "naive baeys class\n",
    "    qiime tools import \\\n",
    "    --type 'FeatureData[Sequence]' \\\n",
    "           --input-path /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/gg_13_5_tax_with_hOTUs_99_reps.fasta \\\n",
    "                          --output-path 99_otus.qza\n",
    "\n",
    "qiime tools import \\\n",
    "    --type 'FeatureData[Taxonomy]' \\\n",
    "           --input-format HeaderlessTSVTaxonomyFormat \\\n",
    "                          --input-path /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/gg_13_5_tax_with_hOTUs_99_reps_nou.taxonomy \\\n",
    "                                         --output-path ref-taxonomy.qza\n",
    "\n",
    "qiime feature-classifier extract-reads \\\n",
    "                         --i-sequences /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/99_otus.qza \\\n",
    "                                         --p-f-primer GTGCCAGCMGCCGCGGTAA \\\n",
    "                                                      --p-r-primer GGACTACHVGGGTWTCTAAT \\\n",
    "                                                                   --p-trunc-len 251 \\\n",
    "                                                                                 --p-min-length 100 \\\n",
    "                                                                                                --p-max-length 400 \\\n",
    "                                                                                                               --o-reads ref-seqs.qza\n",
    "qiime feature-classifier fit-classifier-naive-bayes \\\n",
    "                         --i-reference-reads /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/ref-seqs.qza \\\n",
    "                                             --i-reference-taxonomy /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/ref-taxonomy.qza \\\n",
    "                                                                    --o-classifier classifier.qza\n",
    "qiime feature-classifier classify-sklearn \\\n",
    "                         --i-classifier /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/classifier.2.qza \\\n",
    "                                        --i-reads /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/rep-seqs-dada2.2.qza \\\n",
    "                                                  --o-classification taxonomy.2.qza\n",
    "qiime feature-classifier classify-sklearn \\\n",
    "                         --i-classifier /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/classifier.2.qza \\\n",
    "                         --i-reads /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/rep-seqs-dn-97.2.qza \\\n",
    "                                   --o-classification taxonomy.2.qza\n",
    "qiime metadata tabulate \\\n",
    "               --m-input-file /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/taxonomy.2.qza \\\n",
    "                              --o-visualization taxonomy.2.qzv\n",
    "qiime taxa barplot \\\n",
    "           --i-table /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/table-dn-97.2.qza \\\n",
    "           --i-taxonomy /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/taxonomy.2.qza \\\n",
    "           --m-metadata-file /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/emp-paired-end-sequences/Archbold_Barcode_Sample_Metadata.tsv \\\n",
    "           --o-visualization -taxa-bar-plots.2.qzv\n",
    "qiime tools export \\\n",
    "            --input-path /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/rep-seqs-dn-97.2.qza \\\n",
    "                                        --output-path exported-rep-seqs-dn-97.2\n",
    "\n",
    "qiime diversity beta \\\n",
    "                --i-table /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/table-dn-97.2.qza \\\n",
    "                          --p-metric braycurtis \\\n",
    "                                     --o-distance-matrix bray_curtis_distance_matrix.2.qza\n",
    "qiime diversity pcoa \\\n",
    "                --i-distance-matrix /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/bray_curtis_distance_matrix.2.qza \\\n",
    "                                    --o-pcoa bray_curtis_pcoa_results.2.qza\n",
    "qiime emperor plot \\\n",
    "              --i-pcoa /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/bray_curtis_pcoa_results.2.qza \\\n",
    "                       --m-metadata-file /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/emp-paired-end-sequences/Archbold_Barcode_Sample_Metadata.tsv \\\n",
    "                                         --o-visualization bray_curtis_emperor.qzv\n",
    "\n",
    "qiime diversity beta-phylogenetic \\\n",
    "                --i-table /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/table-dn-97.2.qza \\\n",
    "                          --i-phylogeny /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/rooted-tree.2.qza \\\n",
    "                                        --p-metric weighted_unifrac \\\n",
    "                                                   --o-distance-matrix weighted_unifrac_distance_matrix.2.qza\n",
    "\n",
    "qiime diversity pcoa \\\n",
    "                --i-distance-matrix /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/weighted_unifrac_distance_matrix.2.qza \\\n",
    "                                    --o-pcoa weighted_unifrac_pcoa_results.2.qza\n",
    "\n",
    "qiime emperor plot \\\n",
    "              --i-pcoa /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/weighted_unifrac_pcoa_results.2.qza \\\n",
    "                       --m-metadata-file /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/emp-paired-end-sequences/Archbold_Barcode_Sample_Metadata.tsv \\\n",
    "                                         --o-visualization weighted_unifrac_emperor.2.qzv\n",
    "\n",
    "\n",
    "makeblastdb -in /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/exported-rep-seqs-dn-97.2/dna-sequences.fasta -dbtype nucl -out ref_seqs_db\n",
    "blastn -query exported-rep-seqs/dna-sequences.fasta -db ref_seqs_db -outfmt 6 -out blast_output.txt\n",
    "\n",
    "qiime metadata tabulate \\\n",
    "               --m-input-file /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/blast_output.tsv \\\n",
    "                              --o-visualization blast_output.qzv\n"
   ],
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "source": [
    "\n",
    "\n",
    "qiime tools import \\\n",
    "    --type 'FeatureData[Sequence]' \\\n",
    "           --input-path /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/2022.10.backbone.tax.qza \\\n",
    "           --output-path gg_97_otus.2022.qza\n",
    "\n",
    "qiime metadata tabulate \\\n",
    "               --m-input-file /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/taxonomy.2.qza \\\n",
    "                                                                                                                     --o-visualization taxonomy.2.qzv\n",
    "qiime feature-classifier classify-sklearn \\\n",
    "                         --i-classifier /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/2022.10.backbone.v4.nb.qza \\\n",
    "                         --i-reads /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/rep-seqs-dada2.qza \\\n",
    "                         --o-classification taxonomy2022.qza\n",
    "\n",
    "qiime taxa barplot \\\n",
    "--i-table /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/table-dn-97.2.qza \\\n",
    "--i-taxonomy /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/taxonomy.2.qza \\\n",
    "--m-metadata-file /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/emp-paired-end-sequences/Archbold_Barcode_Sample_Metadata.tsv \\\n",
    "--o-visualization taxa-bar-plots.2.qzv\n",
    "\n",
    "qiime taxa barplot \\\n",
    "--i-table /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/table-dada2.qza \\\n",
    "--i-taxonomy /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/taxonomy2022.qza \\\n",
    "--m-metadata-file /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/emp-paired-end-sequences/Archbold_Barcode_Sample_Metadata.tsv \\\n",
    "--o-visualization taxa-bar-plots-2022.qzv"
   ],
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "source": [
    "qiime vsearch dereplicate-sequences \\\n",
    "              --i-sequences /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/uchime-dn-out/nonchimeras.qza \\\n",
    "                            --o-dereplicated-table table.qza \\\n",
    "                                                   --o-dereplicated-sequences rep-seqs.qza\n",
    "qiime feature-table summarize \\\n",
    "                    --i-table /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/table-dada2.qza \\\n",
    "                              --o-visualization table.qzv \\\n",
    "                                                --m-sample-metadata-file /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/emp-paired-end-sequences/Archbold_Barcode_Sample_Metadata.tsv\n",
    "\n",
    "#visualize results summaries\n",
    "qiime feature-table tabulate-seqs \\\n",
    "                    --i-data rep-seqs.qza \\\n",
    "                             --o-visualization rep-seqs.qzv\n",
    "\n",
    "#visualize denoising stats\n",
    "qiime metadata tabulate \\\n",
    "               --m-input-file denoising-stats.qza \\\n",
    "                              --o-visualization denoising-stats.qzv\n",
    "\n",
    "#everything looks good from files above\n",
    "#lowest depth after above step is AD069 @ 68,943 followed by AD072 @ 87,757\n",
    "#will subsequently still need to remove chloroplast, singletons, etc\n",
    "qiime metadata tabulate \\\n",
    "               --m-input-file /Users/dimitriosgiakoumas/Downloads/Zymo_rawdata_qiime2_Analysis_DG/table120.qza \\\n",
    "                              --o-visualization taxa-table.qzv\n",
    "qiime taxa barplot \\\n",
    "           --i-table /Users/dimitriosgiakoumas/Downloads/Zymo_rawdata_qiime2_Analysis_DG/table120.qza \\\n",
    "                     --i-taxonomy taxonomy.qza \\\n",
    "                                  --m-metadata-file sample-metadata.tsv \\\n",
    "                                                    --o-visualization taxa-bar-plots.qzv\n",
    "\n",
    "qiime feature-classifier classify-sklearn \\\n",
    "                         --i-classifier /Users/dimitriosgiakoumas/Downloads/Zymo_rawdata_qiime2_Analysis_DG/classifier_GG16S_99otus_v3-4_338F-806R.qza \\\n",
    "                         --i-reads /Users/dimitriosgiakoumas/Downloads/Zymo_rawdata_qiime2_Analysis_DG/representative-sequences120.qza \\\n",
    "                                   --o-classification taxonomyGG120.qza\n",
    "qiime feature-classifier classify-sklearn \\\n",
    "                         --i-classifier /Users/dimitriosgiakoumas/Downloads/Zymo_rawdata_qiime2_Analysis_DG/classifier-dada2.qza \\\n",
    "                         --i-reads /Users/dimitriosgiakoumas/Downloads/Zymo_rawdata_qiime2_Analysis_DG/representative-sequences120.qza \\\n",
    "                                   --o-classification taxonomydada2.120.qza\n",
    "\n",
    "#Generate a tree for phylogenetic diversity analyses, start @ 1pm, finished 1:56pm\n",
    "qiime phylogeny align-to-tree-mafft-fasttree \\\n",
    "                --i-sequences /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/rep-seqs-dada2.2.qza \\\n",
    "                              --o-alignment aligned-rep-seqs.2.qza \\\n",
    "                                            --o-masked-alignment masked-aligned-rep-seqs.2.qza \\\n",
    "                                                                 --o-tree unrooted-tree.2.qza \\\n",
    "                                                                          --o-rooted-tree rooted-tree.2.qza\n",
    "\n",
    "\n",
    "#Taxonomy\n",
    "#make sure to use the classifier trained on the right scikit for this to work\n",
    "#can take a few hours\n",
    "qiime feature-classifier classify-sklearn \\\n",
    "                         --i-classifier /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/silva-138-99-515-806-nb-classifier.qza \\\n",
    "                                        --i-reads /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/rep-seqs-dada2.2.qza \\\n",
    "                                                  --o-classification silva-taxonomy.qza\n",
    "\n",
    "\n",
    "qiime metadata tabulate \\\n",
    "               --m-input-file /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/silva-taxonomy2.qza \\\n",
    "                              --o-visualization silva-taxonomy.qzv\n",
    "\n",
    "\n",
    "# need to get rid of chloroplasts! might as well do mitochondria too although not usually an issue\n",
    "qiime taxa filter-table \\\n",
    "           --i-table /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/table-dada2.2.qza \\\n",
    "                     --i-taxonomy /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/silva-taxonomy2.qza \\\n",
    "                                  --p-exclude mitochondria,chloroplast \\\n",
    "                                                           --o-filtered-table silva-filtered-table2.qza\n",
    "\n",
    "#visualize results from feature table\n",
    "qiime feature-table summarize \\\n",
    "                    --i-table /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/silva-filtered-table2.qza \\\n",
    "                              --o-visualization silva-filtered-table.qzv \\\n",
    "                                                --m-sample-metadata-file /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/emp-paired-end-sequences/Archbold_Barcode_Sample_Metadata.tsv\n",
    "\n",
    "\n",
    "\n",
    "#Taxonomy Bar plots\n",
    "#visualize filtered taxa bar plots\n",
    "qiime taxa barplot \\\n",
    "           --i-table /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/silva-filtered-table2.qza \\\n",
    "                     --i-taxonomy /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/silva-taxonomy2.qza \\\n",
    "                                  --m-metadata-file /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/emp-paired-end-sequences/Archbold_Barcode_Sample_Metadata.tsv \\\n",
    "                                                    --o-visualization silva-filtered-taxa-bar-plots.qzv\n",
    "\n",
    "\n",
    "java -Xmx1g -jar /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/rdp_classifier_2.13/dist/classifier.jar train \\\n",
    "                                         -s /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/gg_13_5_tax_with_hOTUs_99_reps_nou.taxonomy \\\n",
    "                                                                             --gene 16S \\\n",
    "                                                                             --seqs /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/gg_13_5_tax_with_hOTUs_99_reps.fasta \\\n",
    "                                                                                                         --output /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4\n",
    "\n",
    "\n",
    "#####Filtering step specific to my experiment (threshold set at 70 reads)\n",
    "\n",
    "qiime feature-table filter-features \\\n",
    "                    --i-table silva-filtered-table.qza \\\n",
    "                              --p-min-frequency 70 \\\n",
    "                                                --o-filtered-table silva-70-filtered-table.qza\n",
    "\n",
    "#visualize\n",
    "qiime feature-table summarize \\\n",
    "                    --i-table silva-70-filtered-table.qza \\\n",
    "                              --o-visualization silva-70-filtered-table.qzv \\\n",
    "                                                --m-sample-metadata-file BZ15_metadata.tsv\n",
    "\n",
    "#Now lowest depth is AD069 @ 56,983 followed by AD067 @ 66,676\n",
    "\n",
    "Taxonomy Bar plots\n",
    "#visualize filtered taxa bar plots\n",
    "qiime taxa barplot \\\n",
    "           --i-table silva-70-filtered-table.qza \\\n",
    "                     --i-taxonomy silva-taxonomy.qza \\\n",
    "                                  --m-metadata-file BZ15_metadata.tsv \\\n",
    "                                                    --o-visualization silva-70-filtered-taxa-bar-plots.qzv\n",
    "\n",
    "\n",
    "#Results indicate that there are still ~50 genera that we only see in CD sampling! this includes streptomyces\n",
    "#continue with analysis, first rarefaction\n",
    "\n",
    "\n",
    "ALPHA RAREFACTION\n",
    "#Alpha rarefaction makes table of diversity at different depths for your curve\n",
    "#for first one, lets look at the unlimited diversity by doing max depth of 266,648\n",
    "qiime diversity alpha-rarefaction \\\n",
    "                --i-table /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/table-dada2.qza \\\n",
    "                          --i-phylogeny /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/rooted-tree.qza \\\n",
    "                                        --p-max-depth 49666 \\\n",
    "                                                      --m-metadata-file /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/emp-paired-end-sequences/Archbold_Barcode_Sample_Metadata.tsv \\\n",
    "                                                                        --o-visualization alpha-rarefaction.qzv\n",
    "\n",
    "#not enough steps, redo with >10 (default)\n",
    "qiime diversity alpha-rarefaction \\\n",
    "                --i-table silva-70-filtered-table.qza \\\n",
    "                          --i-phylogeny rooted-tree.qza \\\n",
    "                                        --p-steps 20 \\\n",
    "                                                  --p-max-depth 266648 \\\n",
    "                                                                --m-metadata-file BZ15_metadata.tsv \\\n",
    "                                                                                  --o-visualization alpha-rarefaction-20steps.qzv\n",
    "\n",
    "\n",
    "\n",
    "#### Lets Rarefy to 56,983 to include all samples and go from there...\n",
    "\n",
    "qiime diversity alpha-rarefaction \\\n",
    "                --i-table silva-70-filtered-table.qza \\\n",
    "                          --i-phylogeny /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/rooted-tree.qza \\\n",
    "                                        --p-max-depth 56983 \\\n",
    "                                                      --m-metadata-file BZ15_metadata.tsv \\\n",
    "                                                                        --o-visualization alpha-rarefaction-56k.qzv\n",
    "\n",
    "#I think it looks okay in terms of being saturated\n",
    "\n",
    "\n",
    "#now do diversity core metrices\n",
    "qiime diversity core-metrics-phylogenetic \\\n",
    "                --i-phylogeny /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/rooted-tree.qza \\\n",
    "                              --i-table /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/table-dada2.qza \\\n",
    "                                        --p-sampling-depth 49666 \\\n",
    "                                                           --m-metadata-file /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/emp-paired-end-sequences/Archbold_Barcode_Sample_Metadata.tsv \\\n",
    "                                                                             --output-dir 56k/core-metrics-results-56k#%%\n",
    "\n",
    "qiime diversity core-metrics-phylogenetic \\\n",
    "                --i-phylogeny /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/rooted-tree.qza \\\n",
    "                                                                                --i-table /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/table-dada2.qza \\\n",
    "                                                                                                                                            --p-sampling-depth 49666 \\\n",
    "                                                                                                                                                               --m-metadata-file /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/Yuan_GT_mapfile_copy.tsv \\\n",
    "                                                                                                                                                                                                                                   --output-dir 56k/core-metrics-results-56k\n",
    "\n",
    "qiime diversity alpha-group-significance \\\n",
    "                --i-alpha-diversity /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/56k/core-metrics-results-56k/evenness_vector.qza \\\n",
    "                                                                                                                         --m-metadata-file /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/Manafest.csv/sample-metadata_zr9410_1V3V4.tsv \\\n",
    "                                                                                                                                                                                             --o-visualization /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/56k/core-metrics-results-56k/evenness_vector.qzv\n",
    "qiime diversity beta-group-significance \\\n",
    "                --i-distance-matrix /path/to/your/distance/matrix.qza \\\n",
    "                --m-metadata-file /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/Manafest.csv/sample-metadata_zr9410_1V3V4.tsv \\\n",
    "                                                                                    --o-visualization /path/to/your/visualization.qzv\n",
    "qiime emperor plot \\\n",
    "              --i-pcoa /path/to/your/pcoa/results.qza \\\n",
    "              --m-metadata-file /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/Manafest.csv/sample-metadata_zr9410_1V3V4.tsv \\\n",
    "                                                                                  --o-visualization /path/to/your/visualization.qzv"
   ],
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "\n",
    "\n",
    "\n",
    "\n",
    "qiime feature-table filter-features \\\n",
    "                    --i-table /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/table-dada2.qza \\\n",
    "                              --i-sequences /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/rep-seqs-dada2.qza \\\n",
    "                                            --p-where \"taxonomy NOT LIKE '%Chloroplast%' AND taxonomy NOT LIKE '%Mitochondria%'\" \\\n",
    "                                                      --o-filtered-table table-no-chimeras.qza \\\n",
    "                                                                         --o-filtered-sequences rep-seqs-no-chimeras.qza\n",
    "qiime vsearch uchime-denovo \\\n",
    "              --i-table /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/table-dada2.qza \\\n",
    "                        --i-sequences /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/uchime-dn-out/nonchimeras.qza \\\n",
    "                                      --output-dir uchime-dn-out\n",
    "\n",
    "qiime metadata tabulate --m-input-file /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/uchime-dn-out/stats.qza \\\n",
    "                                       --o-visualization uchime-dn-out/stats.qzv\n",
    "\n",
    "qiime tools view uchime-dn-out/stats.qzv"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "/Users/dimitriosgiakoumas/Downloads/QIIME2_Workshop_2022.2/Section_1_QIIME2/manifest.tsv\n",
    "qiime tools import --type 'SampleData[PairedEndSequencesWithQuality]' --input-path /Users/dimitriosgiakoumas/Downloads/QIIME2_Workshop_2022.2/Section_1_QIIME2/manifest.csv --output-path output.seqs --input-format 'PairedEndFastqManifestPhred33V2'\n",
    "sample-id forward-absolute-filepath reverse-absolute-filepath\n",
    "\n",
    "qiime tools import --type 'SampleData[PairedEndSequencesWithQuality]' --input-path /Users/dimitriosgiakoumas/Downloads/QIIME2_Workshop_2022.2/Section_1_QIIME2/manifest.csv --output-path raw_sequences.qza --input-format 'PairedEndFastqManifestPhred33V2'\n",
    "qiime tools import \\\n",
    "    --type MultiplexedPairedEndBarcodeInSequence \\\n",
    "           --input-path /Users/dimitriosgiakoumas/Downloads/QIIME2_Workshop_2022.2/Section_1_QIIME2/manifest.csv \\\n",
    "                        --output-path multiplexed-seqs.qza\n",
    "\n",
    "qiime tools import \\\n",
    "    --type MultiplexedPairedEndBarcodeInSequence \\\n",
    "           --input-path /Users/dimitriosgiakoumas/Downloads/QIIME2_Workshop_2022.2/Section_1_QIIME2 \\\n",
    "                                                                                --output-path multiplexed-seqs.qza\n",
    "\n",
    "qiime tools import \\\n",
    "    --type MultiplexedPairedEndBarcodeInSequence \\\n",
    "           --input-path /Users/dimitriosgiakoumas/Downloads/muxed-pe-barcode-in-seq \\\n",
    "                        --output-path multiplexed-seqs.qza\n",
    "\n",
    "qiime tools import \\\n",
    "    --type EMPPairedEndSequences \\\n",
    "           --input-path /Users/dimitriosgiakoumas/Downloads/muxed-pe-barcode-in-seq \\\n",
    "                        --output-path emp-paired-end-sequences.qza\n",
    "\n",
    "\n",
    "qiime demux summarize --i-data /Users/dimitriosgiakoumas/Downloads/muxed-pe-barcode-in-seq/emp-paired-end-sequences.qza --o-visualization output.file\n",
    "\n",
    "qiime demux emp-paired \\\n",
    "            --m-barcodes-file sample-metadata.tsv \\\n",
    "                              --m-barcodes-column barcode-sequence \\\n",
    "                                                  --p-rev-comp-mapping-barcodes \\\n",
    "                                                  --i-seqs /Users/dimitriosgiakoumas/Downloads/muxed-pe-barcode-in-seq/emp-paired-end-sequences.qza \\\n",
    "                                                           --o-per-sample-sequences demux-full.qza \\\n",
    "                                                                                    --o-error-correction-details demux-details.qza\n",
    "\n",
    "\n",
    "biom convert -i /Users/dimitriosgiakoumas/Downloads/Otu_tables 3/otu_table_filtered_L6.biom -o output.tsv --to-tsv\n",
    "\n",
    "\n",
    "python -m venv biom-env\n",
    "\n",
    "\n",
    "source biom-env/bin/activate\n",
    "\n",
    "\n",
    "pip install biom-format\n",
    "pip install \"numpy>=1.16.5,<1.23.0\"\n",
    "pip install scipy\n",
    "\n",
    "\n",
    "biom convert -i \"/Users/dimitriosgiakoumas/Downloads/Otu_tables 3/otu_table_filtered_L6.biom\" -o output.tsv --to-tsv\n",
    "\n",
    "python -m venv biom-env\n",
    "\n",
    "\n",
    "source biom-env/bin/activate\n",
    "biom convert -i \"/Users/dimitriosgiakoumas/Downloads/Otu_tables 3/otu_table_filtered_L5.biom\" -o outputl5.tsv --to-tsv\n",
    "/Users/dimitriosgiakoumas/Downloads/Otu_tables 3/otu_table_filtered_L5.biom\n",
    "\n",
    "python -m venv biom-env\n",
    "\n",
    "\n",
    "source biom-env/bin/activate\n",
    "biom convert -i \"/Users/dimitriosgiakoumas/Downloads/Otu_tables 3/otu_table_filtered_L4.biom\" -o outputl4.tsv --to-tsv\n",
    "\n",
    "\n",
    "python -m venv biom-env\n",
    "\n",
    "\n",
    "source biom-env/bin/activate\n",
    "biom convert -i \"/Users/dimitriosgiakoumas/Downloads/Otu_tables 3/otu_table_filtered_L3.biom\" -o outputl3.tsv --to-tsv\n",
    "\n",
    "python -m venv biom-env\n",
    "\n",
    "\n",
    "source biom-env/bin/activate\n",
    "biom convert -i \"/Users/dimitriosgiakoumas/Downloads/Otu_tables 3/otu_table_filtered_L2.biom\" -o outputl2.tsv --to-tsv\n",
    "python -m venv biom-env\n",
    "\n",
    "\n",
    "source biom-env/bin/activate\n",
    "biom convert -i \"/Users/dimitriosgiakoumas/Desktop/GutMicrobiome_16SV4/otu_table_filtered_even112500.biom\" -o outputeven112500.tsv --to-tsv\n",
    "python -m venv biom-env\n",
    "\n",
    "\n",
    "source biom-env/bin/activate\n",
    "biom convert -i \"/Users/dimitriosgiakoumas/Downloads/Otu_tables 3/otu_table_filtered.biom\" -o outputfiltered.tsv --to-tsv\n",
    "\n",
    "/Users/dimitriosgiakoumas/Downloads/otu_bacteriaonly.biom\n",
    "\n",
    "source biom-env/bin/activate\n",
    "biom convert -i \"/Users/dimitriosgiakoumas/Downloads/otu_bacteriaonly.biom\" -o outputfiltered.tsv --to-tsv\n",
    "\n",
    "source biom-env/bin/activate\n",
    "biom convert -i \"/Users/dimitriosgiakoumas/Desktop/zr9410.16S_230202.zymo/00...AllSamples.Bac16Sv34/DADA2_ASV_Distribution/ASV_Table.biom\" -o outputeven112500.tsv --to-tsv\n",
    "python -m venv biom-env\n",
    "/Users/dimitriosgiakoumas/Desktop/zr9410.16S_230202.zymo/00...AllSamples.Bac16Sv34/DADA2_ASV_Distribution/ASV_Table.biom"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "qiime tools import \\\n",
    "    --type EMPPairedEndSequences \\\n",
    "           --input-path /Users/dimitriosgiakoumas/Downloads/New_Folder_With_Items \\\n",
    "                        --output-path emp-paired-end-sequences.qza\n",
    "\n",
    "qiime tools import \\\n",
    "    --type 'SampleData[PairedEndSequencesWithQuality]' \\\n",
    "    --input-path /Users/dimitriosgiakoumas/Downloads/Zymo_rawdata_qiime2_Analysis_DG/sample-metadata.tsv \\\n",
    "    --output-path paired-end-demux.qza \\\n",
    "\n",
    "    qiime tools import \\\n",
    "    --type 'SampleData[PairedEndSequencesWithQuality]' \\\n",
    "           --input-path /Users/dimitriosgiakoumas/Downloads/Zymo_rawdata_qiime2_Analysis_DG/sample-metadata.tsv \\\n",
    "                        --output-path paired-end-demux.qza \\\n",
    "                                      --source-format PairedEndFastqManifestPhred33V2 \\\n",
    "                                                      --verbose\n",
    "qiime tools import \\\n",
    "    --type 'SampleData[PairedEndSequencesWithQuality]' \\\n",
    "           --input-path /Users/dimitriosgiakoumas/Downloads/Zymo_rawdata_qiime2_Analysis_DG/sample-metadata.tsv \\\n",
    "                        --output-path demux-paired-end.qza \\\n",
    "                                      --source-format PairedEndFastqManifestPhred33V2\n",
    "\n",
    "/Users/dimitriosgiakoumas/Downloads/Zymo_rawdata_qiime2_Analysis_DG\n",
    "unzip -q casava-18-paired-end-demultiplexed.zip                                                                     /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/zr9410_1V3V4_R1.fastq\n",
    "\n",
    "qiime tools import \\\n",
    "    --type 'SampleData[PairedEndSequencesWithQuality]' \\\n",
    "           --input-path /Users/dimitriosgiakoumas/Downloads/Zymo_rawdata_qiime2_Analysis_DG \\\n",
    "                        --input-format CasavaOneEightSingleLanePerSampleDirFmt \\\n",
    "                                       --output-path demux-paired-end.qza\n",
    "\n",
    "\n",
    "/Users/dimitriosgiakoumas/Downloads/casava-18-paired-end-demultiplexed\n",
    "qiime tools import \\\n",
    "--type 'SampleData[PairedEndSequencesWithQuality]' \\\n",
    "--input-path /Users/dimitriosgiakoumas/Downloads/Zymo_rawdata_qiime2_Analysis_DG/Manifest/sample-metadata.tsv \\\n",
    "--output-path paired-end-demux.qza \\\n",
    "--input-format PairedEndFastqManifestPhred33V2\n",
    "/Users/dimitriosgiakoumas/Downloads/Zymo_rawdata_qiime2_Analysis_DG/Manifest/sample-metadata.tsv\n",
    "\n",
    "/Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/zr9410_1V3V4_R2.fastq\n",
    "/Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/zr9410_2V3V4_R1.fastq\n",
    "/Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/zr9410_2V3V4_R2.fastq\n",
    "\n",
    "qiime tools import \\\n",
    "--type 'SampleData[PairedEndSequencesWithQuality]' \\\n",
    "--input-path /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/Manafest.csv/sample-metadata.tsv \\\n",
    "--output-path paired-end-demux.qza \\\n",
    "--input-format PairedEndFastqManifestPhred33V2\n",
    "qiime demux summarize \\\n",
    "            --i-data /Users/dimitriosgiakoumas/Downloads/Zymo_rawdata_qiime2_Analysis_DG/paired-end-demux.qza \\\n",
    "                     --o-visualization demux.qzv\n",
    "\n",
    "qiime demux summarize \\\n",
    "--i-data /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/paired-end-demux.qza \\\n",
    "--o-visualization DEMUX_SEQ_SUMMARY_VIZ.qzv\n",
    "0\n",
    "0\n",
    "\n",
    "0\n",
    "277\n",
    "\n",
    "qiime vsearch join-pairs \\\n",
    "              --i-demultiplexed-seqs /Users/dimitriosgiakoumas/Downloads/Zymo_rawdata_qiime2_Analysis_DG/paired-end-demux.qza \\\n",
    "                                     --o-joined-sequences demux-joined.qza \\\n",
    "                                                          --p-minmergelen 380 \\\n",
    "                                                                          --p-maxmergelen 460 \\\n",
    "                                                                                          --p-maxdiffs 10\n",
    "qiime deblur denoise-16S \\\n",
    "                       --i-demultiplexed-seqs /Users/dimitriosgiakoumas/Downloads/Zymo_rawdata_qiime2_Analysis_DG/paired-end-demux.qza \\\n",
    "                                              --p-trim-length -1 \\\n",
    "                                                              --p-sample-stats \\\n",
    "                                                              --o-representative-sequences representative-sequences.qza \\\n",
    "                                                                                           --o-table table.qza \\\n",
    "                                                                                                     --o-stats denoising-stats.qza\n",
    "\n",
    "qiime dada2 denoise-paired \\\n",
    "--i-demultiplexed-seqs /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/paired-end-demux.qza \\\n",
    "--p-trunc-len-f 321 \\\n",
    "--p-trunc-len-r 281 \\\n",
    "--p-trim-left-f 0 \\\n",
    "--p-trim-left-r 1 \\\n",
    "--o-table table.qza \\\n",
    "--o-representative-sequences rep-seqs.qza \\\n",
    "--o-denoising-stats denoising-stats.qza\n",
    "\n",
    "\n",
    "#visualize results from feature table\n",
    "qiime feature-table summarize \\\n",
    "--i-table /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/table.qza \\\n",
    "--o-visualization table.qzv \\\n",
    "--m-sample-metadata-file BZ15_metadata.tsv\n",
    "\n",
    "#visualize results summaries\n",
    "qiime feature-table tabulate-seqs \\\n",
    "--i-data /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/rep-seqs.qza \\\n",
    "--o-visualization rep-seqs.qzv\n",
    "\n",
    "#visualize denoising stats\n",
    "qiime metadata tabulate \\\n",
    "--m-input-file /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/denoising-stats.qza \\\n",
    "--o-visualization denoising-stats.qzv\n",
    "\n",
    "#Generate a tree for phylogenetic diversity analyses, start @ 1pm, finished 1:56pm\n",
    "qiime phylogeny align-to-tree-mafft-fasttree \\\n",
    "--i-sequences /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/rep-seqs.qza \\\n",
    "--o-alignment aligned-rep-seqs.qza \\\n",
    "--o-masked-alignment masked-aligned-rep-seqs.qza \\\n",
    "--o-tree unrooted-tree.qza \\\n",
    "--o-rooted-tree rooted-tree.qza\n",
    "\n",
    "#Taxonomy\n",
    "#make sure to use the classifier trained on the right scikit for this to work\n",
    "#can take a few hours\n",
    "qiime feature-classifier classify-sklearn \\\n",
    "--i-classifier /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/silva-138-99-515-806-nb-classifier.qza \\\n",
    "--i-reads /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/rep-seqs.qza \\\n",
    "--o-classification silva-taxonomy.qza\n",
    "\n",
    "\n",
    "qiime metadata tabulate \\\n",
    "--m-input-file /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/silva-taxonomy.qza \\\n",
    "--o-visualization silva-taxonomy.qzv\n",
    "\n",
    "\n",
    "# need to get rid of chloroplasts! might as well do mitochondria too although not usually an issue\n",
    "qiime taxa filter-table \\\n",
    "--i-table /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/table.qza \\\n",
    "--i-taxonomy /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/silva-taxonomy.qza \\\n",
    "--p-exclude mitochondria,chloroplast \\\n",
    "--o-filtered-table silva-filtered-table.qza\n",
    "\n",
    "#visualize results from feature table\n",
    "qiime feature-table summarize \\\n",
    "--i-table /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/silva-filtered-table.qza \\\n",
    "--o-visualization silva-filtered-table.qzv \\\n",
    "--m-sample-metadata-file /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/Manafest.csv/sample-metadata_zr9410_1V3V4.tsv\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "#Taxonomy Bar plots\n",
    "#visualize filtered taxa bar plots\n",
    "qiime taxa barplot \\\n",
    "--i-table /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/silva-filtered-table.qza \\\n",
    "--i-taxonomy /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/silva-taxonomy.qza \\\n",
    "--m-metadata-file /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/Manafest.csv/sample-metadata_zr9410_1V3V4.tsv \\\n",
    "--o-visualization silva-filtered-taxa-bar-plots.qzv\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "#####Filtering step specific to my experiment (threshold set at 70 reads)\n",
    "\n",
    "qiime feature-table filter-features \\\n",
    "--i-table silva-filtered-table.qza \\\n",
    "--p-min-frequency 70 \\\n",
    "--o-filtered-table silva-70-filtered-table.qza\n",
    "\n",
    "#visualize\n",
    "qiime feature-table summarize \\\n",
    "--i-table silva-70-filtered-table.qza \\\n",
    "--o-visualization silva-70-filtered-table.qzv \\\n",
    "--m-sample-metadata-file BZ15_metadata.tsv\n",
    "\n",
    "#Now lowest depth is AD069 @ 56,983 followed by AD067 @ 66,676\n",
    "\n",
    "Taxonomy Bar plots\n",
    "#visualize filtered taxa bar plots\n",
    "qiime taxa barplot \\\n",
    "--i-table silva-70-filtered-table.qza \\\n",
    "--i-taxonomy silva-taxonomy.qza \\\n",
    "--m-metadata-file BZ15_metadata.tsv \\\n",
    "--o-visualization silva-70-filtered-taxa-bar-plots.qzv\n",
    "\n",
    "\n",
    "#Results indicate that there are still ~50 genera that we only see in CD sampling! this includes streptomyces\n",
    "#continue with analysis, first rarefaction\n",
    "\n",
    "\n",
    "ALPHA RAREFACTION\n",
    "#Alpha rarefaction makes table of diversity at different depths for your curve\n",
    "#for first one, lets look at the unlimited diversity by doing max depth of 266,648\n",
    "qiime diversity alpha-rarefaction \\\n",
    "--i-table /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/silva-filtered-table.qza \\\n",
    "--i-phylogeny /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/rooted-tree.qza \\\n",
    "--p-max-depth 20000 \\\n",
    "--m-metadata-file /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/Manafest.csv/sample-metadata_zr9410_1V3V4.tsv \\\n",
    "--o-visualization alpha-rarefaction.qzv\n",
    "\n",
    "#not enough steps, redo with >10 (default)\n",
    "qiime diversity alpha-rarefaction \\\n",
    "--i-table silva-70-filtered-table.qza \\\n",
    "--i-phylogeny rooted-tree.qza \\\n",
    "--p-steps 20 \\\n",
    "--p-max-depth 266648 \\\n",
    "--m-metadata-file BZ15_metadata.tsv \\\n",
    "--o-visualization alpha-rarefaction-20steps.qzv\n",
    "\n",
    "\n",
    "\n",
    "#### Lets Rarefy to 56,983 to include all samples and go from there...\n",
    "\n",
    "qiime diversity alpha-rarefaction \\\n",
    "--i-table silva-70-filtered-table.qza \\\n",
    "--i-phylogeny rooted-tree.qza \\\n",
    "--p-max-depth 56983 \\\n",
    "--m-metadata-file BZ15_metadata.tsv \\\n",
    "--o-visualization alpha-rarefaction-56k.qzv\n",
    "\n",
    "#I think it looks okay in terms of being saturated\n",
    "\n",
    "\n",
    "#now do diversity core metrices\n",
    "qiime diversity core-metrics-phylogenetic \\\n",
    "--i-phylogeny /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/rooted-tree.qza \\\n",
    "--i-table /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/silva-filtered-table.qza \\\n",
    "--p-sampling-depth 20000 \\\n",
    "--m-metadata-file /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/Manafest.csv/sample-metadata_zr9410_1V3V4.tsv \\\n",
    "--output-dir 56k/core-metrics-results-56k\n",
    "\n",
    "\n",
    "\n",
    "qiime phylogeny fasttree --i-alignment {input.filtered_aligned_seqs} --o-tree {output.tree} --verbose --p-n-threads {threads}\n",
    "\n",
    "qiime phylogeny midpoint-root --i-tree {input.tree} --o-rooted-tree {output.tree}"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "qiime tools import \\\n",
    "    --type 'SampleData[PairedEndSequencesWithQuality]' \\\n",
    "           --input-path /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/muxed-pe-barcode-in-seq/Undetermined_S0_L001_I1_001.fastq.gz \\\n",
    "           --output-path demux-paired-end.qza \\\n",
    "                         --input-format PairedEndFastqManifestPhred33\n",
    "\n",
    "qiime tools import \\\n",
    "    --type 'SampleData[PairedEndSequencesWithQuality]' \\\n",
    "           --input-path /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/muxed-pe-barcode-in-seq/Undetermined_S0_L001_R2_001.fastq.gz \\\n",
    "           --output-path demux-paired-end.qza \\\n",
    "                         --input-format PairedEndFastqManifestPhred33\n",
    "\n",
    "qiime tools import \\\n",
    "    --type 'SampleData[PairedEndSequencesWithQuality]' \\\n",
    "           --input-path /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/muxed-pe-barcode-in-seq/Undetermined_S0_L001_R1_001.fastq.gz \\\n",
    "           --output-path demux-paired-end.qza \\\n",
    "                         --input-format PairedEndFastqManifestPhred33\n",
    "\n",
    "/Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/muxed-pe-barcode-in-seq/Undetermined_S0_L001_I1_001.fastq.gz\n",
    "/Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/muxed-pe-barcode-in-seq/Undetermined_S0_L001_R1_001.fastq.gz\n",
    "/Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/muxed-pe-barcode-in-seq/Undetermined_S0_L001_R2_001.fastq.gz\n",
    "qiime tools import \\\n",
    "    --type EMPPairedEndSequences \\\n",
    "           --input-path /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/muxed-pe-barcode-in-seq \\\n",
    "                        --output-path emp-paired-end-sequences.qza\n",
    "\n",
    "qiime tools import \\\n",
    "    --type EMPPairedEndSequences \\\n",
    "           --input-path /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/emp-paired-end-sequences \\\n",
    "                        --output-path emp-paired-end-sequences.qza\n",
    "/Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/muxed-pe-barcode-in-seq\n",
    "\n",
    "-rw-r--r-- 1 username groupname 12345 Jan  1 00:00 /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/emp-paired-end-sequences \\\n",
    "\n",
    "\n",
    "qiime demux emp-paired \\\n",
    "--m-barcodes-file /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/emp-paired-end-sequences/Archbold_Barcode_Sample_Metadata.tsv \\\n",
    "--m-barcodes-column barcode-sequence \\\n",
    "--p-rev-comp-mapping-barcodes \\\n",
    "--i-seqs /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/emp-paired-end-sequences/emp-paired-end-sequences.qza \\\n",
    "--o-per-sample-sequences demux-full.qza \\\n",
    "--o-error-correction-details demux-details.qza\n",
    "\n",
    "qiime demux summarize \\\n",
    "--i-data /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/demux-details.qza \\\n",
    "--o-visualization demux-details.qzv\n",
    "\n",
    "denoise step archbold when 25th percetile gets below 30 i will truncate it A common approach is to truncate the sequences at a point where the quality scores start to fall below a certain threshold, such as 30, which corresponds to a 1 in 1000 chance of an error.\n",
    "0 forwawrd left\n",
    "251 forward right\n",
    "\n",
    "reverse reads were pretty bad so i am leaving them out and only denoising the forward reads\n",
    "GPT4 gernerated code:\n",
    "qiime dada2 denoise-single \\\n",
    "--i-demultiplexed-seqs /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/demux-full.qza \\\n",
    "--p-trim-left 0 \\\n",
    "--p-trunc-len 251 \\\n",
    "--o-representative-sequences rep-seqs-dada2.qza \\\n",
    "--o-table table-dada2.qza \\\n",
    "--o-denoising-stats stats-dada2.qza\n",
    "\n",
    "i took pretrained nieve bayes classifiers from qiime2 website\n",
    "Michael S Robeson II, Devon R O’Rourke, Benjamin D Kaehler, Michal Ziemski, Matthew R Dillon, Jeffrey T Foster, Nicholas A Bokulich. RESCRIPt: Reproducible sequence taxonomy reference database management for the masses. bioRxiv 2020.10.05.326504; doi: https://doi.org/10.1101/2020.10.05.326504\n",
    "\n",
    "Bokulich, N.A., Kaehler, B.D., Rideout, J.R. et al. Optimizing taxonomic classification of marker-gene amplicon sequences with QIIME 2’s q2-feature-classifier plugin. Microbiome 6, 90 (2018). https://doi.org/10.1186/s40168-018-0470-z\n",
    "\n",
    "qiime feature-classifier classify-sklearn \\\n",
    "--i-classifier /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/gg-13-8-99-515-806-nb-classifier.qza \\\n",
    "--i-reads /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/rep-seqs-dada2.qza \\\n",
    "--o-classification taxonomy.qza\n",
    "\n",
    "qiime feature-classifier classify-sklearn \\\n",
    "--i-classifier /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/silva-138-99-515-806-nb-classifier.qza \\\n",
    "--i-reads /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/rep-seqs-dada2.qza \\\n",
    "--o-classification taxonomy.qza\n",
    "\n",
    "\n",
    "qiime metadata tabulate \\\n",
    "--m-input-file /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/taxonomy.qza \\\n",
    "--o-visualization taxonomygg.qzv\n",
    "\n",
    "qiime metadata tabulate \\\n",
    "               --m-input-file /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/taxonomysilva.qza \\\n",
    "               --o-visualization taxonomysilva.qzv\n",
    "\n",
    "qiime taxa barplot \\\n",
    "--i-table /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/table-dada2.qza \\\n",
    "--i-taxonomy /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/taxonomysilva.qza \\\n",
    "--m-metadata-file /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/emp-paired-end-sequences/Archbold_Barcode_Sample_Metadata.tsv \\\n",
    "--o-visualization taxa-bar-plotssilva.qzv\n",
    "\n",
    "\n",
    "\n",
    "qiime vsearch uclust \\\n",
    "--i-sequences /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/rep-seqs-dada2.qza \\\n",
    "--o-clustering clustered_sequences.qza \\\n",
    "--p-perc-identity 0.97 \\\n",
    "--p-threads 4\n",
    "qiime vsearch cluster-features-de-novo \\\n",
    "              --i-table /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/table-dada2.2.qza \\\n",
    "                        --i-sequences /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/rep-seqs-dada2.2.qza \\\n",
    "                                      --p-perc-identity 0.97 \\\n",
    "                                                        --o-clustered-table table-dn-97.qza \\\n",
    "                                                                                        --o-clustered-sequences rep-seqs-dn-97.qza\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "qiime feature-table filter-features \\\n",
    "                    --i-table /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/table-dada2.qza \\\n",
    "                              --i-sequences /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/rep-seqs-dada2.qza \\\n",
    "                                            --p-where \"taxonomy NOT LIKE '%Chloroplast%' AND taxonomy NOT LIKE '%Mitochondria%'\" \\\n",
    "                                                      --o-filtered-table table-no-chimeras.qza \\\n",
    "                                                                         --o-filtered-sequences rep-seqs-no-chimeras.qza\n",
    "qiime vsearch uchime-denovo \\\n",
    "              --i-table /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/table-dada2.qza \\\n",
    "                        --i-sequences /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/rep-seqs-dada2.qza \\\n",
    "                                      --output-dir uchime-dn-out\n",
    "\n",
    "qiime metadata tabulate --m-input-file /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/uchime-dn-out/stats.qza \\\n",
    "                                       --o-visualization uchime-dn-out/stats.qzv\n",
    "\n",
    "qiime tools view uchime-dn-out/stats.qzv\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "\n",
    "import pandas as pd\n",
    "\n",
    "# Assuming the first column contains the OTU identifiers\n",
    "data1 = pd.read_excel('/Users/dimitriosgiakoumas/Downloads/Micro/Micro1.xlsx', index_col=0)\n",
    "data2 = pd.read_excel('/Users/dimitriosgiakoumas/Downloads/Micro/Micro2.xlsx', index_col=0)\n",
    "\n",
    "# Combine the two datasets\n",
    "combined_data = pd.concat([data1, data2], axis=1)\n",
    "\n",
    "combined_data.to_excel('/Users/dimitriosgiakoumas/Downloads/micro/compo.xlsx')\n",
    "\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "from Bio import SeqIO\n",
    "\n",
    "def parse_fna_file(file_path):\n",
    "    smallest = float('inf')  # Initialize smallest to positive infinity\n",
    "    largest = float('-inf')  # Initialize largest to negative infinity\n",
    "\n",
    "    # Parse the .fna file with SeqIO.parse\n",
    "    for record in SeqIO.parse(file_path, 'fasta'):\n",
    "        # The description field usually contains the \"size=#\" attribute\n",
    "        description = record.description\n",
    "\n",
    "        # Find the \"size=#\" in the description\n",
    "        size_index = description.find('size=')\n",
    "        if size_index != -1:\n",
    "            # Extract the size value\n",
    "            size_value = int(description[size_index+5:].split()[0])\n",
    "            # Update smallest and largest if necessary\n",
    "            smallest = min(smallest, size_value)\n",
    "            largest = max(largest, size_value)\n",
    "\n",
    "    return smallest, largest\n",
    "\n",
    "# Replace 'your_file.fna' with your actual file path\n",
    "file_path = '/Users/dimitriosgiakoumas/Desktop/zr10770.16S_230509.zymo/00...AllSamples.Bac16Sv34/DADA2_ASV_Distribution/sv.seqs.fna'\n",
    "smallest, largest = parse_fna_file(file_path)\n",
    "\n",
    "print(f'Smallest size value: {smallest}')\n",
    "print(f'Largest size value: {largest}')\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": "!/usr/local/bin/python3.9 -m pip install Bio",
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "qiime diversity alpha-group-significance \\\n",
    "--i-alpha-diversity /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/56k/core-metrics-results-56k/evenness_vector.qza \\\n",
    "--m-metadata-file /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/Manafest.csv/sample-metadata_zr9410_1V3V4.tsv \\\n",
    "--o-visualization /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/56k/core-metrics-results-56k/evenness_vector.qzv\n",
    "qiime diversity beta-group-significance \\\n",
    "--i-distance-matrix /path/to/your/distance/matrix.qza \\\n",
    "--m-metadata-file /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/Manafest.csv/sample-metadata_zr9410_1V3V4.tsv \\\n",
    "--o-visualization /path/to/your/visualization.qzv\n",
    "qiime emperor plot \\\n",
    "              --i-pcoa /path/to/your/pcoa/results.qza \\\n",
    "              --m-metadata-file /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/Manafest.csv/sample-metadata_zr9410_1V3V4.tsv \\\n",
    "              --o-visualization /path/to/your/visualization.qzv\n",
    "\n",
    "\n",
    "/Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/56k/core-metrics-results-56k/evenness_vector.qza\n",
    "/Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/56k/core-metrics-results-56k/shannon_vector.qza\n",
    "/Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/56k/core-metrics-results-56k/observed_features_vector.qza\n",
    "/Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/56k/core-metrics-results-56k/faith_pd_vector.qza\n",
    "/Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/56k/core-metrics-results-56k/bray_curtis_pcoa_results.qza\n",
    "/Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/56k/core-metrics-results-56k/jaccard_pcoa_results.qza\n",
    "/Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/56k/core-metrics-results-56k/weighted_unifrac_pcoa_results.qza\n",
    "/Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/56k/core-metrics-results-56k/unweighted_unifrac_pcoa_results.qza\n",
    "/Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/56k/core-metrics-results-56k/bray_curtis_distance_matrix.qza\n",
    "/Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/56k/core-metrics-results-56k/jaccard_distance_matrix.qza\n",
    "/Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/56k/core-metrics-results-56k/weighted_unifrac_distance_matrix.qza\n",
    "/Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/56k/core-metrics-results-56k/unweighted_unifrac_distance_matrix.qza\n",
    "qiime metadata tabulate \\\n",
    "--m-input-file /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/rooted-tree.qza \\\n",
    "--o-visualization /Users/dimitriosgiakoumas/Downloads/zr9410.rawdata.230202/rooted-tree.qzv\n",
    "chmod +x /Users/dimitriosgiakoumas/Downloads/convertqza.py\n",
    "./Users/dimitriosgiakoumas/Downloads/convertqza.py\n",
    ".//Users/dimitriosgiakoumas/Downloads/convertqza.py"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Save the dataframe to a new excel file\n",
    "df = pd.read_excel('/Users/dimitriosgiakoumas/Downloads/Micro/Micro_combined_data.xlsx')\n",
    "\n",
    "# Fill NaN values with 0\n",
    "df = df.fillna(0)\n",
    "\n",
    "df.to_excel('/Users/dimitriosgiakoumas/Downloads/Micro/Micro_combined_data2.xlsx')\n",
    "\n",
    "\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": "!/usr/local/bin/python3.9 -m pip install openpyxl",
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "source": [
    "qiime feature-table summarize \\\n",
    "                    --i-table table.qza \\\n",
    "                              --o-visualization table.qzv\n",
    "\n",
    "\n",
    "qiime diversity pcoa \\\n",
    "                --i-distance-matrix braycurtis_distance_matrix.qza \\\n",
    "                                    --o-pcoa braycurtis_pcoa_results.qza\n",
    "qiime emperor plot \\\n",
    "              --i-pcoa braycurtis_pcoa_results.qza \\\n",
    "                       --m-metadata-file /Users/dimitriosgiakoumas/Downloads/Zymo_rawdata_qiime2_Analysis_DG/Manifest/sample-metadata.tsv \\\n",
    "                                         --o-visualization braycurtis_pcoa_plot.qzv\n",
    "qiime emperor plot \\\n",
    "              --i-pcoa weighted_unifrac_pcoa_results.qza \\\n",
    "                       --m-metadata-file /Users/dimitriosgiakoumas/Downloads/Zymo_rawdata_qiime2_Analysis_DG/Manifest/sample-metadata.tsv \\\n",
    "                                         --o-visualization weighted_unifrac_emperor.qzv\n",
    "qiime diversity pcoa \\\n",
    "                --i-distance-matrix /Users/dimitriosgiakoumas/Downloads/Zymo_rawdata_qiime2_Analysis_DG/weighted_unifrac_distance_matrix.qza \\\n",
    "                                    --o-pcoa weighted_unifrac_pcoa_results.qza\n",
    "\n",
    "qiime diversity alpha-rarefaction \\\n",
    "                --i-table table.qza \\\n",
    "                          --i-phylogeny rooted-tree.qza \\\n",
    "                                        --p-max-depth 7722 \\\n",
    "                                                      --m-metadata-file /Users/dimitriosgiakoumas/Downloads/Zymo_rawdata_qiime2_Analysis_DG/Manifest/sample-metadata.tsv \\\n",
    "                                                                        --o-visualization alpha-rarefaction.qzv\n",
    "\n",
    "\n",
    "qiime diversity core-metrics-phylogenetic \\\n",
    "                --i-phylogeny rooted-tree.qza \\\n",
    "                              --i-table table.qza \\\n",
    "                                        --p-sampling-depth 7722 \\\n",
    "                                                           --m-metadata-file /Users/dimitriosgiakoumas/Downloads/Zymo_rawdata_qiime2_Analysis_DG/Manifest/sample-metadata.tsv \\\n",
    "                                                                             --output-dir core-metrics-results\n",
    "\n",
    "\n",
    "qiime diversity alpha-group-significance \\\n",
    "                --i-alpha-diversity core-metrics-results/faith_pd_vector.qza \\\n",
    "                                    --m-metadata-file /Users/dimitriosgiakoumas/Downloads/Zymo_rawdata_qiime2_Analysis_DG/Manifest/sample-metadataMVF.tsv \\\n",
    "                                                      --o-visualization core-metrics-results/faith-pd-group-significance.qzv\n",
    "\n",
    "qiime diversity alpha-group-significance \\\n",
    "                --i-alpha-diversity core-metrics-results/evenness_vector.qza \\\n",
    "                                    --m-metadata-file /Users/dimitriosgiakoumas/Downloads/Zymo_rawdata_qiime2_Analysis_DG/Manifest/sample-metadataMVF.tsv\n",
    "qiime diversity alpha-group-significance \\\n",
    "                --i-alpha-diversity core-metrics-results/faith_pd_vector.qza \\\n",
    "                                    --m-metadata-file /Users/dimitriosgiakoumas/Downloads/Zymo_rawdata_qiime2_Analysis_DG/Manifest/sample-metadataMVF.tsv \\\n",
    "                                    --o-visualization core-metrics-results/faith-pd-group-significance.qzv \\\n",
    "                                                      --o-visualization core-metrics-results/evenness-group-significance.qzv\n",
    "\n",
    "qiime diversity alpha-group-significance \\\n",
    "                --i-alpha-diversity core-metrics-results/shannon_vector.qza \\\n",
    "                                    --m-metadata-file\n",
    "qiime diversity alpha-group-significance \\\n",
    "                --i-alpha-diversity core-metrics-results/faith_pd_vector.qza \\\n",
    "                                    --m-metadata-file /Users/dimitriosgiakoumas/Downloads/Zymo_rawdata_qiime2_Analysis_DG/Manifest/sample-metadataMVF.tsv \\\n",
    "                                    --o-visualization core-metrics-results/faith-pd-group-significance.qzv \\\n",
    "                                                      --o-visualization core-metrics-results/shannon_group-significance.qzv\n",
    "\n",
    "\n",
    "\n",
    "qiime diversity alpha-group-significance \\\n",
    "                --i-alpha-diversity core-metrics-results/evenness_vector.qza \\\n",
    "                                    --m-metadata-file /Users/dimitriosgiakoumas/Downloads/Zymo_rawdata_qiime2_Analysis_DG/Manifest/sample-metadataMVF.tsv \\\n",
    "                                                      --o-visualization core-metrics-results/evenness-group-significance.qzv\n",
    "\n",
    "\n",
    "qiime diversity alpha-group-significance \\\n",
    "                --i-alpha-diversity core-metrics-results/shannon_vector.qza \\\n",
    "                                    --m-metadata-file /Users/dimitriosgiakoumas/Downloads/Zymo_rawdata_qiime2_Analysis_DG/Manifest/sample-metadataMVF.tsv \\\n",
    "                                                      --o-visualization core-metrics-results/shannon_group-significance.qzv\n",
    "\n",
    "\n",
    "qiime diversity beta-group-significance \\\n",
    "                --i-distance-matrix core-metrics-results/unweighted_unifrac_distance_matrix.qza \\\n",
    "                                    --m-metadata-file /Users/dimitriosgiakoumas/Downloads/Zymo_rawdata_qiime2_Analysis_DG/Manifest/sample-metadataMVF.tsv \\\n",
    "                                                      --m-metadata-column Sex \\\n",
    "                                                                          --o-visualization core-metrics-results/unweighted-unifrac-body-site-significance.qzv \\\n",
    "                                                                                            --p-pairwise\n",
    "\n",
    "#filter based on body site\n",
    "qiime feature-table filter-samples \\\n",
    "                    --i-table table.qza \\\n",
    "                              --m-metadata-file sample-metadata.tsv \\\n",
    "                                                --p-where \"BodySite='gut'\" \\\n",
    "                                                          --o-filtered-table gut-table.qza\n",
    "\n",
    "\n",
    "\n",
    "qiime feature-classifier extract-reads \\\n",
    "                         --i-sequences /Users/dimitriosgiakoumas/Downloads/Zymo_rawdata_qiime2_Analysis_DG/silva_132_99_v3v4.qza \\\n",
    "                                       --p-f-primer ACTCCTACGGGAGGCAGCAG \\\n",
    "                                                    --p-r-primer GGACTACHVGGGTWTCTAAT \\\n",
    "                                                                 --o-reads silva_132_99_16S_v3v4.qza\n",
    "qiime feature-classifier classify-sklearn \\\n",
    "                         --i-classifier /Users/dimitriosgiakoumas/Downloads/Zymo_rawdata_qiime2_Analysis_DG/classifier_GG16S_99otus_v3-4_338F-806R.qza \\\n",
    "                                        --i-reads /Users/dimitriosgiakoumas/Downloads/Zymo_rawdata_qiime2_Analysis_DG/representative-sequences120.qza \\\n",
    "                                                  --o-classification your_sample_taxonomy.qza\n"
   ],
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "source": [
    "/Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/emp-paired-end-sequences/barcodes.fastq.gz\n",
    "/Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/emp-paired-end-sequences/reverse.fastq.gz\n",
    "/Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/emp-paired-end-sequences/forward.fastq.gz\n",
    "\n",
    "\n",
    "# Import the forward reads\n",
    "qiime tools import \\\n",
    "    --type 'SampleData[PairedEndSequencesWithQuality]' \\\n",
    "           --input-path /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/emp-paired-end-sequences\\\n",
    "                        --output-path forward.qza\n",
    "\n",
    "# Import the reverse reads\n",
    "qiime tools import \\\n",
    "    --type 'SampleData[PairedEndSequencesWithQuality]' \\\n",
    "           --input-path /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/emp-paired-end-sequences/reverse.fastq.gz \\\n",
    "                        --output-path reverse.qza\n",
    "\n",
    "# Import the barcodes\n",
    "qiime tools import \\\n",
    "    --type 'SampleData[BarcodeSequences]' \\\n",
    "           --input-path /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/emp-paired-end-sequences/barcodes.fastq.gz \\\n",
    "                        --output-path barcodes.qza\n",
    "\n",
    "# Demultiplex the reads\n",
    "qiime demux emp-paired \\\n",
    "            --m-barcodes-file barcodes.qza \\\n",
    "                              --m-barcodes-column barcode-sequence \\\n",
    "                                                  --p-rev-comp-mapping-barcodes \\\n",
    "                                                  --i-seqs forward.qza \\\n",
    "                                                           --o-per-sample-sequences demux.qza \\\n",
    "                                                                                    --o-error-correction-details demux-details.qza\n"
   ],
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "source": [
    "bbduk.sh in=/Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/emp-paired-end-sequences/barcodes.fastq.gz in2=/Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/emp-paired-end-sequences/reverse.fastq.gz  inbarcode=/Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/emp-paired-end-sequences/barcodes.fastq.gz out1=out_%.fq out2=out2_%.fq\n",
    "bbdemux.sh in1=forward.fastq.gz in2=reverse.fastq.gz inbarcode=barcodes.fastq.gz out1=out_%.fq out2=out2_%.fq\n"
   ],
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "source": [
    "/Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/emp-paired-end-sequences/barcodes.fastq.gz\n",
    "/Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/emp-paired-end-sequences/reverse.fastq.gz\n",
    "/Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/emp-paired-end-sequences/forward.fastq.gz\n",
    "\n",
    "\n",
    "# Concatenate the forward and reverse reads\n",
    "cat /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/emp-paired-end-sequences/forward.fastq.gz /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/emp-paired-end-sequences/reverse.fastq.gz > combined.fastq.gz\n",
    "\n",
    "# Import the data into QIIME\n",
    "qiime tools import \\\n",
    "    --type 'SampleData[SequencesWithQuality]' \\\n",
    "           --input-path /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/combined.fastq.gz \\\n",
    "                        --output-path combined.qza\n",
    "\n",
    "# Demultiplex the reads\n",
    "qiime cutadapt demux-paired \\\n",
    "               --i-seqs combined.qza \\\n",
    "                        --m-forward-barcodes-file metadata.tsv \\\n",
    "                                                  --m-forward-barcodes-column BarcodeSequence \\\n",
    "                                                                              --p-error-rate 0 \\\n",
    "                                                                                             --o-per-sample-sequences demux.qza \\\n",
    "                                                                                                                      --o-untrimmed-sequences untrimmed.qza\n"
   ],
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "source": [
    "qiime cutadapt demux-paired\n",
    "--i-seqs /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/emp-paired-end-sequences/forward.fastq.gz\n",
    "--m-forward-barcodes-file /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/emp-paired-end-sequences/barcodes.fastq.gz\n",
    "--m-forward-barcodes-column /Users/dimitriosgiakoumas/Documents/GitHub/microbiopy_demo/GutMicrobiome_16SV4/emp-paired-end-sequences/Archbold_Barcode_Sample_Metadata.tsv\n",
    "\n",
    "--o-per-sample-sequences SampleData[PairedEndSequencesWithQuality]\n",
    "--o-untrimmed-sequences MultiplexedPairedEndBarcodeInSequence\n",
    "--output-dir DIRECTORY"
   ],
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "source": [
    "qiime tools export \\\n",
    "--input-path /Users/dimitriosgiakoumas/Downloads/table_2k_abund.qza \\\n",
    "--output-path exported-feature-table\n",
    "biom convert -i exported-feature-table/feature-table.biom -o feature-table.tsv --to-tsv\n",
    "\n",
    "\n",
    "qiime tools export \\\n",
    "            --input-path /Users/dimitriosgiakoumas/Downloads/table_2k.qza \\\n",
    "            --output-path exported-feature-table\n",
    "\n",
    "qiime tools export \\\n",
    "--input-path ./Users/dimitriosgiakoumas/Downloads/table_2k.qza \\ \n",
    "--output-path /Users/dimitriosgiakoumas/Downloads/table_2k.qza\n",
    "\n",
    "qiime feature-table filter-features \\\n",
    "                    --i-table ./table_2k.qza \\\n",
    "                    --p-min-frequency 50 \\\n",
    "                                      --p-min-samples 4 \\\n",
    "                                                      --o-filtered-table ./table_2k_abund.qza\n",
    "\n",
    "\n",
    "biom convert -i /Users/dimitriosgiakoumas/Downloads/ASV_Table.csv -o /Users/dimitriosgiakoumas/Downloads/ASV_Table.biom --table-type=\"OTU table\" --to-hdf5\n",
    "\n",
    "biom convert -i /Users/dimitriosgiakoumas/Downloads/ASV_Table.csv -o /Users/dimitriosgiakoumas/Downloads/ASV_Table.biom --table-type=\"OTU table\" --to-hdf5 --process-obs-metadata taxonomy --tsv-separator \",\"\n",
    "\n",
    "\n",
    "qiime tools import \\\n",
    "    --input-path /Users/dimitriosgiakoumas/Downloads/ASV_Table.biom \\\n",
    "    --type 'FeatureTable[Frequency]' \\\n",
    "           --input-format BIOMV210Format \\\n",
    "                          --output-path /Users/dimitriosgiakoumas/Downloads/ASV_Table.qza\n",
    "\n",
    "\n",
    "qiime feature-table filter-features \\\n",
    "                    --i-table /Users/dimitriosgiakoumas/Downloads/ASV_Table.qza \\\n",
    "                    --p-min-frequency 10 \\\n",
    "                                      --o-filtered-table /Users/dimitriosgiakoumas/Downloads/filtered-feature-table.qza\n",
    "\n",
    "qiime composition add-pseudocount \\\n",
    "                  --i-table /Users/dimitriosgiakoumas/Downloads/filtered-feature-table.qza \\\n",
    "                  --o-composition-table /Users/dimitriosgiakoumas/Downloads/comp-feature-table.qza\n",
    "\n",
    "qiime composition ancom \\\n",
    "                  --i-table /Users/dimitriosgiakoumas/Downloads/ANCOM/comp-feature-table.qza \\\n",
    "                  --m-metadata-file /Users/dimitriosgiakoumas/Downloads/ANCOM/Metadata.tsv \\\n",
    "                  --m-metadata-column group_column \\\n",
    "                                      --o-visualization /Users/dimitriosgiakoumas/Downloads/ANCOM-results.qzv\n",
    "\n",
    "qiime tools view /Users/dimitriosgiakoumas/Downloads/ancom-results.qzv\n",
    "\n",
    "/Users/dimitriosgiakoumas/Downloads/ANCOM\n",
    "\n",
    "qiime gneiss correlation-clustering \\\n",
    "             --i-table /Users/dimitriosgiakoumas/Downloads/ANCOM/comp-feature-table.qza \\\n",
    "             --o-clustering /Users/dimitriosgiakoumas/Downloads/ANCOM/hierarchy.qza\n",
    "\n",
    "qiime tools import \\\n",
    "    --input-path /Users/dimitriosgiakoumas/Downloads/ANCOM/ASV_Table_dense.biom \\\n",
    "    --type 'FeatureTable[Frequency]' \\\n",
    "           --input-format BIOMV210Format \\\n",
    "                          --output-path /Users/dimitriosgiakoumas/Downloads/ANCOM/feature-table.qza\n",
    "\n",
    "qiime gneiss correlation-clustering \\\n",
    "             --i-table /Users/dimitriosgiakoumas/Downloads/ANCOM/feature-table.qza \\\n",
    "             --o-clustering /Users/dimitriosgiakoumas/Downloads/ANCOM/hierarchy.qza\n",
    "\n",
    "\n",
    "qiime gneiss ols-regression \\\n",
    "             --p-formula \"C(sex)\" \\\n",
    "                         --i-table /Users/dimitriosgiakoumas/Downloads/ANCOM/feature-table.qza \\\n",
    "                         --i-tree /Users/dimitriosgiakoumas/Downloads/ANCOM/hierarchy.qza \\\n",
    "                         --m-metadata-file /Users/dimitriosgiakoumas/Downloads/ANCOM/Metadata.tsv \\\n",
    "                         --o-visualization /Users/dimitriosgiakoumas/Downloads/ANCOM/regression_summary.qzv\n",
    "\n",
    "\n",
    "qiime gneiss dendrogram-heatmap \\\n",
    "             --i-table table.qza \\\n",
    "                       --i-tree hierarchy.qza \\\n",
    "                                --m-metadata-file sample-metadata.tsv \\\n",
    "                                                  --m-metadata-column Subject \\\n",
    "                                                                      --p-color-map seismic \\\n",
    "                                                                                    --o-visualization heatmap.qzv\n",
    "\n",
    "\n",
    "\n",
    "qiime tools import \\\n",
    "    --type 'FeatureTable[Frequency]' \\\n",
    "           --input-path /data/your-feature-table.biom \\\n",
    "           --output-path /data/your-imported-table.qza\n",
    "\n",
    "\n",
    "\n",
    "conda install wget\n",
    "wget https://data.qiime2.org/distro/amplicon/qiime2-amplicon-2023.9-py38-osx-conda.yml\n",
    "conda env create -n qiime2-amplicon-2023.9 --file qiime2-amplicon-2023.9-py38-osx-conda.yml\n",
    "conda activate qiime2-<distro>-2023.9"
   ],
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "source": [
    "import biom\n",
    "import pandas as pd\n",
    "\n",
    "def convert_biom_to_csv(biom_file, csv_file):\n",
    "    # Load the BIOM file\n",
    "    table = biom.load_table(biom_file)\n",
    "\n",
    "    # Convert to a Pandas DataFrame\n",
    "    df = table.to_dataframe(dense=True)\n",
    "\n",
    "    # Write the DataFrame to a CSV file\n",
    "    df.to_csv(csv_file)\n",
    "\n",
    "# Example usage\n",
    "biom_file = '/Users/dimitriosgiakoumas/Downloads/ASV_Table.biom'  # Replace with your .biom file path\n",
    "csv_file = '/Users/dimitriosgiakoumas/Downloads/ASV_Table.csv'  # Replace with your desired output path\n",
    "\n",
    "convert_biom_to_csv(biom_file, csv_file)\n"
   ],
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "source": [
    "import csv\n",
    "\n",
    "with open('/Users/dimitriosgiakoumas/Downloads/ASV_Table.csv', 'r') as csv_file:\n",
    "    csv_reader = csv.reader(csv_file)\n",
    "    with open('/Users/dimitriosgiakoumas/Downloads/ASV_Table.tsv', 'w', newline='') as tsv_file:\n",
    "        tsv_writer = csv.writer(tsv_file, delimiter='\\t')\n",
    "        for row in csv_reader:\n",
    "            tsv_writer.writerow(row)\n"
   ],
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "execution_count": null
  },
  {
   "cell_type": "code",
   "source": [
    "import csv\n",
    "\n",
    "# Replace these file paths with your file's paths\n",
    "input_csv_file = '/Users/dimitriosgiakoumas/Downloads/ANCOM/Metadata.csv'\n",
    "output_tsv_file = '/Users/dimitriosgiakoumas/Downloads/ANCOM/Metadata.tsv'\n",
    "\n",
    "with open(input_csv_file, 'r') as csv_file, open(output_tsv_file, 'w', newline='') as tsv_file:\n",
    "    csv_reader = csv.reader(csv_file, delimiter=',')\n",
    "    tsv_writer = csv.writer(tsv_file, delimiter='\\t')\n",
    "\n",
    "    for row in csv_reader:\n",
    "        tsv_writer.writerow(row)\n"
   ],
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": "",
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": [
    "import pandas as pd\n",
    "import plotly.express as px\n",
    "\n",
    "\n",
    "# Example OTU table with 16 samples\n",
    "otu_table = pd.read_excel(\"/Users/dimitriosgiakoumas/Downloads/Micro/Phylum_Merged.xlsx\")\n",
    "\n",
    "otu_table.iloc[:, 1:] *= 100\n",
    "plt.figure(dpi=1200)\n",
    "# Calculate average relative abundance for each OTU\n",
    "#otu_table['Average'] = otu_table.iloc[:, 1:].mean(axis=1)\n",
    "\n",
    "# Melt DataFrame\n",
    "df_melted = pd.melt(otu_table, id_vars=['OTU'], var_name='T', value_name='Relative Abundance')\n",
    "\n",
    "# Set y-axis range to 0-100%\n",
    "#fig.update_yaxes(range=[0, 100])\n",
    "fig = px.bar(df_melted, x='T', y='Relative Abundance', color='OTU',\n",
    "             title='', barmode='stack',\n",
    "             color_discrete_sequence=px.colors.qualitative.Alphabet)\n",
    "fig.update_layout(\n",
    "    xaxis_title=\"Sample ID\"\n",
    ")\n",
    "# Update layout\n",
    "fig.update_layout(legend=dict(orientation=\"h\", yanchor=\"bottom\", y=1.02, xanchor=\"right\", x=1),\n",
    "                  margin=dict(t=100, b=50, l=50, r=50))\n",
    "fig.update_layout(plot_bgcolor='rgba(255, 255, 255, 1)')\n",
    "# Show plot\n",
    "fig.show()\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": [
    "import pandas as pd\n",
    "import plotly.express as px\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "# Example OTU table with 16 samples\n",
    "otu_table = pd.read_excel(\"/Users/dimitriosgiakoumas/Downloads/Micro/Phylum_Merged.xlsx\")\n",
    "\n",
    "otu_table.iloc[:, 1:] *= 100\n",
    "plt.figure(dpi=1200)\n",
    "# Calculate average relative abundance for each OTU\n",
    "#otu_table['Average'] = otu_table.iloc[:, 1:].mean(axis=1)\n",
    "\n",
    "# Melt DataFrame\n",
    "df_melted = pd.melt(otu_table, id_vars=['OTU'], var_name='T', value_name='Relative Abundance')\n",
    "\n",
    "# Set y-axis range to 0-100%\n",
    "#fig.update_yaxes(range=[0, 100])\n",
    "fig = px.bar(df_melted, x='T', y='Relative Abundance', color='OTU',\n",
    "             title='', barmode='stack',\n",
    "             color_discrete_sequence=px.colors.qualitative.Alphabet)\n",
    "fig.update_layout(\n",
    "    xaxis_title=\"Sample ID\"\n",
    ")\n",
    "# Update layout\n",
    "fig.update_layout(legend=dict(orientation=\"h\", yanchor=\"bottom\", y=1.02, xanchor=\"right\", x=1),\n",
    "                  margin=dict(t=100, b=50, l=50, r=50))\n",
    "fig.update_layout(plot_bgcolor='rgba(255, 255, 255, 1)')\n",
    "# Show plot\n",
    "fig.show()\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": [
    "import pandas as pd\n",
    "import plotly.express as px\n",
    "\n",
    "# Load the data\n",
    "otu_table = pd.read_excel(\"/Users/dimitriosgiakoumas/Downloads/Micro/Phylum_Merged.xlsx\")\n",
    "\n",
    "# Calculate average relative abundance for each OTU\n",
    "#otu_table['Average'] = otu_table.iloc[:, 1:].mean(axis=1)\n",
    "\n",
    "# Transform the data to long format for plotting\n",
    "otu_table_long = otu_table.melt(id_vars=\"OTU\", var_name=\"T\", value_name=\"Relative Abundance\")\n",
    "\n",
    "# Convert the values in Relative Abundance column to percentage\n",
    "otu_table_long['Relative Abundance'] = otu_table_long['Relative Abundance'] * 100\n",
    "\n",
    "# Pivot the table for plotly\n",
    "pivot_table = otu_table_long.pivot_table(index='T', columns='OTU', values='Relative Abundance', fill_value=0)\n",
    "\n",
    "# Generate the plot with plotly\n",
    "fig = px.bar(pivot_table, labels={\"value\": \"Relative Abundance (%)\", \"variable\": \"OTU\", \"index\": \"Sample ID\"},\n",
    "             #title=\"Sample ID vs Relative Abundance\",\n",
    "             orientation='v', barmode='stack')\n",
    "\n",
    "# Update layout\n",
    "fig.update_layout(\n",
    "    xaxis=dict(tickangle=-45),\n",
    "    yaxis=dict(range=[0,100]),\n",
    "    legend=dict(orientation=\"h\", yanchor=\"bottom\", y=1.02, xanchor=\"right\", x=1)\n",
    ")\n",
    "\n",
    "fig.show()\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": [
    "import pandas as pd\n",
    "import plotly.express as px\n",
    "\n",
    "\n",
    "# Example OTU table with 16 samples\n",
    "otu_table = pd.read_excel(\"/Users/dimitriosgiakoumas/Downloads/Micro/Phylum_Merged.xlsx\")\n",
    "\n",
    "otu_table.iloc[:, 1:] *= 100\n",
    "plt.figure(dpi=1200)\n",
    "# Calculate average relative abundance for each OTU\n",
    "#otu_table['Average'] = otu_table.iloc[:, 1:].mean(axis=1)\n",
    "\n",
    "# Melt DataFrame\n",
    "df_melted = pd.melt(otu_table, id_vars=['OTU'], var_name='T', value_name='Relative Abundance')\n",
    "\n",
    "# Set y-axis range to 0-100%\n",
    "#fig.update_yaxes(range=[0, 100])\n",
    "fig = px.bar(df_melted, x='T', y='Relative Abundance', color='OTU',\n",
    "             title='', barmode='stack',\n",
    "             color_discrete_sequence=px.colors.qualitative.Alphabet)\n",
    "fig.update_layout(\n",
    "    xaxis_title=\"Sample ID\"\n",
    ")\n",
    "# Update layout\n",
    "fig.update_layout(legend=dict(orientation=\"h\", yanchor=\"bottom\", y=1.02, xanchor=\"right\", x=1),\n",
    "                  margin=dict(t=100, b=50, l=50, r=50))\n",
    "fig.update_layout(plot_bgcolor='rgba(255, 255, 255, 1)')\n",
    "\n",
    "# Save the plot as a PNG file with 300 DPI\n",
    "save_path = '/Users/dimitriosgiakoumas/Library/CloudStorage/OneDrive-FloridaAtlanticUniversity/otu_table.png'\n",
    "fig.write_image(save_path, format='png', width=1200, height=800, scale=3)\n",
    "\n",
    "# Show plot\n",
    "fig.show()\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": [
    "from PIL import Image\n",
    "\n",
    "# Open the PNG file\n",
    "png_image = Image.open('/Users/dimitriosgiakoumas/Library/CloudStorage/OneDrive-FloridaAtlanticUniversity/otu_table.png')\n",
    "\n",
    "# Convert the PNG image to a TIFF image\n",
    "png_image.save('/Users/dimitriosgiakoumas/Library/CloudStorage/OneDrive-FloridaAtlanticUniversity/otu_table.tiff', format='TIFF')\n",
    "\n",
    "# Close the PNG image\n",
    "png_image.close()\n"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-07-18T01:44:26.754049Z",
     "start_time": "2024-07-18T01:44:25.429207Z"
    }
   },
   "cell_type": "code",
   "source": [
    "import plotly.graph_objects as go\n",
    "from PIL import Image\n",
    "\n",
    "# Sample data: taxonomic composition of multiple microbiome samples\n",
    "microbiomes['Sample1'] = {\n",
    "    'None;Other': 0.075742067,\n",
    "    'Euryarchaeota': 0.049116324,\n",
    "    'Actinobacteria': 0.004214359,\n",
    "    'Bacteroidetes': 0.084114591,\n",
    "    'Chlamydiae': 5.78339E-05,\n",
    "    'Chloroflexi': 0.007700276,\n",
    "    'Cyanobacteria': 0.005267594,\n",
    "    'Elusimicrobia': 0.000758561,\n",
    "    'Fibrobacteres': 0.010849173,\n",
    "    'Firmicutes': 0.704400609,\n",
    "    'Fusobacteria': 2.9331E-05,\n",
    "    'Lentisphaerae': 0.006111405,\n",
    "    'NA': 0.003288403,\n",
    "    'Nitrospirae': 3.57752E-05,\n",
    "    'Planctomycetes': 0.009601286,\n",
    "    'Proteobacteria': 0.006894289,\n",
    "    'Saccharibacteria': 1.51341E-05,\n",
    "    'Spirochaetae': 0.001372058,\n",
    "    'Synergistetes': 0.019067822,\n",
    "    'Tenericutes': 0.004268445,\n",
    "    'Verrucomicrobia': 0.007094664\n",
    "}\n",
    "\n",
    "# Set the file path to save the files\n",
    "file_path = \"/Users/dimitriosgiakoumas/Library/CloudStorage/OneDrive-FloridaAtlanticUniversity/\"\n",
    "\n",
    "# Set the DPI of the saved image\n",
    "dpi = 300\n",
    "\n",
    "# Set the size of the plot\n",
    "width = 1400\n",
    "height = 600\n",
    "\n",
    "# Create a pie chart for each sample\n",
    "for sample in microbiomes:\n",
    "    # Get the taxonomic composition of the sample\n",
    "    composition = microbiomes[sample]\n",
    "\n",
    "    # Create a list of labels and values for the pie chart\n",
    "    labels = []\n",
    "    values = []\n",
    "    for domain in composition:\n",
    "        percentage = composition[domain] * 100\n",
    "        labels.append('{}: {:.2f}%'.format(domain, percentage))\n",
    "        values.append(composition[domain])\n",
    "\n",
    "    # Create a pie chart using Plotly\n",
    "    fig = go.Figure(data=[go.Pie(labels=labels, values=values)])\n",
    "    fig.update_layout(title_text='Total Taxonomic Composition', legend=dict(orientation=\"h\", x=-1, y=.8), width=width, height=height)\n",
    "\n",
    "    # Save the plot as a PNG file with the specified DPI\n",
    "    fig.write_image(file_path + \"plot.png\", scale=dpi/72)\n",
    "\n",
    "    # Convert the PNG file to a TIFF file with the specified DPI\n",
    "    with Image.open(file_path + \"plot.png\") as im:\n",
    "        im.save(file_path + \"plot.tiff\", dpi=(dpi, dpi))\n"
   ],
   "outputs": [],
   "execution_count": 21
  },
  {
   "metadata": {},
   "cell_type": "code",
   "outputs": [],
   "execution_count": null,
   "source": ""
  }
 ],
 "metadata": {
  "kernelspec": {
   "name": "python3",
   "language": "python",
   "display_name": "Python 3 (ipykernel)"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
